// igniteui-angular-charts/esm5/lib/igx-weighted-close-indicator-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxItemwiseStrategyBasedIndicatorComponent } from "./igx-itemwise-strategy-based-indicator-component";
import { IgxStrategyBasedIndicatorComponent } from "./igx-strategy-based-indicator-component";
import { IgxFinancialIndicatorComponent } from "./igx-financial-indicator-component";
import { IgxFinancialSeriesComponent } from "./igx-financial-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { WeightedCloseIndicator } from "./WeightedCloseIndicator";
/**
 * Represents a IgxDataChartComponent Weigted Close indicator series.
 * The weighted close indicator shows an average of the high low and close
 * for a day but with the closing price counted twice in the average.
 * Default required members: High, Low, Close
*
* The `IgxWeightedCloseIndicatorComponent` is similar to the `IgxTypicalPriceIndicatorComponent` in that it represents an average of the high price, low price, and closing price for a day.
* However, with the `IgxWeightedCloseIndicatorComponent`, more emphasis is placed on the closing price and it is included twice when calculating the arithmetic average.
*
* ```html
* <igx-data-chart
*     [dataSource]="data">
*     <igx-category-x-axis #xAxis
*         label="label">
*     </igx-category-x-axis>
*     <igx-numeric-y-axis #yAxis>
*     </igx-numeric-y-axis>
*     <igx-weighted-close-indicator
*         [xAxis]="xAxis"
*         [yAxis]="yAxis"
*         openMemberPath="open"
*         highMemberPath="high"
*         lowMemberPath="low"
*         closeMemberPath="close">
*     </igx-weighted-close-indicator>
* </igx-data-chart>
* ```
*
* ```ts
* let series = new IgxWeightedCloseIndicatorComponent();
* series.xAxis = this.xAxis;
* series.yAxis = this.yAxis;
* series.openMemberPath = "open";
* series.highMemberPath = "high";
* series.lowMemberPath = "low";
* series.closeMemberPath = "close";
* this.chart.series.add(series);
* ```
*/
export var IgxWeightedCloseIndicatorComponent_PROVIDERS = [{ provide: IgxItemwiseStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxWeightedCloseIndicatorComponent; }) }, { provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxWeightedCloseIndicatorComponent; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxWeightedCloseIndicatorComponent; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxWeightedCloseIndicatorComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxWeightedCloseIndicatorComponent; }) }];
var IgxWeightedCloseIndicatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxWeightedCloseIndicatorComponent, _super);
    function IgxWeightedCloseIndicatorComponent() {
        return _super.call(this) || this;
    }
    IgxWeightedCloseIndicatorComponent_1 = IgxWeightedCloseIndicatorComponent;
    IgxWeightedCloseIndicatorComponent.prototype.createImplementation = function () {
        return new WeightedCloseIndicator();
    };
    Object.defineProperty(IgxWeightedCloseIndicatorComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    var IgxWeightedCloseIndicatorComponent_1;
    IgxWeightedCloseIndicatorComponent = IgxWeightedCloseIndicatorComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-weighted-close-indicator',
            template: "",
            providers: [{ provide: IgxItemwiseStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxWeightedCloseIndicatorComponent_1; }) }, { provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxWeightedCloseIndicatorComponent_1; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxWeightedCloseIndicatorComponent_1; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxWeightedCloseIndicatorComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxWeightedCloseIndicatorComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxWeightedCloseIndicatorComponent);
    return IgxWeightedCloseIndicatorComponent;
}(IgxItemwiseStrategyBasedIndicatorComponent));
export { IgxWeightedCloseIndicatorComponent };
