// igniteui-angular-charts/esm5/lib/igx-spline-series-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxSplineSeriesBaseComponent } from "./igx-spline-series-base-component";
import { IgxHorizontalAnchoredCategorySeriesComponent } from "./igx-horizontal-anchored-category-series-component";
import { IgxAnchoredCategorySeriesComponent } from "./igx-anchored-category-series-component";
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { SplineSeries } from "./SplineSeries";
/**
 * Represents a IgxDataChartComponent spline series.
*
* ```html
* <igx-data-chart
*     [dataSource]="data">
*     <igx-category-x-axis
*         label="label"
*         #xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
*         #yAxis>
*     </igx-numeric-y-axis>
*     <igx-spline-series
*         [xAxis]="xAxis"
*         [yAxis]="yAxis"
*         valueMemberPath="value">
*     </igx-spline-series>
* </igx-data-chart>
* ```
*/
export var IgxSplineSeriesComponent_PROVIDERS = [{ provide: IgxSplineSeriesBaseComponent, useExisting: forwardRef(function () { return IgxSplineSeriesComponent; }) }, { provide: IgxHorizontalAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxSplineSeriesComponent; }) }, { provide: IgxAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxSplineSeriesComponent; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxSplineSeriesComponent; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxSplineSeriesComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxSplineSeriesComponent; }) }];
var IgxSplineSeriesComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxSplineSeriesComponent, _super);
    function IgxSplineSeriesComponent() {
        return _super.call(this) || this;
    }
    IgxSplineSeriesComponent_1 = IgxSplineSeriesComponent;
    IgxSplineSeriesComponent.prototype.createImplementation = function () {
        return new SplineSeries();
    };
    Object.defineProperty(IgxSplineSeriesComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    var IgxSplineSeriesComponent_1;
    IgxSplineSeriesComponent = IgxSplineSeriesComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-spline-series',
            template: "",
            providers: [{ provide: IgxSplineSeriesBaseComponent, useExisting: forwardRef(function () { return IgxSplineSeriesComponent_1; }) }, { provide: IgxHorizontalAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxSplineSeriesComponent_1; }) }, { provide: IgxAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxSplineSeriesComponent_1; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxSplineSeriesComponent_1; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxSplineSeriesComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxSplineSeriesComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxSplineSeriesComponent);
    return IgxSplineSeriesComponent;
}(IgxSplineSeriesBaseComponent));
export { IgxSplineSeriesComponent };
