// igniteui-angular-charts/esm5/lib/igx-stacked-100-line-series-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxStackedLineSeriesComponent } from "./igx-stacked-line-series-component";
import { IgxHorizontalStackedSeriesBaseComponent } from "./igx-horizontal-stacked-series-base-component";
import { IgxStackedSeriesBaseComponent } from "./igx-stacked-series-base-component";
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { Stacked100LineSeries } from "./Stacked100LineSeries";
/**
 * Represents a IgxDataChartComponent stacked100 line series.
*/
export var IgxStacked100LineSeriesComponent_PROVIDERS = [{ provide: IgxStackedLineSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100LineSeriesComponent; }) }, { provide: IgxHorizontalStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStacked100LineSeriesComponent; }) }, { provide: IgxStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStacked100LineSeriesComponent; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxStacked100LineSeriesComponent; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100LineSeriesComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100LineSeriesComponent; }) }];
var IgxStacked100LineSeriesComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxStacked100LineSeriesComponent, _super);
    function IgxStacked100LineSeriesComponent() {
        return _super.call(this) || this;
    }
    IgxStacked100LineSeriesComponent_1 = IgxStacked100LineSeriesComponent;
    IgxStacked100LineSeriesComponent.prototype.createImplementation = function () {
        return new Stacked100LineSeries();
    };
    Object.defineProperty(IgxStacked100LineSeriesComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    var IgxStacked100LineSeriesComponent_1;
    IgxStacked100LineSeriesComponent = IgxStacked100LineSeriesComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-stacked-100-line-series',
            template: "",
            providers: [{ provide: IgxStackedLineSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100LineSeriesComponent_1; }) }, { provide: IgxHorizontalStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStacked100LineSeriesComponent_1; }) }, { provide: IgxStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStacked100LineSeriesComponent_1; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxStacked100LineSeriesComponent_1; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100LineSeriesComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100LineSeriesComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxStacked100LineSeriesComponent);
    return IgxStacked100LineSeriesComponent;
}(IgxStackedLineSeriesComponent));
export { IgxStacked100LineSeriesComponent };
