// igniteui-angular-charts/esm5/lib/igx-detrended-price-oscillator-indicator-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxStrategyBasedIndicatorComponent } from "./igx-strategy-based-indicator-component";
import { IgxFinancialIndicatorComponent } from "./igx-financial-indicator-component";
import { IgxFinancialSeriesComponent } from "./igx-financial-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { DetrendedPriceOscillatorIndicator } from "./DetrendedPriceOscillatorIndicator";
/**
 * Represents a IgxDataChartComponent Detrended Price Oscillator indicator series.
 * Default required members: Close
*
* The `IgxDetrendedPriceOscillatorIndicatorComponent` class represents a IgxDataChartComponent Detrended Price Oscillator indicator series.
*
* ```html
* <igx-data-chart
*     [dataSource]="data" >
*     <igx-category-x-axis #xAxis
*         label="label">
*     </igx-category-x-axis>
*     <igx-numeric-y-axis #yAxis>
*     </igx-numeric-y-axis>
*     <igx-detrended-price-oscillator-indicator
*         [xAxis]="xAxis"
*         [yAxis]="yAxis"
*         openMemberPath="open"
*         highMemberPath="high"
*         lowMemberPath="low"
*         closeMemberPath="close">
*     </igx-detrended-price-oscillator-indicator>
* </igx-data-chart>
* ```
*
* ```ts
* let series = new IgxDetrendedPriceOscillatorIndicatorComponent();
* series.xAxis = this.xAxis;
* series.yAxis = this.yAxis;
* series.openMemberPath = "open";
* series.highMemberPath = "high";
* series.lowMemberPath = "low";
* series.closeMemberPath = "close";
* this.chart.series.add(series);
* ```
*/
export var IgxDetrendedPriceOscillatorIndicatorComponent_PROVIDERS = [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxDetrendedPriceOscillatorIndicatorComponent; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxDetrendedPriceOscillatorIndicatorComponent; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxDetrendedPriceOscillatorIndicatorComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxDetrendedPriceOscillatorIndicatorComponent; }) }];
var IgxDetrendedPriceOscillatorIndicatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxDetrendedPriceOscillatorIndicatorComponent, _super);
    function IgxDetrendedPriceOscillatorIndicatorComponent() {
        return _super.call(this) || this;
    }
    IgxDetrendedPriceOscillatorIndicatorComponent_1 = IgxDetrendedPriceOscillatorIndicatorComponent;
    IgxDetrendedPriceOscillatorIndicatorComponent.prototype.createImplementation = function () {
        return new DetrendedPriceOscillatorIndicator();
    };
    Object.defineProperty(IgxDetrendedPriceOscillatorIndicatorComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxDetrendedPriceOscillatorIndicatorComponent.prototype, "period", {
        /**
         * Gets or sets the moving average period for the current DetrendedPriceOscillatorIndicator object.
         * The typical, and initial, value for DPO periods is 20.
        *
        * You can use the `period` to set the current moving average period.
        *
        * ```html
        * <igx-data-chart
        *     [dataSource]="data" >
        *     <igx-category-x-axis #xAxis
        *         label="label">
        *     </igx-category-x-axis>
        *     <igx-numeric-y-axis #yAxis>
        *     </igx-numeric-y-axis>
        *     <igx-detrended-price-oscillator-indicator
        *         [xAxis]="xAxis"
        *         [yAxis]="yAxis"
        *         openMemberPath="open"
        *         highMemberPath="high"
        *         lowMemberPath="low"
        *         closeMemberPath="close"
        * 		period=30>
        *     </igx-detrended-price-oscillator-indicator>
        * </igx-data-chart>
        * ```
        *
        * ```ts
        * this.series.period = 30;
        * ```
        */
        get: function () {
            return this.i.qy;
        },
        set: function (v) {
            this.i.qy = +v;
        },
        enumerable: true,
        configurable: true
    });
    var IgxDetrendedPriceOscillatorIndicatorComponent_1;
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxDetrendedPriceOscillatorIndicatorComponent.prototype, "period", null);
    IgxDetrendedPriceOscillatorIndicatorComponent = IgxDetrendedPriceOscillatorIndicatorComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-detrended-price-oscillator-indicator',
            template: "",
            providers: [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxDetrendedPriceOscillatorIndicatorComponent_1; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxDetrendedPriceOscillatorIndicatorComponent_1; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxDetrendedPriceOscillatorIndicatorComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxDetrendedPriceOscillatorIndicatorComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxDetrendedPriceOscillatorIndicatorComponent);
    return IgxDetrendedPriceOscillatorIndicatorComponent;
}(IgxStrategyBasedIndicatorComponent));
export { IgxDetrendedPriceOscillatorIndicatorComponent };
