// igniteui-angular-charts/esm5/lib/igx-stacked-100-column-series-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxStackedColumnSeriesComponent } from "./igx-stacked-column-series-component";
import { IgxHorizontalStackedSeriesBaseComponent } from "./igx-horizontal-stacked-series-base-component";
import { IgxStackedSeriesBaseComponent } from "./igx-stacked-series-base-component";
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { Stacked100ColumnSeries } from "./Stacked100ColumnSeries";
/**
 * Represents a IgxDataChartComponent stacked100 column series.
*/
export var IgxStacked100ColumnSeriesComponent_PROVIDERS = [{ provide: IgxStackedColumnSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100ColumnSeriesComponent; }) }, { provide: IgxHorizontalStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStacked100ColumnSeriesComponent; }) }, { provide: IgxStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStacked100ColumnSeriesComponent; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxStacked100ColumnSeriesComponent; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100ColumnSeriesComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100ColumnSeriesComponent; }) }];
var IgxStacked100ColumnSeriesComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxStacked100ColumnSeriesComponent, _super);
    function IgxStacked100ColumnSeriesComponent() {
        return _super.call(this) || this;
    }
    IgxStacked100ColumnSeriesComponent_1 = IgxStacked100ColumnSeriesComponent;
    IgxStacked100ColumnSeriesComponent.prototype.createImplementation = function () {
        return new Stacked100ColumnSeries();
    };
    Object.defineProperty(IgxStacked100ColumnSeriesComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    var IgxStacked100ColumnSeriesComponent_1;
    IgxStacked100ColumnSeriesComponent = IgxStacked100ColumnSeriesComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-stacked-100-column-series',
            template: "",
            providers: [{ provide: IgxStackedColumnSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100ColumnSeriesComponent_1; }) }, { provide: IgxHorizontalStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStacked100ColumnSeriesComponent_1; }) }, { provide: IgxStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStacked100ColumnSeriesComponent_1; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxStacked100ColumnSeriesComponent_1; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100ColumnSeriesComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100ColumnSeriesComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxStacked100ColumnSeriesComponent);
    return IgxStacked100ColumnSeriesComponent;
}(IgxStackedColumnSeriesComponent));
export { IgxStacked100ColumnSeriesComponent };
