// igniteui-angular-charts/esm5/lib/igx-bollinger-band-width-indicator-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxStrategyBasedIndicatorComponent } from "./igx-strategy-based-indicator-component";
import { IgxFinancialIndicatorComponent } from "./igx-financial-indicator-component";
import { IgxFinancialSeriesComponent } from "./igx-financial-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { BollingerBandWidthIndicator } from "./BollingerBandWidthIndicator";
/**
 * Represents a IgxDataChartComponent Bollinger Bandwidth indicator series.
 * Represents the normalized width of the Bollinger bands for each provided value.
 * For more info see: BollingerBandsOverlay
 * Default required members: High, Low, Close
*
* You can use the `IgxBollingerBandWidthIndicatorComponent` to represents a IgxDataChartComponent Bollinger Bandwidth indicator series.
*
* ```html
* <igx-data-chart
* 	#chart
*     [dataSource]="data" >
*     <igx-category-x-axis
* 		label="label"
* 		#xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
* 		#yAxis>
*     </igx-numeric-y-axis>
*     <igx-bollinger-band-width-indicator
* 		[xAxis]="xAxis"
* 		[yAxis]="yAxis"
* 		openMemberPath="open"
* 		highMemberPath="high"
* 		lowMemberPath="low"
* 		closeMemberPath="close">
*     </igx-bollinger-band-width-indicator>
* </igx-data-chart>
* ```
*
* ```ts
* let series = new IgxBollingerBandWidthIndicatorComponent();
* series.xAxis = this.xAxis;
* series.yAxis = this.yAxis;
* series.openMemberPath = "open";
* series.highMemberPath = "high";
* series.lowMemberPath = "low";
* series.closeMemberPath = "close";
* this.chart.series.add(series);
* ```
*/
export var IgxBollingerBandWidthIndicatorComponent_PROVIDERS = [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxBollingerBandWidthIndicatorComponent; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxBollingerBandWidthIndicatorComponent; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxBollingerBandWidthIndicatorComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxBollingerBandWidthIndicatorComponent; }) }];
var IgxBollingerBandWidthIndicatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxBollingerBandWidthIndicatorComponent, _super);
    function IgxBollingerBandWidthIndicatorComponent() {
        return _super.call(this) || this;
    }
    IgxBollingerBandWidthIndicatorComponent_1 = IgxBollingerBandWidthIndicatorComponent;
    IgxBollingerBandWidthIndicatorComponent.prototype.createImplementation = function () {
        return new BollingerBandWidthIndicator();
    };
    Object.defineProperty(IgxBollingerBandWidthIndicatorComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxBollingerBandWidthIndicatorComponent.prototype, "period", {
        /**
         * Gets or sets the moving average period for the Bollinger Band Width Indicator.
         * The typical, and initial, value for CCI periods is 20.
        *
        * You can use the `period` to set the moving average.
        *
        * ```html
        * <igx-data-chart
        * 	#chart
        *     [dataSource]="data" >
        *     <igx-category-x-axis
        * 		label="label"
        * 		#xAxis>
        *     </igx-category-x-axis>
        *     <igx-numeric-y-axis
        * 		#yAxis>
        *     </igx-numeric-y-axis>
        *     <igx-bollinger-band-width-indicator
        * 		[xAxis]="xAxis"
        * 		[yAxis]="yAxis"
        * 		openMemberPath="open"
        * 		highMemberPath="high"
        * 		lowMemberPath="low"
        * 		closeMemberPath="close"
        * 		period="14">
        *     </igx-bollinger-band-width-indicator>
        * </igx-data-chart>
        * ```
        *
        * ```ts
        * this.series.period = 14;
        * ```
        */
        get: function () {
            return this.i.qz;
        },
        set: function (v) {
            this.i.qz = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxBollingerBandWidthIndicatorComponent.prototype, "multiplier", {
        /**
         * Gets or sets the multiplier for the Bollinger Band width.
         * The typical, and initial, value for Bollinger Band width multiplier is 2.
        *
        * Use the `multiplier` property for the Bollinger Band width. The typical, and initial, value for Bollinger Band width multiplier is 2.
        *
        * ```ts
        * this.series.multiplier = 14;
        * ```
        */
        get: function () {
            return this.i.qy;
        },
        set: function (v) {
            this.i.qy = +v;
        },
        enumerable: true,
        configurable: true
    });
    var IgxBollingerBandWidthIndicatorComponent_1;
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxBollingerBandWidthIndicatorComponent.prototype, "period", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxBollingerBandWidthIndicatorComponent.prototype, "multiplier", null);
    IgxBollingerBandWidthIndicatorComponent = IgxBollingerBandWidthIndicatorComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-bollinger-band-width-indicator',
            template: "",
            providers: [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxBollingerBandWidthIndicatorComponent_1; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxBollingerBandWidthIndicatorComponent_1; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxBollingerBandWidthIndicatorComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxBollingerBandWidthIndicatorComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxBollingerBandWidthIndicatorComponent);
    return IgxBollingerBandWidthIndicatorComponent;
}(IgxStrategyBasedIndicatorComponent));
export { IgxBollingerBandWidthIndicatorComponent };
