// igniteui-angular-charts/esm5/lib/igx-spline-area-fragment-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxSplineFragmentBaseComponent } from "./igx-spline-fragment-base-component";
import { IgxSplineSeriesBaseComponent } from "./igx-spline-series-base-component";
import { IgxHorizontalAnchoredCategorySeriesComponent } from "./igx-horizontal-anchored-category-series-component";
import { IgxAnchoredCategorySeriesComponent } from "./igx-anchored-category-series-component";
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { SplineAreaFragment } from "./SplineAreaFragment";
/**
 * Represents one part of a StackedSplineAreaSeries.
*/
export var IgxSplineAreaFragmentComponent_PROVIDERS = [{ provide: IgxSplineFragmentBaseComponent, useExisting: forwardRef(function () { return IgxSplineAreaFragmentComponent; }) }, { provide: IgxSplineSeriesBaseComponent, useExisting: forwardRef(function () { return IgxSplineAreaFragmentComponent; }) }, { provide: IgxHorizontalAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxSplineAreaFragmentComponent; }) }, { provide: IgxAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxSplineAreaFragmentComponent; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxSplineAreaFragmentComponent; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxSplineAreaFragmentComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxSplineAreaFragmentComponent; }) }];
var IgxSplineAreaFragmentComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxSplineAreaFragmentComponent, _super);
    function IgxSplineAreaFragmentComponent() {
        return _super.call(this) || this;
    }
    IgxSplineAreaFragmentComponent_1 = IgxSplineAreaFragmentComponent;
    IgxSplineAreaFragmentComponent.prototype.createImplementation = function () {
        return new SplineAreaFragment();
    };
    Object.defineProperty(IgxSplineAreaFragmentComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxSplineAreaFragmentComponent.prototype, "isAreaOrLine", {
        /**
         * Gets whether the current series shows an area or line shape.
        */
        get: function () {
            return this.i.ck;
        },
        enumerable: true,
        configurable: true
    });
    var IgxSplineAreaFragmentComponent_1;
    IgxSplineAreaFragmentComponent = IgxSplineAreaFragmentComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-spline-area-fragment',
            template: "",
            providers: [{ provide: IgxSplineFragmentBaseComponent, useExisting: forwardRef(function () { return IgxSplineAreaFragmentComponent_1; }) }, { provide: IgxSplineSeriesBaseComponent, useExisting: forwardRef(function () { return IgxSplineAreaFragmentComponent_1; }) }, { provide: IgxHorizontalAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxSplineAreaFragmentComponent_1; }) }, { provide: IgxAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxSplineAreaFragmentComponent_1; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxSplineAreaFragmentComponent_1; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxSplineAreaFragmentComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxSplineAreaFragmentComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxSplineAreaFragmentComponent);
    return IgxSplineAreaFragmentComponent;
}(IgxSplineFragmentBaseComponent));
export { IgxSplineAreaFragmentComponent };
