// igniteui-angular-charts/esm5/lib/igx-item-tool-tip-layer-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxSeriesComponent } from "./igx-series-component";
import { IgxAnnotationLayerComponent } from "./igx-annotation-layer-component";
import { ItemToolTipLayer } from "./ItemToolTipLayer";
import { ensureBool } from "igniteui-angular-core";
/**
 * Represents an annotation layer that displays tooltips for all target series individually.
*/
export var IgxItemToolTipLayerComponent_PROVIDERS = [{ provide: IgxAnnotationLayerComponent, useExisting: forwardRef(function () { return IgxItemToolTipLayerComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxItemToolTipLayerComponent; }) }];
var IgxItemToolTipLayerComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxItemToolTipLayerComponent, _super);
    function IgxItemToolTipLayerComponent() {
        return _super.call(this) || this;
    }
    IgxItemToolTipLayerComponent_1 = IgxItemToolTipLayerComponent;
    IgxItemToolTipLayerComponent.prototype.createImplementation = function () {
        return new ItemToolTipLayer();
    };
    Object.defineProperty(IgxItemToolTipLayerComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxItemToolTipLayerComponent.prototype, "isDefaultTooltipBehaviorDisabled", {
        /**
         * Gets whether the default tooltip behaviors for the chart are disabled if this layer is present.
        */
        get: function () {
            return this.i.cn;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxItemToolTipLayerComponent.prototype, "targetSeries", {
        /**
         * Gets or sets the series to target this annotation to. If null, this annotation targets all series simultaneously.
        */
        get: function () {
            if (this.i.m6 == null) {
                return null;
            }
            if (!this.i.m6.externalObject) {
                var e = IgxSeriesComponent._createFromInternal(this.i.m6);
                if (e) {
                    e._implementation = this.i.m6;
                }
                this.i.m6.externalObject = e;
            }
            return this.i.m6.externalObject;
        },
        set: function (v) {
            if (v != null && this._stylingContainer && v._styling)
                v._styling(this._stylingContainer, this, this);
            v == null ? this.i.m6 = null : this.i.m6 = v.i;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxItemToolTipLayerComponent.prototype, "useInterpolation", {
        /**
         * Gets or sets whether to use value interpolation when drawing the tooltips.
        */
        get: function () {
            return this.i.m9;
        },
        set: function (v) {
            this.i.m9 = ensureBool(v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxItemToolTipLayerComponent.prototype, "skipUnknownValues", {
        /**
         * Gets or sets whether to skip past unknown values when searching for series values.
        */
        get: function () {
            return this.i.m8;
        },
        set: function (v) {
            this.i.m8 = ensureBool(v);
        },
        enumerable: true,
        configurable: true
    });
    IgxItemToolTipLayerComponent.prototype.findByName = function (name) {
        var baseResult = _super.prototype.findByName.call(this, name);
        if (baseResult) {
            return baseResult;
        }
        if (this.targetSeries && this.targetSeries.name && this.targetSeries.name == name) {
            return this.targetSeries;
        }
        return null;
    };
    IgxItemToolTipLayerComponent.prototype._styling = function (container, component, parent) {
        _super.prototype._styling.call(this, container, component, parent);
        this._inStyling = true;
        if (this.targetSeries && this.targetSeries._styling) {
            this.targetSeries._styling(container, component, this);
        }
        this._inStyling = false;
    };
    /**
     * Hides any tooltips presented by the layer, if any.
    
    */
    IgxItemToolTipLayerComponent.prototype.hideToolTips = function () {
        this.i.ib();
    };
    var IgxItemToolTipLayerComponent_1;
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", IgxSeriesComponent),
        tslib_1.__metadata("design:paramtypes", [IgxSeriesComponent])
    ], IgxItemToolTipLayerComponent.prototype, "targetSeries", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [Boolean])
    ], IgxItemToolTipLayerComponent.prototype, "useInterpolation", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [Boolean])
    ], IgxItemToolTipLayerComponent.prototype, "skipUnknownValues", null);
    IgxItemToolTipLayerComponent = IgxItemToolTipLayerComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-item-tool-tip-layer',
            template: "",
            providers: [{ provide: IgxAnnotationLayerComponent, useExisting: forwardRef(function () { return IgxItemToolTipLayerComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxItemToolTipLayerComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxItemToolTipLayerComponent);
    return IgxItemToolTipLayerComponent;
}(IgxAnnotationLayerComponent));
export { IgxItemToolTipLayerComponent };
