// igniteui-angular-charts/esm5/lib/igx-line-fragment-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxFragmentBaseComponent } from "./igx-fragment-base-component";
import { IgxHorizontalAnchoredCategorySeriesComponent } from "./igx-horizontal-anchored-category-series-component";
import { IgxAnchoredCategorySeriesComponent } from "./igx-anchored-category-series-component";
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { LineFragment } from "./LineFragment";
/**
 * Represents one part of a StackedLineSeries.
*/
export var IgxLineFragmentComponent_PROVIDERS = [{ provide: IgxFragmentBaseComponent, useExisting: forwardRef(function () { return IgxLineFragmentComponent; }) }, { provide: IgxHorizontalAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxLineFragmentComponent; }) }, { provide: IgxAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxLineFragmentComponent; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxLineFragmentComponent; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxLineFragmentComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxLineFragmentComponent; }) }];
var IgxLineFragmentComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxLineFragmentComponent, _super);
    function IgxLineFragmentComponent() {
        return _super.call(this) || this;
    }
    IgxLineFragmentComponent_1 = IgxLineFragmentComponent;
    IgxLineFragmentComponent.prototype.createImplementation = function () {
        return new LineFragment();
    };
    Object.defineProperty(IgxLineFragmentComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxLineFragmentComponent.prototype, "isAreaOrLine", {
        /**
         * Gets whether the current series shows an area or line shape.
        */
        get: function () {
            return this.i.ck;
        },
        enumerable: true,
        configurable: true
    });
    var IgxLineFragmentComponent_1;
    IgxLineFragmentComponent = IgxLineFragmentComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-line-fragment',
            template: "",
            providers: [{ provide: IgxFragmentBaseComponent, useExisting: forwardRef(function () { return IgxLineFragmentComponent_1; }) }, { provide: IgxHorizontalAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxLineFragmentComponent_1; }) }, { provide: IgxAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxLineFragmentComponent_1; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxLineFragmentComponent_1; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxLineFragmentComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxLineFragmentComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxLineFragmentComponent);
    return IgxLineFragmentComponent;
}(IgxFragmentBaseComponent));
export { IgxLineFragmentComponent };
