// igniteui-angular-charts/esm5/lib/igx-moving-average-convergence-divergence-indicator-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxStrategyBasedIndicatorComponent } from "./igx-strategy-based-indicator-component";
import { IgxFinancialIndicatorComponent } from "./igx-financial-indicator-component";
import { IgxFinancialSeriesComponent } from "./igx-financial-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { MovingAverageConvergenceDivergenceIndicator } from "./MovingAverageConvergenceDivergenceIndicator";
/**
 * Represents a IgxDataChartComponent Moving Average Convergence/Divergence (MACD) indicator series.
 * Default required members: High, Low, Close
*
* Represents a Ignite UIDataChart Moving Average Convergence/Divergence (MACD) indicator series.
*
* Example:
*
* ```html
* <igx-data-chart
*     [dataSource]="data">
* 	<igx-category-x-axis
* 		label="label"
* 		#xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
* 		#yAxis>
*     </igx-numeric-y-axis>
* 	<igx-moving-average-convergence-divergence-indicator
*         #series
*         [xAxis]="xAxis"
*         [yAxis]="yAxis"
* 		openMemberPath="open"
*         highMemberPath="high"
* 		lowMemberPath="low"
* 		closeMemberPath="close"
* 		volumeMemberPath="volume">
*     </gx-moving-average-convergence-divergence-indicatorr>
* </igx-data-chart>
* ```
*
* ```ts
* let series = new IgxMovingAverageConvergenceDivergenceIndicatorComponent();
* series.xAxis = this.xAxis;
* series.yAxis = this.yAxis;
* series.openMemberPath="open"
* series.highMemberPath = "high";
* series.lowMemberPath = "low";
* series.closeMemberPath="close"
* series.volumeMemberPath= "volume";
* this.chart.series.add(series);
* ```
*/
export var IgxMovingAverageConvergenceDivergenceIndicatorComponent_PROVIDERS = [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxMovingAverageConvergenceDivergenceIndicatorComponent; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxMovingAverageConvergenceDivergenceIndicatorComponent; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxMovingAverageConvergenceDivergenceIndicatorComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxMovingAverageConvergenceDivergenceIndicatorComponent; }) }];
var IgxMovingAverageConvergenceDivergenceIndicatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxMovingAverageConvergenceDivergenceIndicatorComponent, _super);
    function IgxMovingAverageConvergenceDivergenceIndicatorComponent() {
        return _super.call(this) || this;
    }
    IgxMovingAverageConvergenceDivergenceIndicatorComponent_1 = IgxMovingAverageConvergenceDivergenceIndicatorComponent;
    IgxMovingAverageConvergenceDivergenceIndicatorComponent.prototype.createImplementation = function () {
        return new MovingAverageConvergenceDivergenceIndicator();
    };
    Object.defineProperty(IgxMovingAverageConvergenceDivergenceIndicatorComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxMovingAverageConvergenceDivergenceIndicatorComponent.prototype, "shortPeriod", {
        /**
         * Gets or sets the short moving average period for the current MovingAverageConvergenceDivergenceIndicator object.
         * The typical, and initial, value for short MACD periods is 10.
        *
        * Gets or sets the short moving average period for the current MovingAverageConvergenceDivergenceIndicator object.
        *
        * Example:
        *
        * ```html
        * <igx-data-chart
        *     [dataSource]="data">
        * 	<igx-category-x-axis
        * 		label="label"
        * 		#xAxis>
        *     </igx-category-x-axis>
        *     <igx-numeric-y-axis
        * 		#yAxis>
        *     </igx-numeric-y-axis>
        * 	<igx-moving-average-convergence-divergence-indicator
        *         #series
        *         [xAxis]="xAxis"
        *         [yAxis]="yAxis"
        * 		openMemberPath="open"
        *         highMemberPath="high"
        * 		lowMemberPath="low"
        * 		closeMemberPath="close"
        * 		volumeMemberPath="volume"
        * 		shortPeriod="10">
        *     </gx-moving-average-convergence-divergence-indicatorr>
        * </igx-data-chart>
        * ```
        */
        get: function () {
            return this.i.shortPeriod;
        },
        set: function (v) {
            this.i.shortPeriod = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxMovingAverageConvergenceDivergenceIndicatorComponent.prototype, "longPeriod", {
        /**
         * Gets or sets the long moving average period for the current MovingAverageConvergenceDivergenceIndicator object.
         * The typical, and initial, value for long MACD periods is 30.
        *
        * Gets or sets the long moving average period for the current MovingAverageConvergenceDivergenceIndicator object.
        *
        * Example:
        *
        * ```html
        * <igx-data-chart
        *     [dataSource]="data">
        * 	<igx-category-x-axis
        * 		label="label"
        * 		#xAxis>
        *     </igx-category-x-axis>
        *     <igx-numeric-y-axis
        * 		#yAxis>
        *     </igx-numeric-y-axis>
        * 	<igx-moving-average-convergence-divergence-indicator
        *         #series
        *         [xAxis]="xAxis"
        *         [yAxis]="yAxis"
        * 		openMemberPath="open"
        *         highMemberPath="high"
        * 		lowMemberPath="low"
        * 		closeMemberPath="close"
        * 		volumeMemberPath="volume"
        * 		longPeriod="30">
        *     </gx-moving-average-convergence-divergence-indicatorr>
        * </igx-data-chart>
        * ```
        *
        * ```ts
        * this.series.longPeriod="30" ;
        * ```
        */
        get: function () {
            return this.i.longPeriod;
        },
        set: function (v) {
            this.i.longPeriod = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxMovingAverageConvergenceDivergenceIndicatorComponent.prototype, "signalPeriod", {
        /**
         * Gets or sets the long moving average period for the current MovingAverageConvergenceDivergenceIndicator object.
         * The typical, and initial, value for long PVO periods is 30.
        *
        * Gets or sets the long moving average period for the current MovingAverageConvergenceDivergenceIndicator object.
        *
        * Example:
        *
        * ```html
        * <igx-data-chart
        *     [dataSource]="data">
        * 	<igx-category-x-axis
        * 		label="label"
        * 		#xAxis>
        *     </igx-category-x-axis>
        *     <igx-numeric-y-axis
        * 		#yAxis>
        *     </igx-numeric-y-axis>
        * 	<igx-moving-average-convergence-divergence-indicator
        *         #series
        *         [xAxis]="xAxis"
        *         [yAxis]="yAxis"
        * 		openMemberPath="open"
        *         highMemberPath="high"
        * 		lowMemberPath="low"
        * 		closeMemberPath="close"
        * 		volumeMemberPath="volume"
        * 		signalPeriod="30">
        *     </gx-moving-average-convergence-divergence-indicatorr>
        * </igx-data-chart>
        * ```
        */
        get: function () {
            return this.i.signalPeriod;
        },
        set: function (v) {
            this.i.signalPeriod = +v;
        },
        enumerable: true,
        configurable: true
    });
    var IgxMovingAverageConvergenceDivergenceIndicatorComponent_1;
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxMovingAverageConvergenceDivergenceIndicatorComponent.prototype, "shortPeriod", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxMovingAverageConvergenceDivergenceIndicatorComponent.prototype, "longPeriod", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxMovingAverageConvergenceDivergenceIndicatorComponent.prototype, "signalPeriod", null);
    IgxMovingAverageConvergenceDivergenceIndicatorComponent = IgxMovingAverageConvergenceDivergenceIndicatorComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-moving-average-convergence-divergence-indicator',
            template: "",
            providers: [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxMovingAverageConvergenceDivergenceIndicatorComponent_1; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxMovingAverageConvergenceDivergenceIndicatorComponent_1; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxMovingAverageConvergenceDivergenceIndicatorComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxMovingAverageConvergenceDivergenceIndicatorComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxMovingAverageConvergenceDivergenceIndicatorComponent);
    return IgxMovingAverageConvergenceDivergenceIndicatorComponent;
}(IgxStrategyBasedIndicatorComponent));
export { IgxMovingAverageConvergenceDivergenceIndicatorComponent };
