// igniteui-angular-charts/esm5/lib/igx-pie-chart-base-component.js
import * as tslib_1 from "tslib";
import { EventEmitter, Output, Input } from '@angular/core';
import { delegateCombine } from "igniteui-angular-core";
import { LabelsPosition, LabelsPosition_$type } from "./LabelsPosition";
import { Visibility, Visibility_$type } from "igniteui-angular-core";
import { LeaderLineType, LeaderLineType_$type } from "./LeaderLineType";
import { OthersCategoryType, OthersCategoryType_$type } from "./OthersCategoryType";
import { IgxIndexCollection } from "./igx-index-collection";
import { SweepDirection, SweepDirection_$type } from "igniteui-angular-core";
import { SliceSelectionMode, SliceSelectionMode_$type } from "./SliceSelectionMode";
import { IgxObjectCollection } from "igniteui-angular-core";
import { IgxLabelClickEventArgs } from "./igx-label-click-event-args";
import { IgxSelectedItemChangingEventArgs } from "./igx-selected-item-changing-event-args";
import { IgxSelectedItemsChangingEventArgs } from "./igx-selected-items-changing-event-args";
import { IgxSelectedItemChangedEventArgs } from "./igx-selected-item-changed-event-args";
import { IgxSelectedItemsChangedEventArgs } from "./igx-selected-items-changed-event-args";
import { IgxSliceClickEventArgs } from "./igx-slice-click-event-args";
import { TypeRegistrar, Number_$type, Base } from "igniteui-angular-core";
import { NamePatcher, ensureEnum, brushToString, stringToBrush, ensureBool, toBrushCollection, fromBrushCollection, arrayFindByName, toSpinal, initializePropertiesFromCss, toPoint } from "igniteui-angular-core";
import { IndexCollection as IndexCollection_internal } from "./IndexCollection";
import { SyncableObservableCollection$1 } from "igniteui-angular-core";
import { Style } from "igniteui-angular-core";
import { ObjectCollection as ObjectCollection_internal } from "igniteui-angular-core";
/**
 * Represents the base class for the pie chart.
*/
export var IgxPieChartBaseComponent_PROVIDERS = [];
var IgxPieChartBaseComponent = /** @class */ (function () {
    function IgxPieChartBaseComponent() {
        this._explodedSlices = null;
        this._selectedItems = null;
        this.__p = null;
        this._hasUserValues = new Set();
        this._stylingContainer = null;
        this._stylingParent = null;
        this._inStyling = false;
        this._labelClick = null;
        this._selectedItemChanging = null;
        this._selectedItemsChanging = null;
        this._selectedItemChanged = null;
        this._selectedItemsChanged = null;
        this._sliceClick = null;
        this._zoneRunner = null;
        if (this._styling) {
            NamePatcher.ensureStylablePatched(Object.getPrototypeOf(this));
        }
        this._implementation = this.createImplementation();
        this._implementation.externalObject = this;
        this.onImplementationCreated();
    }
    IgxPieChartBaseComponent.prototype.createImplementation = function () {
        return null;
    };
    IgxPieChartBaseComponent.prototype.ngOnInit = function () {
    };
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "i", {
        /**
         * @hidden
         */
        get: function () {
            return this._implementation;
        } /**
                                     * @hidden
                                     */,
        enumerable: true,
        configurable: true
    });
    IgxPieChartBaseComponent._createFromInternal = function (internal) {
        if (!internal) {
            return null;
        }
        if (!internal.$type) {
            return null;
        }
        var name = internal.$type.name;
        var externalName = "Igx" + name + "Component";
        if (!TypeRegistrar.isRegistered(externalName)) {
            return null;
        }
        return TypeRegistrar.create(externalName);
    };
    IgxPieChartBaseComponent.prototype.onImplementationCreated = function () {
    };
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "innerExtent", {
        /**
         * Gets or sets the amount of space between the center and the start of the arc. This property should be set between 0 and 1.
         * If InnerExtent is set to a value greater than 1, it will be interpreted as a percentage, effectively divided by 100.
        */
        get: function () {
            return this.i.ct;
        },
        set: function (v) {
            this.i.ct = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "valueMemberPath", {
        /**
         * Gets or Sets the property name that contains the values.
        */
        get: function () {
            return this.i.e0;
        },
        set: function (v) {
            this.i.e0 = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "labelMemberPath", {
        /**
         * Gets or sets the property name that contains the labels.
        */
        get: function () {
            return this.i.du;
        },
        set: function (v) {
            this.i.du = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "legendLabelMemberPath", {
        /**
         * Gets or sets the property name that contains the legend labels.
        */
        get: function () {
            return this.i.d9;
        },
        set: function (v) {
            this.i.d9 = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "labelsPosition", {
        /**
         * Gets or sets the position of chart labels.
        */
        get: function () {
            return this.i.as;
        },
        set: function (v) {
            this.i.as = ensureEnum(LabelsPosition_$type, v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "labelOuterColor", {
        /**
         * Gets or sets the color for labels rendered outside of the pie chart.
        */
        get: function () {
            return brushToString(this.i.hq);
        },
        set: function (v) {
            this.i.hq = stringToBrush(v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "labelInnerColor", {
        /**
         * Gets or sets the color for labels rendered inside of the pie chart.
        */
        get: function () {
            return brushToString(this.i.hp);
        },
        set: function (v) {
            this.i.hp = stringToBrush(v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "leaderLineVisibility", {
        /**
         * Gets or sets whether the leader lines are visible.
        */
        get: function () {
            return this.i.h2;
        },
        set: function (v) {
            this.i.h2 = ensureEnum(Visibility_$type, v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "leaderLineType", {
        /**
         * Gets or sets what type of leader lines will be used for the outside end labels.
        */
        get: function () {
            return this.i.at;
        },
        set: function (v) {
            this.i.at = ensureEnum(LeaderLineType_$type, v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "leaderLineMargin", {
        /**
         * Gets or sets the margin between a label and the end of its leader line. The default is 6 pixels.
        */
        get: function () {
            return this.i.cv;
        },
        set: function (v) {
            this.i.cv = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "toolTip", {
        /**
         * Gets or sets the ToolTip for the chart.
        */
        get: function () {
            return this.i.c8;
        },
        set: function (v) {
            this.i.c8 = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "othersCategoryThreshold", {
        /**
         * Gets or sets the threshold value that determines if slices are grouped into the Others slice.
        */
        get: function () {
            return this.i.cw;
        },
        set: function (v) {
            this.i.cw = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "othersCategoryType", {
        /**
         * Gets or sets whether to use numeric or percent-based threshold value.
        */
        get: function () {
            return this.i.au;
        },
        set: function (v) {
            this.i.au = ensureEnum(OthersCategoryType_$type, v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "othersCategoryText", {
        /**
         * Gets or sets the label of the Others slice.
        */
        get: function () {
            return this.i.eg;
        },
        set: function (v) {
            this.i.eg = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "explodedRadius", {
        /**
         * Determines how much the exploded slice is offset from the center. Value between 0 and 1.
        */
        get: function () {
            return this.i.co;
        },
        set: function (v) {
            this.i.co = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "radiusFactor", {
        /**
         * Gets or sets the scaling factor of the chart's radius. Value between 0 and 1.
        */
        get: function () {
            return this.i.cz;
        },
        set: function (v) {
            this.i.cz = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "allowSliceSelection", {
        /**
         * Gets or sets whether the slices can be selected.
        */
        get: function () {
            return this.i.by;
        },
        set: function (v) {
            this.i.by = ensureBool(v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "allowSliceExplosion", {
        /**
         * Gets or sets whether the slices can be exploded.
        */
        get: function () {
            return this.i.bx;
        },
        set: function (v) {
            this.i.bx = ensureBool(v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "explodedSlices", {
        /**
         * Gets or sets the collection of exploded slice indices.
        */
        get: function () {
            if (this._explodedSlices === null) {
                var coll = new IgxIndexCollection();
                var innerColl = this.i.explodedSlices;
                if (!innerColl) {
                    innerColl = new IndexCollection_internal();
                }
                this._explodedSlices = coll._fromInner(innerColl);
                this.i.explodedSlices = innerColl;
            }
            return this._explodedSlices;
        },
        set: function (v) {
            if (this._explodedSlices !== null) {
                this._explodedSlices._setSyncTarget(null);
                this._explodedSlices = null;
            }
            var coll = new IgxIndexCollection();
            this._explodedSlices = coll._fromOuter(v);
            var syncColl = new SyncableObservableCollection$1(Number_$type);
            var innerColl = this.i.explodedSlices;
            if (!innerColl) {
                innerColl = new IndexCollection_internal();
            }
            syncColl._inner = innerColl;
            syncColl.clear();
            this._explodedSlices._setSyncTarget(syncColl);
            this.i.explodedSlices = innerColl;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "labelExtent", {
        /**
         * Gets or sets the pixel amount by which the labels are offset from the edge of the slices.
        */
        get: function () {
            return this.i.cu;
        },
        set: function (v) {
            this.i.cu = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "startAngle", {
        /**
         * Gets or sets the starting angle of the chart.
         * The default zero value is equivalent to 3 o'clock.
        */
        get: function () {
            return this.i.c1;
        },
        set: function (v) {
            this.i.c1 = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "sweepDirection", {
        /**
         * Gets or sets the rotational direction of the chart.
        */
        get: function () {
            return this.i.hr;
        },
        set: function (v) {
            this.i.hr = ensureEnum(SweepDirection_$type, v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "othersCategoryFill", {
        /**
         * Gets or sets the fill brush.
        */
        get: function () {
            return this.i.hz ? this.i.hz.fill : null;
        },
        set: function (v) {
            this.ensureOthersCategoryStyle();
            this.i.hz.fill = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "othersCategoryStroke", {
        /**
         * Gets or sets the stroke brush.
        */
        get: function () {
            return this.i.hz ? this.i.hz.stroke : null;
        },
        set: function (v) {
            this.ensureOthersCategoryStyle();
            this.i.hz.stroke = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "othersCategoryStrokeThickness", {
        /**
         * Gets or sets the stroke thickness.
        */
        get: function () {
            return this.i.hz ? this.i.hz.strokeThickness : NaN;
        },
        set: function (v) {
            this.ensureOthersCategoryStyle();
            this.i.hz.strokeThickness = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "othersCategoryOpacity", {
        /**
         * Gets or sets the opacity.
        */
        get: function () {
            return this.i.hz ? this.i.hz.opacity : NaN;
        },
        set: function (v) {
            this.ensureOthersCategoryStyle();
            this.i.hz.opacity = +v;
        },
        enumerable: true,
        configurable: true
    });
    IgxPieChartBaseComponent.prototype.ensureOthersCategoryStyle = function () {
        if (this.i.hz) {
            return;
        }
        this.i.hz = new Style();
    };
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "selectedSliceFill", {
        /**
         * Gets or sets the fill brush.
        */
        get: function () {
            return this.i.h0 ? this.i.h0.fill : null;
        },
        set: function (v) {
            this.ensureSelectedStyle();
            this.i.h0.fill = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "selectedSliceStroke", {
        /**
         * Gets or sets the stroke brush.
        */
        get: function () {
            return this.i.h0 ? this.i.h0.stroke : null;
        },
        set: function (v) {
            this.ensureSelectedStyle();
            this.i.h0.stroke = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "selectedSliceStrokeThickness", {
        /**
         * Gets or sets the stroke thickness.
        */
        get: function () {
            return this.i.h0 ? this.i.h0.strokeThickness : NaN;
        },
        set: function (v) {
            this.ensureSelectedStyle();
            this.i.h0.strokeThickness = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "selectedSliceOpacity", {
        /**
         * Gets or sets the opacity.
        */
        get: function () {
            return this.i.h0 ? this.i.h0.opacity : NaN;
        },
        set: function (v) {
            this.ensureSelectedStyle();
            this.i.h0.opacity = +v;
        },
        enumerable: true,
        configurable: true
    });
    IgxPieChartBaseComponent.prototype.ensureSelectedStyle = function () {
        if (this.i.h0) {
            return;
        }
        this.i.h0 = new Style();
    };
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "brushes", {
        /**
         * Gets or sets the palette of brushes to use for coloring the slices.
        */
        get: function () {
            return fromBrushCollection(this.i.ak);
        },
        set: function (v) {
            this.i.ak = toBrushCollection(v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "outlines", {
        /**
         * Gets or sets the palette of brushes to use for outlines on the slices.
        */
        get: function () {
            return fromBrushCollection(this.i.al);
        },
        set: function (v) {
            this.i.al = toBrushCollection(v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "labelFormat", {
        /**
         * Gets or sets the label format string to use for the label.
        */
        get: function () {
            return this.i.dq;
        },
        set: function (v) {
            this.i.dq = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "labelFormatSpecifiers", {
        /**
         * Gets or sets the format specifiers to use with the LabelFormat string.
        */
        get: function () {
            return this.i.ag;
        },
        set: function (v) {
            this.i.ag = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "othersLabelFormat", {
        /**
         * Gets or sets the label format string to use for the label.
        */
        get: function () {
            return this.i.ek;
        },
        set: function (v) {
            this.i.ek = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "othersLabelFormatSpecifiers", {
        /**
         * Gets or sets the format specifiers to use with the OthersLabelFormat string.
        */
        get: function () {
            return this.i.aj;
        },
        set: function (v) {
            this.i.aj = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "legendLabelFormat", {
        /**
         * Gets or sets the label format string to use for the label.
        */
        get: function () {
            return this.i.d6;
        },
        set: function (v) {
            this.i.d6 = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "legendLabelFormatSpecifiers", {
        /**
         * Gets or sets the format specifiers to use with the LegendLabelFormat string.
        */
        get: function () {
            return this.i.ah;
        },
        set: function (v) {
            this.i.ah = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "legendOthersLabelFormat", {
        /**
         * Gets or sets the label format string to use for the label.
        */
        get: function () {
            return this.i.eb;
        },
        set: function (v) {
            this.i.eb = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "legendOthersLabelFormatSpecifiers", {
        /**
         * Gets or sets the format specifiers to use with the LegendOthersLabelFormat string.
        */
        get: function () {
            return this.i.ai;
        },
        set: function (v) {
            this.i.ai = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "legendItemTemplate", {
        /**
         * Gets or sets the LegendItemTemplate property.
         * The legend item control content is created according to the LegendItemTemplate on-demand by
         * the chart object itself.
        */
        get: function () {
            return this.i.legendItemTemplate;
        },
        set: function (v) {
            this.i.legendItemTemplate = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "legendItemBadgeTemplate", {
        /**
         * Gets or sets the LegendItemBadgeTemplate property.
         * The legend item badge is created according to the LegendItemBadgeTemplate on-demand by
         * the chart object itself.
        */
        get: function () {
            return this.i.legendItemBadgeTemplate;
        },
        set: function (v) {
            this.i.legendItemBadgeTemplate = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "isSurfaceInteractionDisabled", {
        /**
         * Gets or sets whether all surface interactions with the plot area should be disabled.
        */
        get: function () {
            return this.i.b3;
        },
        set: function (v) {
            this.i.b3 = ensureBool(v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "formatLabel", {
        /**
         * Sets or gets a function which takes an object that produces a formatted label for displaying in the chart.
        */
        get: function () {
            return this.i.ax;
        },
        set: function (v) {
            this.i.ax = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "formatLegendLabel", {
        /**
         * Sets or gets a function which takes an object that produces a formatted label for displaying in the chart's legend.
        */
        get: function () {
            return this.i.ay;
        },
        set: function (v) {
            this.i.ay = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "pixelScalingRatio", {
        /**
         * Gets or sets the scaling value used to affect the pixel density of the control.
         * A higher scaling ratio will produce crisper visuals at the expense of memory.  Lower values will cause the control
         * to appear blurry.
        */
        get: function () {
            return this.i.cy;
        },
        set: function (v) {
            this.i.cy = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "actualPixelScalingRatio", {
        /**
         * Resolved pixel scaling ratio. Unless explicitly overridden by the PieChartBase.PixelScalingRatio property,
         * this one returns the default ratio enforced by device. High resolution devices will initialize this property
         * to a higher value.
        */
        get: function () {
            return this.i.cl;
        },
        set: function (v) {
            this.i.cl = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "selectionMode", {
        /**
         * Gets or sets which mode to use for selecting slices.
        */
        get: function () {
            return this.i.a7;
        },
        set: function (v) {
            this.i.a7 = ensureEnum(SliceSelectionMode_$type, v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "selectedItem", {
        /**
         * Gets or sets the currently selected data item.  If a different data item is provided the pie chart
         * will select the slice associated with the new item.
        */
        get: function () {
            return this.i.selectedItem;
        },
        set: function (v) {
            this.i.selectedItem = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "selectedItems", {
        /**
         * Gets the currently selected data items.  Adding or removing data items from this collection will
         * select or deselect the slices associated with those items.
        */
        get: function () {
            if (this._selectedItems === null) {
                var coll = new IgxObjectCollection();
                var innerColl = this.i.selectedItems;
                if (!innerColl) {
                    innerColl = new ObjectCollection_internal(0);
                }
                this._selectedItems = coll._fromInner(innerColl);
                this.i.selectedItems = innerColl;
            }
            return this._selectedItems;
        },
        set: function (v) {
            if (this._selectedItems !== null) {
                this._selectedItems._setSyncTarget(null);
                this._selectedItems = null;
            }
            var coll = new IgxObjectCollection();
            this._selectedItems = coll._fromOuter(v);
            var syncColl = new SyncableObservableCollection$1(Base.$type);
            var innerColl = this.i.selectedItems;
            if (!innerColl) {
                innerColl = new ObjectCollection_internal(0);
            }
            syncColl._inner = innerColl;
            syncColl.clear();
            this._selectedItems._setSyncTarget(syncColl);
            this.i.selectedItems = innerColl;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "textStyle", {
        /**
         * Manually overrides the style to use for the labels.
        */
        get: function () {
            return this.i.ev;
        },
        set: function (v) {
            this.i.ev = v;
        },
        enumerable: true,
        configurable: true
    });
    IgxPieChartBaseComponent.prototype.findByName = function (name) {
        if (this.legend && this.legend.name && this.legend.name == name) {
            return this.legend;
        }
        if (this.labelFormatSpecifiers != null && arrayFindByName(this.labelFormatSpecifiers, name)) {
            return arrayFindByName(this.labelFormatSpecifiers, name);
        }
        if (this.othersLabelFormatSpecifiers != null && arrayFindByName(this.othersLabelFormatSpecifiers, name)) {
            return arrayFindByName(this.othersLabelFormatSpecifiers, name);
        }
        if (this.legendLabelFormatSpecifiers != null && arrayFindByName(this.legendLabelFormatSpecifiers, name)) {
            return arrayFindByName(this.legendLabelFormatSpecifiers, name);
        }
        if (this.legendOthersLabelFormatSpecifiers != null && arrayFindByName(this.legendOthersLabelFormatSpecifiers, name)) {
            return arrayFindByName(this.legendOthersLabelFormatSpecifiers, name);
        }
        return null;
    };
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "hasUserValues", {
        get: function () {
            return this._hasUserValues;
        },
        enumerable: true,
        configurable: true
    });
    IgxPieChartBaseComponent.prototype.__m = function (propertyName) {
        if (!this._inStyling) {
            this._hasUserValues.add(propertyName);
        }
    };
    IgxPieChartBaseComponent.prototype._styling = function (container, component, parent) {
        if (this._inStyling) {
            return;
        }
        this._inStyling = true;
        this._stylingContainer = container;
        this._stylingParent = component;
        var genericPrefix = "";
        var typeName = this.i.$type.name;
        if (typeName.indexOf("Xam") === 0) {
            typeName = typeName.substring(3);
        }
        genericPrefix = toSpinal("PieChartBaseComponent");
        var additionalPrefixes = [];
        var prefix = toSpinal(typeName);
        additionalPrefixes.push(prefix + "-");
        var b = this.i.$type.baseType;
        while (b && b.name != "Object" &&
            b.name != "Base" &&
            b.name != "Control" &&
            b.Name != "DependencyObject" &&
            b.Name != "FrameworkElement") {
            typeName = b.name;
            if (typeName.indexOf("Xam") === 0) {
                typeName = typeName.substring(3);
            }
            var basePrefix = toSpinal(typeName);
            additionalPrefixes.push(basePrefix + "-");
            b = b.baseType;
        }
        if (parent) {
            var parentTypeName = parent.i.$type.name;
            if (parentTypeName.indexOf("Xam") === 0) {
                parentTypeName = parentTypeName.substring(3);
            }
            var parentPrefix = toSpinal(parentTypeName);
            additionalPrefixes.push(parentPrefix + "-" + genericPrefix + "-");
            additionalPrefixes.push(parentPrefix + "-" + prefix + "-");
        }
        initializePropertiesFromCss(container, this, genericPrefix + "-", this.hasUserValues, false, additionalPrefixes);
        if (this._otherStyling) {
            this._otherStyling(container, component, parent);
        }
        this._inStyling = false;
    };
    /**
     * Returns the chart visuals expressed as a serialized string.
    
    */
    IgxPieChartBaseComponent.prototype.exportSerializedVisualData = function () {
        var iv = this.i.dg();
        return (iv);
    };
    /**
     * Simulates a mouse click action at the specified point.
    
    * @param point  * The point to click at.
    */
    IgxPieChartBaseComponent.prototype.simulateLeftClick = function (point) {
        this.i.f7(toPoint(point));
    };
    /**
     * Invoked by the containing object to provide reference to the container.
    
    * @param container  * Reference to the container.
    */
    IgxPieChartBaseComponent.prototype.provideContainer = function (container) {
        this.i.provideContainer(container);
    };
    /**
     * Notification from the containing object that the container has been resized.
    
    */
    IgxPieChartBaseComponent.prototype.notifyContainerResized = function () {
        this.i.notifyContainerResized();
    };
    /**
     * Invoked when the instance is being destroyed.
    
    */
    IgxPieChartBaseComponent.prototype.destroy = function () {
        this.i.destroy();
    };
    /**
     * Use to force the pie chart to finish any deferred work before printing or evaluating its visual.
     * This should only be called if the visual of the pie chart needs to be synchronously saved or evaluated.
     * Calling this method too often will hinder the performance of the pie chart.
    
    */
    IgxPieChartBaseComponent.prototype.flush = function () {
        this.i.e5();
    };
    IgxPieChartBaseComponent.prototype.notifySetItem = function (source_, index, oldItem, newItem) {
        this.i.fe(source_, index, oldItem, newItem);
    };
    /**
     * Manually notifies the pie chart's data source that the data it has bound to has been cleared and needs to be re-examined.
     * This should not be called if the data that the pie chart is bound to is already observable.
    
    * @param source_  * The data source.
    */
    IgxPieChartBaseComponent.prototype.notifyClearItems = function (source_) {
        this.i.fa(source_);
    };
    IgxPieChartBaseComponent.prototype.notifyInsertItem = function (source_, index, newItem) {
        this.i.fc(source_, index, newItem);
    };
    IgxPieChartBaseComponent.prototype.notifyRemoveItem = function (source_, index, oldItem) {
        this.i.fd(source_, index, oldItem);
    };
    /**
     * Assigns data source for the items.
    
    * @param source_  * Array of the data source items.
    */
    IgxPieChartBaseComponent.prototype.setWidgetLevelDataSource = function (source_) {
        this.i.f6(source_);
    };
    /**
     * Resets items data source.
    
    */
    IgxPieChartBaseComponent.prototype.removeWidgetLevelDataSource = function () {
        this.i.fx();
    };
    /**
     * Invoked when style is updated.
    
    */
    IgxPieChartBaseComponent.prototype.styleUpdated = function () {
        this.i.gb();
    };
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "labelClick", {
        /**
         * Raised when the slice's label is clicked.
        */
        get: function () {
            var _this = this;
            if (this._labelClick == null) {
                this._labelClick = new EventEmitter();
                this.i.labelClick = delegateCombine(this.i.labelClick, function (o, e) {
                    _this._runInZone(function () {
                        var outerArgs = new IgxLabelClickEventArgs();
                        outerArgs._provideImplementation(e);
                        if (_this.beforeLabelClick) {
                            _this.beforeLabelClick(_this, outerArgs);
                        }
                        _this._labelClick.emit({
                            sender: _this,
                            args: outerArgs
                        });
                    });
                });
            }
            return this._labelClick;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "selectedItemChanging", {
        /**
         * Raised when the SelectedItem property is changing.  This event is cancelable.
        */
        get: function () {
            var _this = this;
            if (this._selectedItemChanging == null) {
                this._selectedItemChanging = new EventEmitter();
                this.i.selectedItemChanging = delegateCombine(this.i.selectedItemChanging, function (o, e) {
                    _this._runInZone(function () {
                        var outerArgs = new IgxSelectedItemChangingEventArgs();
                        outerArgs._provideImplementation(e);
                        if (_this.beforeSelectedItemChanging) {
                            _this.beforeSelectedItemChanging(_this, outerArgs);
                        }
                        _this._selectedItemChanging.emit({
                            sender: _this,
                            args: outerArgs
                        });
                    });
                });
            }
            return this._selectedItemChanging;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "selectedItemsChanging", {
        /**
         * Raised when the SelectedItems collection is changing.  This event is cancelable.
        */
        get: function () {
            var _this = this;
            if (this._selectedItemsChanging == null) {
                this._selectedItemsChanging = new EventEmitter();
                this.i.selectedItemsChanging = delegateCombine(this.i.selectedItemsChanging, function (o, e) {
                    _this._runInZone(function () {
                        var outerArgs = new IgxSelectedItemsChangingEventArgs();
                        outerArgs._provideImplementation(e);
                        if (_this.beforeSelectedItemsChanging) {
                            _this.beforeSelectedItemsChanging(_this, outerArgs);
                        }
                        _this._selectedItemsChanging.emit({
                            sender: _this,
                            args: outerArgs
                        });
                    });
                });
            }
            return this._selectedItemsChanging;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "selectedItemChanged", {
        /**
         * Raised when the SelectedItem property has changed.
        */
        get: function () {
            var _this = this;
            if (this._selectedItemChanged == null) {
                this._selectedItemChanged = new EventEmitter();
                this.i.selectedItemChanged = delegateCombine(this.i.selectedItemChanged, function (o, e) {
                    _this._runInZone(function () {
                        var outerArgs = new IgxSelectedItemChangedEventArgs();
                        outerArgs._provideImplementation(e);
                        if (_this.beforeSelectedItemChanged) {
                            _this.beforeSelectedItemChanged(_this, outerArgs);
                        }
                        _this._selectedItemChanged.emit({
                            sender: _this,
                            args: outerArgs
                        });
                    });
                });
            }
            return this._selectedItemChanged;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "selectedItemsChanged", {
        /**
         * Raised when the SelectedItems collection has changed.
        */
        get: function () {
            var _this = this;
            if (this._selectedItemsChanged == null) {
                this._selectedItemsChanged = new EventEmitter();
                this.i.selectedItemsChanged = delegateCombine(this.i.selectedItemsChanged, function (o, e) {
                    _this._runInZone(function () {
                        var outerArgs = new IgxSelectedItemsChangedEventArgs();
                        outerArgs._provideImplementation(e);
                        if (_this.beforeSelectedItemsChanged) {
                            _this.beforeSelectedItemsChanged(_this, outerArgs);
                        }
                        _this._selectedItemsChanged.emit({
                            sender: _this,
                            args: outerArgs
                        });
                    });
                });
            }
            return this._selectedItemsChanged;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPieChartBaseComponent.prototype, "sliceClick", {
        /**
         * Raised when the slice is clicked.
        */
        get: function () {
            var _this = this;
            if (this._sliceClick == null) {
                this._sliceClick = new EventEmitter();
                this.i.sliceClick = delegateCombine(this.i.sliceClick, function (o, e) {
                    _this._runInZone(function () {
                        var outerArgs = new IgxSliceClickEventArgs();
                        outerArgs._provideImplementation(e);
                        if (_this.beforeSliceClick) {
                            _this.beforeSliceClick(_this, outerArgs);
                        }
                        _this._sliceClick.emit({
                            sender: _this,
                            args: outerArgs
                        });
                    });
                });
            }
            return this._sliceClick;
        },
        enumerable: true,
        configurable: true
    });
    IgxPieChartBaseComponent.prototype._runInZone = function (act) {
        if (this._zoneRunner != null) {
            this._zoneRunner(act);
        }
        else {
            act();
        }
    };
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPieChartBaseComponent.prototype, "innerExtent", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxPieChartBaseComponent.prototype, "valueMemberPath", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxPieChartBaseComponent.prototype, "labelMemberPath", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxPieChartBaseComponent.prototype, "legendLabelMemberPath", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPieChartBaseComponent.prototype, "labelsPosition", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxPieChartBaseComponent.prototype, "labelOuterColor", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxPieChartBaseComponent.prototype, "labelInnerColor", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPieChartBaseComponent.prototype, "leaderLineVisibility", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPieChartBaseComponent.prototype, "leaderLineType", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPieChartBaseComponent.prototype, "leaderLineMargin", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Object),
        tslib_1.__metadata("design:paramtypes", [Object])
    ], IgxPieChartBaseComponent.prototype, "toolTip", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPieChartBaseComponent.prototype, "othersCategoryThreshold", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPieChartBaseComponent.prototype, "othersCategoryType", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxPieChartBaseComponent.prototype, "othersCategoryText", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPieChartBaseComponent.prototype, "explodedRadius", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPieChartBaseComponent.prototype, "radiusFactor", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [Boolean])
    ], IgxPieChartBaseComponent.prototype, "allowSliceSelection", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [Boolean])
    ], IgxPieChartBaseComponent.prototype, "allowSliceExplosion", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", IgxIndexCollection),
        tslib_1.__metadata("design:paramtypes", [IgxIndexCollection])
    ], IgxPieChartBaseComponent.prototype, "explodedSlices", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPieChartBaseComponent.prototype, "labelExtent", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPieChartBaseComponent.prototype, "startAngle", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPieChartBaseComponent.prototype, "sweepDirection", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxPieChartBaseComponent.prototype, "othersCategoryFill", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxPieChartBaseComponent.prototype, "othersCategoryStroke", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPieChartBaseComponent.prototype, "othersCategoryStrokeThickness", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPieChartBaseComponent.prototype, "othersCategoryOpacity", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxPieChartBaseComponent.prototype, "selectedSliceFill", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxPieChartBaseComponent.prototype, "selectedSliceStroke", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPieChartBaseComponent.prototype, "selectedSliceStrokeThickness", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPieChartBaseComponent.prototype, "selectedSliceOpacity", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Array),
        tslib_1.__metadata("design:paramtypes", [Array])
    ], IgxPieChartBaseComponent.prototype, "brushes", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Array),
        tslib_1.__metadata("design:paramtypes", [Array])
    ], IgxPieChartBaseComponent.prototype, "outlines", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxPieChartBaseComponent.prototype, "labelFormat", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Array),
        tslib_1.__metadata("design:paramtypes", [Array])
    ], IgxPieChartBaseComponent.prototype, "labelFormatSpecifiers", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxPieChartBaseComponent.prototype, "othersLabelFormat", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Array),
        tslib_1.__metadata("design:paramtypes", [Array])
    ], IgxPieChartBaseComponent.prototype, "othersLabelFormatSpecifiers", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxPieChartBaseComponent.prototype, "legendLabelFormat", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Array),
        tslib_1.__metadata("design:paramtypes", [Array])
    ], IgxPieChartBaseComponent.prototype, "legendLabelFormatSpecifiers", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxPieChartBaseComponent.prototype, "legendOthersLabelFormat", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Array),
        tslib_1.__metadata("design:paramtypes", [Array])
    ], IgxPieChartBaseComponent.prototype, "legendOthersLabelFormatSpecifiers", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Object),
        tslib_1.__metadata("design:paramtypes", [Object])
    ], IgxPieChartBaseComponent.prototype, "legendItemTemplate", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Object),
        tslib_1.__metadata("design:paramtypes", [Object])
    ], IgxPieChartBaseComponent.prototype, "legendItemBadgeTemplate", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [Boolean])
    ], IgxPieChartBaseComponent.prototype, "isSurfaceInteractionDisabled", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Function),
        tslib_1.__metadata("design:paramtypes", [Function])
    ], IgxPieChartBaseComponent.prototype, "formatLabel", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Function),
        tslib_1.__metadata("design:paramtypes", [Function])
    ], IgxPieChartBaseComponent.prototype, "formatLegendLabel", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPieChartBaseComponent.prototype, "pixelScalingRatio", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPieChartBaseComponent.prototype, "actualPixelScalingRatio", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPieChartBaseComponent.prototype, "selectionMode", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Object),
        tslib_1.__metadata("design:paramtypes", [Object])
    ], IgxPieChartBaseComponent.prototype, "selectedItem", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", IgxObjectCollection),
        tslib_1.__metadata("design:paramtypes", [IgxObjectCollection])
    ], IgxPieChartBaseComponent.prototype, "selectedItems", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxPieChartBaseComponent.prototype, "textStyle", null);
    tslib_1.__decorate([
        Output(),
        tslib_1.__metadata("design:type", EventEmitter),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxPieChartBaseComponent.prototype, "labelClick", null);
    tslib_1.__decorate([
        Output(),
        tslib_1.__metadata("design:type", EventEmitter),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxPieChartBaseComponent.prototype, "selectedItemChanging", null);
    tslib_1.__decorate([
        Output(),
        tslib_1.__metadata("design:type", EventEmitter),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxPieChartBaseComponent.prototype, "selectedItemsChanging", null);
    tslib_1.__decorate([
        Output(),
        tslib_1.__metadata("design:type", EventEmitter),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxPieChartBaseComponent.prototype, "selectedItemChanged", null);
    tslib_1.__decorate([
        Output(),
        tslib_1.__metadata("design:type", EventEmitter),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxPieChartBaseComponent.prototype, "selectedItemsChanged", null);
    tslib_1.__decorate([
        Output(),
        tslib_1.__metadata("design:type", EventEmitter),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxPieChartBaseComponent.prototype, "sliceClick", null);
    return IgxPieChartBaseComponent;
}());
export { IgxPieChartBaseComponent };
