// igniteui-angular-charts/esm5/lib/igx-stacked-area-series-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxHorizontalStackedSeriesBaseComponent } from "./igx-horizontal-stacked-series-base-component";
import { IgxStackedSeriesBaseComponent } from "./igx-stacked-series-base-component";
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { StackedAreaSeries } from "./StackedAreaSeries";
/**
 * Represents a IgxDataChartComponent stacked area series.
*/
export var IgxStackedAreaSeriesComponent_PROVIDERS = [{ provide: IgxHorizontalStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStackedAreaSeriesComponent; }) }, { provide: IgxStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStackedAreaSeriesComponent; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxStackedAreaSeriesComponent; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxStackedAreaSeriesComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxStackedAreaSeriesComponent; }) }];
var IgxStackedAreaSeriesComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxStackedAreaSeriesComponent, _super);
    function IgxStackedAreaSeriesComponent() {
        return _super.call(this) || this;
    }
    IgxStackedAreaSeriesComponent_1 = IgxStackedAreaSeriesComponent;
    IgxStackedAreaSeriesComponent.prototype.createImplementation = function () {
        return new StackedAreaSeries();
    };
    Object.defineProperty(IgxStackedAreaSeriesComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    var IgxStackedAreaSeriesComponent_1;
    IgxStackedAreaSeriesComponent = IgxStackedAreaSeriesComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-stacked-area-series',
            template: "",
            providers: [{ provide: IgxHorizontalStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStackedAreaSeriesComponent_1; }) }, { provide: IgxStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStackedAreaSeriesComponent_1; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxStackedAreaSeriesComponent_1; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxStackedAreaSeriesComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxStackedAreaSeriesComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxStackedAreaSeriesComponent);
    return IgxStackedAreaSeriesComponent;
}(IgxHorizontalStackedSeriesBaseComponent));
export { IgxStackedAreaSeriesComponent };
