// igniteui-angular-charts/esm5/lib/igx-median-price-indicator-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxItemwiseStrategyBasedIndicatorComponent } from "./igx-itemwise-strategy-based-indicator-component";
import { IgxStrategyBasedIndicatorComponent } from "./igx-strategy-based-indicator-component";
import { IgxFinancialIndicatorComponent } from "./igx-financial-indicator-component";
import { IgxFinancialSeriesComponent } from "./igx-financial-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { MedianPriceIndicator } from "./MedianPriceIndicator";
/**
 * Represents a IgxDataChartComponent Median Price indicator series.
 * Default required members: High, Low
*
* Represents a Ignite UIDataChart Median Price indicator series.
*
* Example:
*
* ```html
* <igx-data-chart
*     [dataSource]="data">
* 	<igx-category-x-axis
* 		label="label"
* 		#xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
* 		#yAxis>
*     </igx-numeric-y-axis>
* 	<igx-median-price-indicator
*         #series
*         [xAxis]="xAxis"
*         [yAxis]="yAxis"
*         highMemberPath="high"
* 		lowMemberPath="low"
* 		volumeMemberPath="volume">
*     </igx-median-price-indicator>
* </igx-data-chart>
* ```
*
* ```ts
* let series = new IgxMedianPriceIndicatorComponent();
* series.xAxis = this.xAxis;
* series.yAxis = this.yAxis;
* series.highMemberPath = "high";
* series.lowMemberPath = "low";
* series.volumeMemberPath= "volume";
* this.chart.series.add(series);
* ```
*/
export var IgxMedianPriceIndicatorComponent_PROVIDERS = [{ provide: IgxItemwiseStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxMedianPriceIndicatorComponent; }) }, { provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxMedianPriceIndicatorComponent; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxMedianPriceIndicatorComponent; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxMedianPriceIndicatorComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxMedianPriceIndicatorComponent; }) }];
var IgxMedianPriceIndicatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxMedianPriceIndicatorComponent, _super);
    function IgxMedianPriceIndicatorComponent() {
        return _super.call(this) || this;
    }
    IgxMedianPriceIndicatorComponent_1 = IgxMedianPriceIndicatorComponent;
    IgxMedianPriceIndicatorComponent.prototype.createImplementation = function () {
        return new MedianPriceIndicator();
    };
    Object.defineProperty(IgxMedianPriceIndicatorComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    var IgxMedianPriceIndicatorComponent_1;
    IgxMedianPriceIndicatorComponent = IgxMedianPriceIndicatorComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-median-price-indicator',
            template: "",
            providers: [{ provide: IgxItemwiseStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxMedianPriceIndicatorComponent_1; }) }, { provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxMedianPriceIndicatorComponent_1; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxMedianPriceIndicatorComponent_1; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxMedianPriceIndicatorComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxMedianPriceIndicatorComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxMedianPriceIndicatorComponent);
    return IgxMedianPriceIndicatorComponent;
}(IgxItemwiseStrategyBasedIndicatorComponent));
export { IgxMedianPriceIndicatorComponent };
