// igniteui-angular-charts/esm5/lib/igx-polar-base-component.js
import * as tslib_1 from "tslib";
import { forwardRef, Input } from '@angular/core';
import { IgxNumericAngleAxisComponent } from "./igx-numeric-angle-axis-component";
import { IgxNumericRadiusAxisComponent } from "./igx-numeric-radius-axis-component";
import { TrendLineType, TrendLineType_$type } from "igniteui-angular-core";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { ensureBool, ensureEnum, brushToString, stringToBrush, toPoint } from "igniteui-angular-core";
/**
 * Represents the base class from which all IgxDataChartComponent polar series are derived.
*/
export var IgxPolarBaseComponent_PROVIDERS = [{ provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxPolarBaseComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxPolarBaseComponent; }) }];
var IgxPolarBaseComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxPolarBaseComponent, _super);
    function IgxPolarBaseComponent() {
        return _super.call(this) || this;
    }
    Object.defineProperty(IgxPolarBaseComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPolarBaseComponent.prototype, "isPolar", {
        /**
         * Checks if this series is a polar series
        */
        get: function () {
            return this.i.c0;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPolarBaseComponent.prototype, "angleMemberPath", {
        /**
         * Gets or sets the path to use to find the angle values for the series.
        */
        get: function () {
            return this.i.n0;
        },
        set: function (v) {
            this.i.n0 = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPolarBaseComponent.prototype, "radiusMemberPath", {
        /**
         * Gets or sets the path to use to get the radius values for the series.
        */
        get: function () {
            return this.i.n6;
        },
        set: function (v) {
            this.i.n6 = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPolarBaseComponent.prototype, "angleAxis", {
        /**
         * Gets the effective angle axis for the current series object.
        */
        get: function () {
            if (this.i.m7 == null) {
                return null;
            }
            if (!this.i.m7.externalObject) {
                var e = IgxNumericAngleAxisComponent._createFromInternal(this.i.m7);
                if (e) {
                    e._implementation = this.i.m7;
                }
                this.i.m7.externalObject = e;
            }
            return this.i.m7.externalObject;
        },
        set: function (v) {
            if (v != null && this._stylingContainer && v._styling)
                v._styling(this._stylingContainer, this, this);
            v == null ? this.i.m7 = null : this.i.m7 = v.i;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPolarBaseComponent.prototype, "radiusAxis", {
        /**
         * Gets the effective radius axis for the current series object.
        */
        get: function () {
            if (this.i.m8 == null) {
                return null;
            }
            if (!this.i.m8.externalObject) {
                var e = IgxNumericRadiusAxisComponent._createFromInternal(this.i.m8);
                if (e) {
                    e._implementation = this.i.m8;
                }
                this.i.m8.externalObject = e;
            }
            return this.i.m8.externalObject;
        },
        set: function (v) {
            if (v != null && this._stylingContainer && v._styling)
                v._styling(this._stylingContainer, this, this);
            v == null ? this.i.m8 = null : this.i.m8 = v.i;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPolarBaseComponent.prototype, "useCartesianInterpolation", {
        /**
         * Gets or sets whether Cartesian Interpolation should be used rather than Archimedian
         * spiral based interpolation.
        */
        get: function () {
            return this.i.ns;
        },
        set: function (v) {
            this.i.ns = ensureBool(v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPolarBaseComponent.prototype, "maximumMarkers", {
        /**
         * Gets or sets the maximum number of markers displayed by the current series.
         * If more than the specified number of markers are visible, the polar series will automatically
         * choose a representative set.
        */
        get: function () {
            return this.i.nv;
        },
        set: function (v) {
            this.i.nv = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPolarBaseComponent.prototype, "trendLineType", {
        /**
         * Gets or sets the trend type for the current series object.
        */
        get: function () {
            return this.i.nj;
        },
        set: function (v) {
            this.i.nj = ensureEnum(TrendLineType_$type, v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPolarBaseComponent.prototype, "trendLineBrush", {
        /**
         * Gets or sets the brush that specifies how the current series
         * object's trend line is drawn.
        */
        get: function () {
            return brushToString(this.i.ow);
        },
        set: function (v) {
            this.i.ow = stringToBrush(v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPolarBaseComponent.prototype, "actualTrendLineBrush", {
        /**
         * Gets the effective TrendLineBrush for this series.
        */
        get: function () {
            return brushToString(this.i.ov);
        },
        set: function (v) {
            this.i.ov = stringToBrush(v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPolarBaseComponent.prototype, "trendLineThickness", {
        /**
         * Gets or sets the thickness of the current series object's trend line.
        */
        get: function () {
            return this.i.nu;
        },
        set: function (v) {
            this.i.nu = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPolarBaseComponent.prototype, "trendLinePeriod", {
        /**
         * Gets or sets the moving average period for the current series object.
        */
        get: function () {
            return this.i.nw;
        },
        set: function (v) {
            this.i.nw = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPolarBaseComponent.prototype, "trendLineZIndex", {
        /**
         * Sets or Gets the Trendline Z index.
        */
        get: function () {
            return this.i.nx;
        },
        set: function (v) {
            this.i.nx = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPolarBaseComponent.prototype, "clipSeriesToBounds", {
        /**
         * Gets or sets whether to clip the series to the bounds.
         * Setting this to true can effect performance.
        */
        get: function () {
            return this.i.nq;
        },
        set: function (v) {
            this.i.nq = ensureBool(v);
        },
        enumerable: true,
        configurable: true
    });
    IgxPolarBaseComponent.prototype.findByName = function (name) {
        var baseResult = _super.prototype.findByName.call(this, name);
        if (baseResult) {
            return baseResult;
        }
        if (this.angleAxis && this.angleAxis.name && this.angleAxis.name == name) {
            return this.angleAxis;
        }
        if (this.radiusAxis && this.radiusAxis.name && this.radiusAxis.name == name) {
            return this.radiusAxis;
        }
        return null;
    };
    IgxPolarBaseComponent.prototype._styling = function (container, component, parent) {
        _super.prototype._styling.call(this, container, component, parent);
        this._inStyling = true;
        if (this.angleAxis && this.angleAxis._styling) {
            this.angleAxis._styling(container, component, this);
        }
        if (this.radiusAxis && this.radiusAxis._styling) {
            this.radiusAxis._styling(container, component, this);
        }
        this._inStyling = false;
    };
    /**
     * Scrolls the requested item into view, if possible.
    
    * @param item  * The item to scroll into view.
    */
    IgxPolarBaseComponent.prototype.scrollIntoView = function (item) {
        var iv = this.i.c9(item);
        return (iv);
    };
    /**
     * Gets the item that is the best match for the specified world coordinates.
    
    * @param world  * The world coordinates to use.
    */
    IgxPolarBaseComponent.prototype.getItem = function (world) {
        var iv = this.i.fp(toPoint(world));
        return (iv);
    };
    /**
     * Gets the index of the item that resides at the provided world coordinates.
    
    * @param world  * The world coordinates of the requested item.
    */
    IgxPolarBaseComponent.prototype.getItemIndex = function (world) {
        var iv = this.i.e8(toPoint(world));
        return (iv);
    };
    IgxPolarBaseComponent.prototype.getItemValue = function (item, memberPathName) {
        var iv = this.i.fs(item, memberPathName);
        return (iv);
    };
    /**
     * Determine if object can be used as RadiusAxis
    
    * @param axis  * The object to check
    */
    IgxPolarBaseComponent.prototype.canUseAsRadiusAxis = function (axis) {
        var iv = this.i.np(axis);
        return (iv);
    };
    /**
     * Determine if object can be used as AngleAxis
    
    * @param axis  * The object to check
    */
    IgxPolarBaseComponent.prototype.canUseAsAngleAxis = function (axis) {
        var iv = this.i.no(axis);
        return (iv);
    };
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxPolarBaseComponent.prototype, "angleMemberPath", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxPolarBaseComponent.prototype, "radiusMemberPath", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", IgxNumericAngleAxisComponent),
        tslib_1.__metadata("design:paramtypes", [IgxNumericAngleAxisComponent])
    ], IgxPolarBaseComponent.prototype, "angleAxis", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", IgxNumericRadiusAxisComponent),
        tslib_1.__metadata("design:paramtypes", [IgxNumericRadiusAxisComponent])
    ], IgxPolarBaseComponent.prototype, "radiusAxis", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [Boolean])
    ], IgxPolarBaseComponent.prototype, "useCartesianInterpolation", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPolarBaseComponent.prototype, "maximumMarkers", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPolarBaseComponent.prototype, "trendLineType", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxPolarBaseComponent.prototype, "trendLineBrush", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxPolarBaseComponent.prototype, "actualTrendLineBrush", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPolarBaseComponent.prototype, "trendLineThickness", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPolarBaseComponent.prototype, "trendLinePeriod", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPolarBaseComponent.prototype, "trendLineZIndex", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [Boolean])
    ], IgxPolarBaseComponent.prototype, "clipSeriesToBounds", null);
    return IgxPolarBaseComponent;
}(IgxMarkerSeriesComponent));
export { IgxPolarBaseComponent };
