// igniteui-angular-charts/esm5/lib/igx-mass-index-indicator-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxStrategyBasedIndicatorComponent } from "./igx-strategy-based-indicator-component";
import { IgxFinancialIndicatorComponent } from "./igx-financial-indicator-component";
import { IgxFinancialSeriesComponent } from "./igx-financial-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { MassIndexIndicator } from "./MassIndexIndicator";
/**
 * Represents a IgxDataChartComponent Mass Index indicator series.
 * Default required members: High, Low
*
* Represents a Ignite UIDataChart Mass Index indicator series.
*
* Example:
*
* ```html
* <igx-data-chart
*     [dataSource]="data">
* 	<igx-category-x-axis
* 		label="label"
* 		#xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
* 		#yAxis>
*     </igx-numeric-y-axis>
* 	<igx-mass-index-indicator
*         #series
*         [xAxis]="xAxis"
*         [yAxis]="yAxis"
*         highMemberPath="high"
* 		lowMemberPath="low"
* 		volumeMemberPath="volume">
*     </igx-mass-index-indicator>
* </igx-data-chart>
* ```
*
* ```ts
* let series = new IgxMassIndexIndicatorComponent();
* series.xAxis = this.xAxis;
* series.yAxis = this.yAxis;
* series.highMemberPath = "high";
* series.lowMemberPath = "low";
* series.volumeMemberPath= "volume";
* this.chart.series.add(series);
* ```
*/
export var IgxMassIndexIndicatorComponent_PROVIDERS = [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxMassIndexIndicatorComponent; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxMassIndexIndicatorComponent; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxMassIndexIndicatorComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxMassIndexIndicatorComponent; }) }];
var IgxMassIndexIndicatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxMassIndexIndicatorComponent, _super);
    function IgxMassIndexIndicatorComponent() {
        return _super.call(this) || this;
    }
    IgxMassIndexIndicatorComponent_1 = IgxMassIndexIndicatorComponent;
    IgxMassIndexIndicatorComponent.prototype.createImplementation = function () {
        return new MassIndexIndicator();
    };
    Object.defineProperty(IgxMassIndexIndicatorComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    var IgxMassIndexIndicatorComponent_1;
    IgxMassIndexIndicatorComponent = IgxMassIndexIndicatorComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-mass-index-indicator',
            template: "",
            providers: [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxMassIndexIndicatorComponent_1; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxMassIndexIndicatorComponent_1; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxMassIndexIndicatorComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxMassIndexIndicatorComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxMassIndexIndicatorComponent);
    return IgxMassIndexIndicatorComponent;
}(IgxStrategyBasedIndicatorComponent));
export { IgxMassIndexIndicatorComponent };
