// igniteui-angular-charts/esm5/lib/igx-trix-indicator-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxStrategyBasedIndicatorComponent } from "./igx-strategy-based-indicator-component";
import { IgxFinancialIndicatorComponent } from "./igx-financial-indicator-component";
import { IgxFinancialSeriesComponent } from "./igx-financial-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { TRIXIndicator } from "./TRIXIndicator";
/**
 * Represents a IgxDataChartComponent TRIX indicator series.
 * Default required members: Close
*
* The `IgxTRIXIndicatorComponent` class specify the series as Trix Indicator series.
*
* ```html
* <igx-data-chart
*     [dataSource]="data">
* 	<igx-category-x-axis
* 		label="label"
* 		#xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
* 		#yAxis>
*     </igx-numeric-y-axis>
* 	<igx-trix-indicator
*         #series
*         [xAxis]="xAxis"
*         [yAxis]="yAxis"
*         highMemberPath="high"
* 		lowMemberPath="low"
* 		volumeMemberPath="volume">
*     </igx-trix-indicator>
* </igx-data-chart>
* ```
*
* ```ts
* let series = new IgxTrixIndicatorComponent();
* series.xAxis = this.xAxis;
* series.yAxis = this.yAxis;
* series.openMemberPath = "open";
* series.highMemberPath = "high";
* series.lowMemberPath = "low";
* series.closeMemberPath = "close";
* this.chart.series.add(series);
* ```
*/
export var IgxTRIXIndicatorComponent_PROVIDERS = [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxTRIXIndicatorComponent; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxTRIXIndicatorComponent; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxTRIXIndicatorComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxTRIXIndicatorComponent; }) }];
var IgxTRIXIndicatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxTRIXIndicatorComponent, _super);
    function IgxTRIXIndicatorComponent() {
        return _super.call(this) || this;
    }
    IgxTRIXIndicatorComponent_1 = IgxTRIXIndicatorComponent;
    IgxTRIXIndicatorComponent.prototype.createImplementation = function () {
        return new TRIXIndicator();
    };
    Object.defineProperty(IgxTRIXIndicatorComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxTRIXIndicatorComponent.prototype, "period", {
        /**
         * Gets or sets the moving average period for the current TRIXIndicator object.
         * The typical, and initial, value for TRIX periods is 15.
        *
        * The `period` property is used for moving average period of the current TRIXIndicator object.
        *
        * ```html
        * <igx-data-chart
        *     [dataSource]="data">
        * 	<igx-category-x-axis
        * 		label="label"
        * 		#xAxis>
        *     </igx-category-x-axis>
        *     <igx-numeric-y-axis
        * 		#yAxis>
        *     </igx-numeric-y-axis>
        * 	<igx-trix-indicator
        *         #series
        *         [xAxis]="xAxis"
        *         [yAxis]="yAxis"
        *         highMemberPath="high"
        * 		lowMemberPath="low"
        * 		volumeMemberPath="volume"
        * 		period="7">
        *     </igx-trix-indicator>
        * </igx-data-chart>
        * ```
        *
        * ```ts
        * this.series.period = 30;
        * ```
        */
        get: function () {
            return this.i.qy;
        },
        set: function (v) {
            this.i.qy = +v;
        },
        enumerable: true,
        configurable: true
    });
    var IgxTRIXIndicatorComponent_1;
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxTRIXIndicatorComponent.prototype, "period", null);
    IgxTRIXIndicatorComponent = IgxTRIXIndicatorComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-trix-indicator',
            template: "",
            providers: [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxTRIXIndicatorComponent_1; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxTRIXIndicatorComponent_1; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxTRIXIndicatorComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxTRIXIndicatorComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxTRIXIndicatorComponent);
    return IgxTRIXIndicatorComponent;
}(IgxStrategyBasedIndicatorComponent));
export { IgxTRIXIndicatorComponent };
