// igniteui-angular-charts/esm5/lib/igx-stacked-line-series-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxHorizontalStackedSeriesBaseComponent } from "./igx-horizontal-stacked-series-base-component";
import { IgxStackedSeriesBaseComponent } from "./igx-stacked-series-base-component";
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { StackedLineSeries } from "./StackedLineSeries";
/**
 * Represents a stacked series, where the points in each series are connected with a line.
*/
export var IgxStackedLineSeriesComponent_PROVIDERS = [{ provide: IgxHorizontalStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStackedLineSeriesComponent; }) }, { provide: IgxStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStackedLineSeriesComponent; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxStackedLineSeriesComponent; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxStackedLineSeriesComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxStackedLineSeriesComponent; }) }];
var IgxStackedLineSeriesComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxStackedLineSeriesComponent, _super);
    function IgxStackedLineSeriesComponent() {
        return _super.call(this) || this;
    }
    IgxStackedLineSeriesComponent_1 = IgxStackedLineSeriesComponent;
    IgxStackedLineSeriesComponent.prototype.createImplementation = function () {
        return new StackedLineSeries();
    };
    Object.defineProperty(IgxStackedLineSeriesComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    var IgxStackedLineSeriesComponent_1;
    IgxStackedLineSeriesComponent = IgxStackedLineSeriesComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-stacked-line-series',
            template: "",
            providers: [{ provide: IgxHorizontalStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStackedLineSeriesComponent_1; }) }, { provide: IgxStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStackedLineSeriesComponent_1; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxStackedLineSeriesComponent_1; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxStackedLineSeriesComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxStackedLineSeriesComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxStackedLineSeriesComponent);
    return IgxStackedLineSeriesComponent;
}(IgxHorizontalStackedSeriesBaseComponent));
export { IgxStackedLineSeriesComponent };
