// igniteui-angular-charts/esm5/lib/igx-area-fragment-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxFragmentBaseComponent } from "./igx-fragment-base-component";
import { IgxHorizontalAnchoredCategorySeriesComponent } from "./igx-horizontal-anchored-category-series-component";
import { IgxAnchoredCategorySeriesComponent } from "./igx-anchored-category-series-component";
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { AreaFragment } from "./AreaFragment";
/**
 * Represents one part of a StackedAreaSeries.
*/
export var IgxAreaFragmentComponent_PROVIDERS = [{ provide: IgxFragmentBaseComponent, useExisting: forwardRef(function () { return IgxAreaFragmentComponent; }) }, { provide: IgxHorizontalAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxAreaFragmentComponent; }) }, { provide: IgxAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxAreaFragmentComponent; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxAreaFragmentComponent; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxAreaFragmentComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxAreaFragmentComponent; }) }];
var IgxAreaFragmentComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxAreaFragmentComponent, _super);
    function IgxAreaFragmentComponent() {
        return _super.call(this) || this;
    }
    IgxAreaFragmentComponent_1 = IgxAreaFragmentComponent;
    IgxAreaFragmentComponent.prototype.createImplementation = function () {
        return new AreaFragment();
    };
    Object.defineProperty(IgxAreaFragmentComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxAreaFragmentComponent.prototype, "isAreaOrLine", {
        /**
         * Gets whether the current series shows an area or line shape.
        */
        get: function () {
            return this.i.ck;
        },
        enumerable: true,
        configurable: true
    });
    var IgxAreaFragmentComponent_1;
    IgxAreaFragmentComponent = IgxAreaFragmentComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-area-fragment',
            template: "",
            providers: [{ provide: IgxFragmentBaseComponent, useExisting: forwardRef(function () { return IgxAreaFragmentComponent_1; }) }, { provide: IgxHorizontalAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxAreaFragmentComponent_1; }) }, { provide: IgxAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxAreaFragmentComponent_1; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxAreaFragmentComponent_1; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxAreaFragmentComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxAreaFragmentComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxAreaFragmentComponent);
    return IgxAreaFragmentComponent;
}(IgxFragmentBaseComponent));
export { IgxAreaFragmentComponent };
