// igniteui-angular-charts/esm5/lib/igx-accumulation-distribution-indicator-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxStrategyBasedIndicatorComponent } from "./igx-strategy-based-indicator-component";
import { IgxFinancialIndicatorComponent } from "./igx-financial-indicator-component";
import { IgxFinancialSeriesComponent } from "./igx-financial-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { AccumulationDistributionIndicator } from "./AccumulationDistributionIndicator";
/**
 * Represents a IgxDataChartComponent Accumulation/Distribution indicator series.
 * Default required members: Close, Low, High, Volume
*
* You can use the `IgxAccumulationDistributionIndicatorComponent` to evaluate the suppy and demand of a stock, security, or index.
*
* ```html
* <igx-data-chart
* 	#chart
*     [dataSource]="data" >
*     <igx-category-x-axis
* 		label="label"
* 		#xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
* 		#yAxis>
*     </igx-numeric-y-axis>
*     <igx-accumulation-distribution-indicator
* 		[xAxis]="xAxis"
* 		[yAxis]="yAxis"
* 		openMemberPath="open"
* 		highMemberPath="high"
* 		lowMemberPath="low"
* 		closeMemberPath="close">
*     </igx-accumulation-distribution-indicator>
* </igx-data-chart>
* ```
*
* ```ts
* let series = new IgxAccumulationDistributionIndicatorComponent();
* series.xAxis = this.xAxis;
* series.yAxis = this.yAxis;
* series.openMemberPath = "open";
* series.highMemberPath = "high";
* series.lowMemberPath = "low";
* series.closeMemberPath = "close";
* this.chart.series.add(series);
* ```
*/
export var IgxAccumulationDistributionIndicatorComponent_PROVIDERS = [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxAccumulationDistributionIndicatorComponent; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxAccumulationDistributionIndicatorComponent; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxAccumulationDistributionIndicatorComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxAccumulationDistributionIndicatorComponent; }) }];
var IgxAccumulationDistributionIndicatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxAccumulationDistributionIndicatorComponent, _super);
    function IgxAccumulationDistributionIndicatorComponent() {
        return _super.call(this) || this;
    }
    IgxAccumulationDistributionIndicatorComponent_1 = IgxAccumulationDistributionIndicatorComponent;
    IgxAccumulationDistributionIndicatorComponent.prototype.createImplementation = function () {
        return new AccumulationDistributionIndicator();
    };
    Object.defineProperty(IgxAccumulationDistributionIndicatorComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    var IgxAccumulationDistributionIndicatorComponent_1;
    IgxAccumulationDistributionIndicatorComponent = IgxAccumulationDistributionIndicatorComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-accumulation-distribution-indicator',
            template: "",
            providers: [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxAccumulationDistributionIndicatorComponent_1; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxAccumulationDistributionIndicatorComponent_1; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxAccumulationDistributionIndicatorComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxAccumulationDistributionIndicatorComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxAccumulationDistributionIndicatorComponent);
    return IgxAccumulationDistributionIndicatorComponent;
}(IgxStrategyBasedIndicatorComponent));
export { IgxAccumulationDistributionIndicatorComponent };
