// igniteui-angular-charts/esm5/lib/igx-stacked-100-bar-series-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxStackedBarSeriesComponent } from "./igx-stacked-bar-series-component";
import { IgxVerticalStackedSeriesBaseComponent } from "./igx-vertical-stacked-series-base-component";
import { IgxStackedSeriesBaseComponent } from "./igx-stacked-series-base-component";
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { Stacked100BarSeries } from "./Stacked100BarSeries";
/**
 * Represents a IgxDataChartComponent stacked100 bar series.
*/
export var IgxStacked100BarSeriesComponent_PROVIDERS = [{ provide: IgxStackedBarSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100BarSeriesComponent; }) }, { provide: IgxVerticalStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStacked100BarSeriesComponent; }) }, { provide: IgxStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStacked100BarSeriesComponent; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxStacked100BarSeriesComponent; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100BarSeriesComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100BarSeriesComponent; }) }];
var IgxStacked100BarSeriesComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxStacked100BarSeriesComponent, _super);
    function IgxStacked100BarSeriesComponent() {
        return _super.call(this) || this;
    }
    IgxStacked100BarSeriesComponent_1 = IgxStacked100BarSeriesComponent;
    IgxStacked100BarSeriesComponent.prototype.createImplementation = function () {
        return new Stacked100BarSeries();
    };
    Object.defineProperty(IgxStacked100BarSeriesComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    var IgxStacked100BarSeriesComponent_1;
    IgxStacked100BarSeriesComponent = IgxStacked100BarSeriesComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-stacked-100-bar-series',
            template: "",
            providers: [{ provide: IgxStackedBarSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100BarSeriesComponent_1; }) }, { provide: IgxVerticalStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStacked100BarSeriesComponent_1; }) }, { provide: IgxStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStacked100BarSeriesComponent_1; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxStacked100BarSeriesComponent_1; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100BarSeriesComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100BarSeriesComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxStacked100BarSeriesComponent);
    return IgxStacked100BarSeriesComponent;
}(IgxStackedBarSeriesComponent));
export { IgxStacked100BarSeriesComponent };
