// igniteui-angular-charts/esm5/lib/igx-scale-legend-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxLegendBaseComponent } from "./igx-legend-base-component";
import { ScaleLegend } from "./ScaleLegend";
/**
 * Represents a legend that indicates the size and the color scale for a collection of series.
*/
export var IgxScaleLegendComponent_PROVIDERS = [{ provide: IgxLegendBaseComponent, useExisting: forwardRef(function () { return IgxScaleLegendComponent; }) }];
var IgxScaleLegendComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxScaleLegendComponent, _super);
    function IgxScaleLegendComponent() {
        return _super.call(this) || this;
    }
    IgxScaleLegendComponent_1 = IgxScaleLegendComponent;
    IgxScaleLegendComponent.prototype.createImplementation = function () {
        return new ScaleLegend();
    };
    Object.defineProperty(IgxScaleLegendComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    var IgxScaleLegendComponent_1;
    IgxScaleLegendComponent = IgxScaleLegendComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-scale-legend',
            template: "",
            providers: [{ provide: IgxLegendBaseComponent, useExisting: forwardRef(function () { return IgxScaleLegendComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxScaleLegendComponent);
    return IgxScaleLegendComponent;
}(IgxLegendBaseComponent));
export { IgxScaleLegendComponent };
