// igniteui-angular-charts/esm5/lib/igx-polar-spline-area-series-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxPolarLineSeriesBaseComponent } from "./igx-polar-line-series-base-component";
import { IgxPolarBaseComponent } from "./igx-polar-base-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { PolarSplineAreaSeries } from "./PolarSplineAreaSeries";
/**
 * Represents a IgxDataChartComponent polar spline area series.
*/
export var IgxPolarSplineAreaSeriesComponent_PROVIDERS = [{ provide: IgxPolarLineSeriesBaseComponent, useExisting: forwardRef(function () { return IgxPolarSplineAreaSeriesComponent; }) }, { provide: IgxPolarBaseComponent, useExisting: forwardRef(function () { return IgxPolarSplineAreaSeriesComponent; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxPolarSplineAreaSeriesComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxPolarSplineAreaSeriesComponent; }) }];
var IgxPolarSplineAreaSeriesComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxPolarSplineAreaSeriesComponent, _super);
    function IgxPolarSplineAreaSeriesComponent() {
        return _super.call(this) || this;
    }
    IgxPolarSplineAreaSeriesComponent_1 = IgxPolarSplineAreaSeriesComponent;
    IgxPolarSplineAreaSeriesComponent.prototype.createImplementation = function () {
        return new PolarSplineAreaSeries();
    };
    Object.defineProperty(IgxPolarSplineAreaSeriesComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPolarSplineAreaSeriesComponent.prototype, "stiffness", {
        /**
         * Gets or sets the Stiffness property.
        */
        get: function () {
            return this.i.pi;
        },
        set: function (v) {
            this.i.pi = +v;
        },
        enumerable: true,
        configurable: true
    });
    var IgxPolarSplineAreaSeriesComponent_1;
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPolarSplineAreaSeriesComponent.prototype, "stiffness", null);
    IgxPolarSplineAreaSeriesComponent = IgxPolarSplineAreaSeriesComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-polar-spline-area-series',
            template: "",
            providers: [{ provide: IgxPolarLineSeriesBaseComponent, useExisting: forwardRef(function () { return IgxPolarSplineAreaSeriesComponent_1; }) }, { provide: IgxPolarBaseComponent, useExisting: forwardRef(function () { return IgxPolarSplineAreaSeriesComponent_1; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxPolarSplineAreaSeriesComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxPolarSplineAreaSeriesComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxPolarSplineAreaSeriesComponent);
    return IgxPolarSplineAreaSeriesComponent;
}(IgxPolarLineSeriesBaseComponent));
export { IgxPolarSplineAreaSeriesComponent };
