// igniteui-angular-charts/esm5/lib/igx-stacked-100-spline-series-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxStackedSplineSeriesComponent } from "./igx-stacked-spline-series-component";
import { IgxHorizontalStackedSeriesBaseComponent } from "./igx-horizontal-stacked-series-base-component";
import { IgxStackedSeriesBaseComponent } from "./igx-stacked-series-base-component";
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { Stacked100SplineSeries } from "./Stacked100SplineSeries";
/**
 * Represents a IgxDataChartComponent stacked100 spline series.
*/
export var IgxStacked100SplineSeriesComponent_PROVIDERS = [{ provide: IgxStackedSplineSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100SplineSeriesComponent; }) }, { provide: IgxHorizontalStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStacked100SplineSeriesComponent; }) }, { provide: IgxStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStacked100SplineSeriesComponent; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxStacked100SplineSeriesComponent; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100SplineSeriesComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100SplineSeriesComponent; }) }];
var IgxStacked100SplineSeriesComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxStacked100SplineSeriesComponent, _super);
    function IgxStacked100SplineSeriesComponent() {
        return _super.call(this) || this;
    }
    IgxStacked100SplineSeriesComponent_1 = IgxStacked100SplineSeriesComponent;
    IgxStacked100SplineSeriesComponent.prototype.createImplementation = function () {
        return new Stacked100SplineSeries();
    };
    Object.defineProperty(IgxStacked100SplineSeriesComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    var IgxStacked100SplineSeriesComponent_1;
    IgxStacked100SplineSeriesComponent = IgxStacked100SplineSeriesComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-stacked-100-spline-series',
            template: "",
            providers: [{ provide: IgxStackedSplineSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100SplineSeriesComponent_1; }) }, { provide: IgxHorizontalStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStacked100SplineSeriesComponent_1; }) }, { provide: IgxStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStacked100SplineSeriesComponent_1; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxStacked100SplineSeriesComponent_1; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100SplineSeriesComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxStacked100SplineSeriesComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxStacked100SplineSeriesComponent);
    return IgxStacked100SplineSeriesComponent;
}(IgxStackedSplineSeriesComponent));
export { IgxStacked100SplineSeriesComponent };
