// igniteui-angular-charts/esm5/lib/igx-fast-stochastic-oscillator-indicator-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxStrategyBasedIndicatorComponent } from "./igx-strategy-based-indicator-component";
import { IgxFinancialIndicatorComponent } from "./igx-financial-indicator-component";
import { IgxFinancialSeriesComponent } from "./igx-financial-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { FastStochasticOscillatorIndicator } from "./FastStochasticOscillatorIndicator";
/**
 * Represents a IgxDataChartComponent Fast Stochastic Oscillator indicator series.
 * Default required members: High, Low, Close
*/
export var IgxFastStochasticOscillatorIndicatorComponent_PROVIDERS = [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxFastStochasticOscillatorIndicatorComponent; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxFastStochasticOscillatorIndicatorComponent; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxFastStochasticOscillatorIndicatorComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxFastStochasticOscillatorIndicatorComponent; }) }];
var IgxFastStochasticOscillatorIndicatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxFastStochasticOscillatorIndicatorComponent, _super);
    function IgxFastStochasticOscillatorIndicatorComponent() {
        return _super.call(this) || this;
    }
    IgxFastStochasticOscillatorIndicatorComponent_1 = IgxFastStochasticOscillatorIndicatorComponent;
    IgxFastStochasticOscillatorIndicatorComponent.prototype.createImplementation = function () {
        return new FastStochasticOscillatorIndicator();
    };
    Object.defineProperty(IgxFastStochasticOscillatorIndicatorComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxFastStochasticOscillatorIndicatorComponent.prototype, "period", {
        /**
         * Gets or sets the moving average period for the current FastStochasticOscillatorIndicator object.
         * The typical, and initial, value for FastStochasticOscillatorIndicator periods is 14.
        */
        get: function () {
            return this.i.qy;
        },
        set: function (v) {
            this.i.qy = +v;
        },
        enumerable: true,
        configurable: true
    });
    var IgxFastStochasticOscillatorIndicatorComponent_1;
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxFastStochasticOscillatorIndicatorComponent.prototype, "period", null);
    IgxFastStochasticOscillatorIndicatorComponent = IgxFastStochasticOscillatorIndicatorComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-fast-stochastic-oscillator-indicator',
            template: "",
            providers: [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxFastStochasticOscillatorIndicatorComponent_1; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxFastStochasticOscillatorIndicatorComponent_1; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxFastStochasticOscillatorIndicatorComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxFastStochasticOscillatorIndicatorComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxFastStochasticOscillatorIndicatorComponent);
    return IgxFastStochasticOscillatorIndicatorComponent;
}(IgxStrategyBasedIndicatorComponent));
export { IgxFastStochasticOscillatorIndicatorComponent };
