// igniteui-angular-charts/esm5/lib/igx-scatter-spline-series-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxScatterBaseComponent } from "./igx-scatter-base-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { ScatterSplineSeries } from "./ScatterSplineSeries";
/**
 * Represents a IgxDataChartComponent scatter spline series
*/
export var IgxScatterSplineSeriesComponent_PROVIDERS = [{ provide: IgxScatterBaseComponent, useExisting: forwardRef(function () { return IgxScatterSplineSeriesComponent; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxScatterSplineSeriesComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxScatterSplineSeriesComponent; }) }];
var IgxScatterSplineSeriesComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxScatterSplineSeriesComponent, _super);
    function IgxScatterSplineSeriesComponent() {
        return _super.call(this) || this;
    }
    IgxScatterSplineSeriesComponent_1 = IgxScatterSplineSeriesComponent;
    IgxScatterSplineSeriesComponent.prototype.createImplementation = function () {
        return new ScatterSplineSeries();
    };
    Object.defineProperty(IgxScatterSplineSeriesComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxScatterSplineSeriesComponent.prototype, "stiffness", {
        /**
         * Gets or sets the Stiffness property.
        */
        get: function () {
            return this.i.o2;
        },
        set: function (v) {
            this.i.o2 = +v;
        },
        enumerable: true,
        configurable: true
    });
    var IgxScatterSplineSeriesComponent_1;
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxScatterSplineSeriesComponent.prototype, "stiffness", null);
    IgxScatterSplineSeriesComponent = IgxScatterSplineSeriesComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-scatter-spline-series',
            template: "",
            providers: [{ provide: IgxScatterBaseComponent, useExisting: forwardRef(function () { return IgxScatterSplineSeriesComponent_1; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxScatterSplineSeriesComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxScatterSplineSeriesComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxScatterSplineSeriesComponent);
    return IgxScatterSplineSeriesComponent;
}(IgxScatterBaseComponent));
export { IgxScatterSplineSeriesComponent };
