// igniteui-angular-charts/esm5/lib/igx-average-directional-index-indicator-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxStrategyBasedIndicatorComponent } from "./igx-strategy-based-indicator-component";
import { IgxFinancialIndicatorComponent } from "./igx-financial-indicator-component";
import { IgxFinancialSeriesComponent } from "./igx-financial-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { AverageDirectionalIndexIndicator } from "./AverageDirectionalIndexIndicator";
/**
 * Represents a IgxDataChartComponent Average Directional indicator series.
 * Default required members: High, Low, Close
*
* You can use the `IgxAverageDirectionalIndexIndicatorComponent` to measures trend strength without regard to trend direction.
*
* ```html
* <igx-data-chart
* 	#chart
*     [dataSource]="data" >
*     <igx-category-x-axis
* 		label="label"
* 		#xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
* 		#yAxis>
*     </igx-numeric-y-axis>
*     <igx-average-directional-index-indicator
* 		[xAxis]="xAxis"
* 		[yAxis]="yAxis"
* 		openMemberPath="open"
* 		highMemberPath="high"
* 		lowMemberPath="low"
* 		closeMemberPath="close">
*     </igx-average-directional-index-indicator>
* </igx-data-chart>
* ```
*
* ```ts
* let series = new IgxAverageDirectionalIndexIndicatorComponent();
* series.xAxis = this.xAxis;
* series.yAxis = this.yAxis;
* series.openMemberPath = "open";
* series.highMemberPath = "high";
* series.lowMemberPath = "low";
* series.closeMemberPath = "close";
* this.chart.series.add(series);
* ```
*/
export var IgxAverageDirectionalIndexIndicatorComponent_PROVIDERS = [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxAverageDirectionalIndexIndicatorComponent; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxAverageDirectionalIndexIndicatorComponent; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxAverageDirectionalIndexIndicatorComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxAverageDirectionalIndexIndicatorComponent; }) }];
var IgxAverageDirectionalIndexIndicatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxAverageDirectionalIndexIndicatorComponent, _super);
    function IgxAverageDirectionalIndexIndicatorComponent() {
        return _super.call(this) || this;
    }
    IgxAverageDirectionalIndexIndicatorComponent_1 = IgxAverageDirectionalIndexIndicatorComponent;
    IgxAverageDirectionalIndexIndicatorComponent.prototype.createImplementation = function () {
        return new AverageDirectionalIndexIndicator();
    };
    Object.defineProperty(IgxAverageDirectionalIndexIndicatorComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxAverageDirectionalIndexIndicatorComponent.prototype, "period", {
        /**
         * Gets or sets the moving average period for the current AverageDirectionalIndexIndicator object.
         * The typical, and initial, value for AverageDirectionalIndexIndicator periods is 14.
        *
        * You can use the `period` to set the moving average.
        *
        * ```html
        * <igx-data-chart
        *     [dataSource]="data" >
        *     <igx-category-x-axis
        * 		label="label"
        * 		#xAxis>
        *     </igx-category-x-axis>
        *     <igx-numeric-y-axis
        * 		#yAxis>
        *     </igx-numeric-y-axis>
        *     <igx-average-directional-index-indicator
        * 		#series
        * 		[xAxis]="xAxis"
        * 		[yAxis]="yAxis"
        * 		openMemberPath="open"
        * 		highMemberPath="high"
        * 		lowMemberPath="low"
        * 		closeMemberPath="close"
        * 		period="14">
        *     </igx-average-directional-index-indicator>
        * </igx-data-chart>
        * ```
        *
        * ```ts
        * this.series.period = 14;
        * ```
        */
        get: function () {
            return this.i.qy;
        },
        set: function (v) {
            this.i.qy = +v;
        },
        enumerable: true,
        configurable: true
    });
    var IgxAverageDirectionalIndexIndicatorComponent_1;
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxAverageDirectionalIndexIndicatorComponent.prototype, "period", null);
    IgxAverageDirectionalIndexIndicatorComponent = IgxAverageDirectionalIndexIndicatorComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-average-directional-index-indicator',
            template: "",
            providers: [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxAverageDirectionalIndexIndicatorComponent_1; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxAverageDirectionalIndexIndicatorComponent_1; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxAverageDirectionalIndexIndicatorComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxAverageDirectionalIndexIndicatorComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxAverageDirectionalIndexIndicatorComponent);
    return IgxAverageDirectionalIndexIndicatorComponent;
}(IgxStrategyBasedIndicatorComponent));
export { IgxAverageDirectionalIndexIndicatorComponent };
