// igniteui-angular-charts/esm5/lib/igx-stacked-spline-series-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxHorizontalStackedSeriesBaseComponent } from "./igx-horizontal-stacked-series-base-component";
import { IgxStackedSeriesBaseComponent } from "./igx-stacked-series-base-component";
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { StackedSplineSeries } from "./StackedSplineSeries";
/**
 * Represents a IgxDataChartComponent stacked spline series.
*/
export var IgxStackedSplineSeriesComponent_PROVIDERS = [{ provide: IgxHorizontalStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStackedSplineSeriesComponent; }) }, { provide: IgxStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStackedSplineSeriesComponent; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxStackedSplineSeriesComponent; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxStackedSplineSeriesComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxStackedSplineSeriesComponent; }) }];
var IgxStackedSplineSeriesComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxStackedSplineSeriesComponent, _super);
    function IgxStackedSplineSeriesComponent() {
        return _super.call(this) || this;
    }
    IgxStackedSplineSeriesComponent_1 = IgxStackedSplineSeriesComponent;
    IgxStackedSplineSeriesComponent.prototype.createImplementation = function () {
        return new StackedSplineSeries();
    };
    Object.defineProperty(IgxStackedSplineSeriesComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    var IgxStackedSplineSeriesComponent_1;
    IgxStackedSplineSeriesComponent = IgxStackedSplineSeriesComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-stacked-spline-series',
            template: "",
            providers: [{ provide: IgxHorizontalStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStackedSplineSeriesComponent_1; }) }, { provide: IgxStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStackedSplineSeriesComponent_1; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxStackedSplineSeriesComponent_1; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxStackedSplineSeriesComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxStackedSplineSeriesComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxStackedSplineSeriesComponent);
    return IgxStackedSplineSeriesComponent;
}(IgxHorizontalStackedSeriesBaseComponent));
export { IgxStackedSplineSeriesComponent };
