// igniteui-angular-charts/esm5/lib/igx-percent-change-y-axis-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxNumericYAxisComponent } from "./igx-numeric-y-axis-component";
import { IgxStraightNumericAxisBaseComponent } from "./igx-straight-numeric-axis-base-component";
import { IgxNumericAxisBaseComponent } from "./igx-numeric-axis-base-component";
import { IgxAxisComponent } from "./igx-axis-component";
import { PercentChangeYAxis } from "./PercentChangeYAxis";
/**
 * A horizontal axis that uses a DateTime scale.
*
* You can use the `IgxPercentChangeYAxisComponent` show percentage values.
*
* ```html
* <igx-data-chart
*     [dataSource]="data">
*     <igx-percent-change-y-axis
* 		#yAxis>
*     </igx-percent-change-y-axis>
* </igx-data-chart>
* ```
*
* ```ts
* let yAxis = new IgxPercentChangeYAxisComponent();
* this.chart.axes.add(yAxis);
* ```
*/
export var IgxPercentChangeYAxisComponent_PROVIDERS = [{ provide: IgxNumericYAxisComponent, useExisting: forwardRef(function () { return IgxPercentChangeYAxisComponent; }) }, { provide: IgxStraightNumericAxisBaseComponent, useExisting: forwardRef(function () { return IgxPercentChangeYAxisComponent; }) }, { provide: IgxNumericAxisBaseComponent, useExisting: forwardRef(function () { return IgxPercentChangeYAxisComponent; }) }, { provide: IgxAxisComponent, useExisting: forwardRef(function () { return IgxPercentChangeYAxisComponent; }) }];
var IgxPercentChangeYAxisComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxPercentChangeYAxisComponent, _super);
    function IgxPercentChangeYAxisComponent() {
        return _super.call(this) || this;
    }
    IgxPercentChangeYAxisComponent_1 = IgxPercentChangeYAxisComponent;
    IgxPercentChangeYAxisComponent.prototype.createImplementation = function () {
        return new PercentChangeYAxis();
    };
    Object.defineProperty(IgxPercentChangeYAxisComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    var IgxPercentChangeYAxisComponent_1;
    IgxPercentChangeYAxisComponent = IgxPercentChangeYAxisComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-percent-change-y-axis',
            template: "",
            providers: [{ provide: IgxNumericYAxisComponent, useExisting: forwardRef(function () { return IgxPercentChangeYAxisComponent_1; }) }, { provide: IgxStraightNumericAxisBaseComponent, useExisting: forwardRef(function () { return IgxPercentChangeYAxisComponent_1; }) }, { provide: IgxNumericAxisBaseComponent, useExisting: forwardRef(function () { return IgxPercentChangeYAxisComponent_1; }) }, { provide: IgxAxisComponent, useExisting: forwardRef(function () { return IgxPercentChangeYAxisComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxPercentChangeYAxisComponent);
    return IgxPercentChangeYAxisComponent;
}(IgxNumericYAxisComponent));
export { IgxPercentChangeYAxisComponent };
