// igniteui-angular-charts/esm5/lib/igx-hierarchical-ring-series-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxRingSeriesBaseComponent } from "./igx-ring-series-base-component";
import { HierarchicalRingSeries } from "./HierarchicalRingSeries";
/**
 * Represents a IgxDoughnutChartComponent hierarchical series.
*/
export var IgxHierarchicalRingSeriesComponent_PROVIDERS = [{ provide: IgxRingSeriesBaseComponent, useExisting: forwardRef(function () { return IgxHierarchicalRingSeriesComponent; }) }];
var IgxHierarchicalRingSeriesComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxHierarchicalRingSeriesComponent, _super);
    function IgxHierarchicalRingSeriesComponent() {
        return _super.call(this) || this;
    }
    IgxHierarchicalRingSeriesComponent_1 = IgxHierarchicalRingSeriesComponent;
    IgxHierarchicalRingSeriesComponent.prototype.createImplementation = function () {
        return new HierarchicalRingSeries();
    };
    Object.defineProperty(IgxHierarchicalRingSeriesComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxHierarchicalRingSeriesComponent.prototype, "childrenMemberPath", {
        /**
         * Gets or sets the property name that contains the children items collection.
        */
        get: function () {
            return this.i.eq;
        },
        set: function (v) {
            this.i.eq = v;
        },
        enumerable: true,
        configurable: true
    });
    var IgxHierarchicalRingSeriesComponent_1;
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxHierarchicalRingSeriesComponent.prototype, "childrenMemberPath", null);
    IgxHierarchicalRingSeriesComponent = IgxHierarchicalRingSeriesComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-hierarchical-ring-series',
            template: "",
            providers: [{ provide: IgxRingSeriesBaseComponent, useExisting: forwardRef(function () { return IgxHierarchicalRingSeriesComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxHierarchicalRingSeriesComponent);
    return IgxHierarchicalRingSeriesComponent;
}(IgxRingSeriesBaseComponent));
export { IgxHierarchicalRingSeriesComponent };
