// igniteui-angular-charts/esm5/lib/igx-typical-price-indicator-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxStrategyBasedIndicatorComponent } from "./igx-strategy-based-indicator-component";
import { IgxFinancialIndicatorComponent } from "./igx-financial-indicator-component";
import { IgxFinancialSeriesComponent } from "./igx-financial-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { TypicalPriceIndicator } from "./TypicalPriceIndicator";
/**
 * Represents a IgxDataChartComponent Typical Price indicator series.
 * The typical price indicator is represented as a arithmetic average
 * of the High, Low, and Close for a day.
 * Default required members: High, Low, Close
*
* The `IgxTypicalPriceIndicatorComponent` class represents a IgxDataChartComponent Typical Price indicator series.
*
* ```html
* <igx-data-chart
*     [dataSource]="data">
* 	<igx-category-x-axis
* 		label="label"
* 		#xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
* 		#yAxis>
*     </igx-numeric-y-axis>
* 	<igx-typical-price-indicator
*         #series
*         [xAxis]="xAxis"
*         [yAxis]="yAxis"
*         highMemberPath="high"
* 		lowMemberPath="low"
* 		volumeMemberPath="volume">
*     </igx-typical-price-indicator>
* </igx-data-chart>
* ```
*
* ```ts
* let series = new IgxTypicalPriceIndicatorComponent();
* series.xAxis = this.xAxis;
* series.yAxis = this.yAxis;
* series.openMemberPath = "open";
* series.highMemberPath = "high";
* series.lowMemberPath = "low";
* series.closeMemberPath = "close";
* this.chart.series.add(series);
* ```
*/
export var IgxTypicalPriceIndicatorComponent_PROVIDERS = [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxTypicalPriceIndicatorComponent; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxTypicalPriceIndicatorComponent; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxTypicalPriceIndicatorComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxTypicalPriceIndicatorComponent; }) }];
var IgxTypicalPriceIndicatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxTypicalPriceIndicatorComponent, _super);
    function IgxTypicalPriceIndicatorComponent() {
        return _super.call(this) || this;
    }
    IgxTypicalPriceIndicatorComponent_1 = IgxTypicalPriceIndicatorComponent;
    IgxTypicalPriceIndicatorComponent.prototype.createImplementation = function () {
        return new TypicalPriceIndicator();
    };
    Object.defineProperty(IgxTypicalPriceIndicatorComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    var IgxTypicalPriceIndicatorComponent_1;
    IgxTypicalPriceIndicatorComponent = IgxTypicalPriceIndicatorComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-typical-price-indicator',
            template: "",
            providers: [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxTypicalPriceIndicatorComponent_1; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxTypicalPriceIndicatorComponent_1; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxTypicalPriceIndicatorComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxTypicalPriceIndicatorComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxTypicalPriceIndicatorComponent);
    return IgxTypicalPriceIndicatorComponent;
}(IgxStrategyBasedIndicatorComponent));
export { IgxTypicalPriceIndicatorComponent };
