// igniteui-angular-charts/esm5/lib/igx-stacked-spline-area-series-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxHorizontalStackedSeriesBaseComponent } from "./igx-horizontal-stacked-series-base-component";
import { IgxStackedSeriesBaseComponent } from "./igx-stacked-series-base-component";
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { StackedSplineAreaSeries } from "./StackedSplineAreaSeries";
/**
 * Represents a IgxDataChartComponent stacked spline area series.
*/
export var IgxStackedSplineAreaSeriesComponent_PROVIDERS = [{ provide: IgxHorizontalStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStackedSplineAreaSeriesComponent; }) }, { provide: IgxStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStackedSplineAreaSeriesComponent; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxStackedSplineAreaSeriesComponent; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxStackedSplineAreaSeriesComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxStackedSplineAreaSeriesComponent; }) }];
var IgxStackedSplineAreaSeriesComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxStackedSplineAreaSeriesComponent, _super);
    function IgxStackedSplineAreaSeriesComponent() {
        return _super.call(this) || this;
    }
    IgxStackedSplineAreaSeriesComponent_1 = IgxStackedSplineAreaSeriesComponent;
    IgxStackedSplineAreaSeriesComponent.prototype.createImplementation = function () {
        return new StackedSplineAreaSeries();
    };
    Object.defineProperty(IgxStackedSplineAreaSeriesComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    var IgxStackedSplineAreaSeriesComponent_1;
    IgxStackedSplineAreaSeriesComponent = IgxStackedSplineAreaSeriesComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-stacked-spline-area-series',
            template: "",
            providers: [{ provide: IgxHorizontalStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStackedSplineAreaSeriesComponent_1; }) }, { provide: IgxStackedSeriesBaseComponent, useExisting: forwardRef(function () { return IgxStackedSplineAreaSeriesComponent_1; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxStackedSplineAreaSeriesComponent_1; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxStackedSplineAreaSeriesComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxStackedSplineAreaSeriesComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxStackedSplineAreaSeriesComponent);
    return IgxStackedSplineAreaSeriesComponent;
}(IgxHorizontalStackedSeriesBaseComponent));
export { IgxStackedSplineAreaSeriesComponent };
