// igniteui-angular-charts/esm5/lib/igx-positive-volume-index-indicator-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxStrategyBasedIndicatorComponent } from "./igx-strategy-based-indicator-component";
import { IgxFinancialIndicatorComponent } from "./igx-financial-indicator-component";
import { IgxFinancialSeriesComponent } from "./igx-financial-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { PositiveVolumeIndexIndicator } from "./PositiveVolumeIndexIndicator";
/**
 * Represents a IgxDataChartComponent Positive Volume Index (PVI) indicator series.
 * Default required members: Close, Volume
*
* You can use the `IgxPositiveVolumeIndexIndicatorComponent` with the 'NegativeVolumeIndexIndicator' to identify bull markets.
*
* ```html
* <igx-data-chart
*     [dataSource]="data" >
*     <igx-category-x-axis
* 		label="label"
* 		#xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
* 		#yAxis>
*     </igx-numeric-y-axis>
*     <igx-positive-volume-index-indicator
* 		[xAxis]="xAxis"
* 		[yAxis]="yAxis"
* 		openMemberPath="open"
* 		highMemberPath="high"
* 		lowMemberPath="low"
* 		closeMemberPath="close">
*     </igx-positive-volume-index-indicator>
* </igx-data-chart>
* ```
*
* ```ts
* let series = new IgxPositiveVolumeIndexIndicatorComponent();
* series.xAxis = this.xAxis;
* series.yAxis = this.yAxis;
* series.openMemberPath = "open";
* series.highMemberPath = "high";
* series.lowMemberPath = "low";
* series.closeMemberPath = "close";
* this.chart.series.add(series);
* ```
*/
export var IgxPositiveVolumeIndexIndicatorComponent_PROVIDERS = [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxPositiveVolumeIndexIndicatorComponent; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxPositiveVolumeIndexIndicatorComponent; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxPositiveVolumeIndexIndicatorComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxPositiveVolumeIndexIndicatorComponent; }) }];
var IgxPositiveVolumeIndexIndicatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxPositiveVolumeIndexIndicatorComponent, _super);
    function IgxPositiveVolumeIndexIndicatorComponent() {
        return _super.call(this) || this;
    }
    IgxPositiveVolumeIndexIndicatorComponent_1 = IgxPositiveVolumeIndexIndicatorComponent;
    IgxPositiveVolumeIndexIndicatorComponent.prototype.createImplementation = function () {
        return new PositiveVolumeIndexIndicator();
    };
    Object.defineProperty(IgxPositiveVolumeIndexIndicatorComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    var IgxPositiveVolumeIndexIndicatorComponent_1;
    IgxPositiveVolumeIndexIndicatorComponent = IgxPositiveVolumeIndexIndicatorComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-positive-volume-index-indicator',
            template: "",
            providers: [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxPositiveVolumeIndexIndicatorComponent_1; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxPositiveVolumeIndexIndicatorComponent_1; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxPositiveVolumeIndexIndicatorComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxPositiveVolumeIndexIndicatorComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxPositiveVolumeIndexIndicatorComponent);
    return IgxPositiveVolumeIndexIndicatorComponent;
}(IgxStrategyBasedIndicatorComponent));
export { IgxPositiveVolumeIndexIndicatorComponent };
