// igniteui-angular-charts/esm5/lib/igx-line-series-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { UnknownValuePlotting, UnknownValuePlotting_$type } from "igniteui-angular-core";
import { IgxHorizontalAnchoredCategorySeriesComponent } from "./igx-horizontal-anchored-category-series-component";
import { IgxAnchoredCategorySeriesComponent } from "./igx-anchored-category-series-component";
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { LineSeries } from "./LineSeries";
import { ensureEnum } from "igniteui-angular-core";
/**
 * Represents a IgxDataChartComponent line series.
 * Displays trend over time or ordered categories.Useful when there are many data points and the order is important.
*
* `IgxLineSeriesComponent` belongs to a group of `IgxCategorySeriesComponent` and it is rendered using a collection of points connected by straight line segments.
*
* ```html
*  <igx-data-chart
*     [dataSource]="data">
*     <igx-category-x-axis
*         label="label"
*         #xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
*         #yAxis>
*     </igx-numeric-y-axis>
*     <igx-line-series
*         [xAxis]="xAxis"
*         [yAxis]="yAxis"
*         valueMemberPath="value">
*     </igx-line-series>
* </igx-data-chart>
* ```
*
* ```ts
* let series: IgxLineSeriesComponent = new IgxLineSeriesComponent();
* series.xAxis = this.xAxis;
* series.yAxis = this.yAxis;
* series.valueMemberPath = "value";
* this.chart.series.add(series);
* ```
*/
export var IgxLineSeriesComponent_PROVIDERS = [{ provide: IgxHorizontalAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxLineSeriesComponent; }) }, { provide: IgxAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxLineSeriesComponent; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxLineSeriesComponent; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxLineSeriesComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxLineSeriesComponent; }) }];
var IgxLineSeriesComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxLineSeriesComponent, _super);
    function IgxLineSeriesComponent() {
        return _super.call(this) || this;
    }
    IgxLineSeriesComponent_1 = IgxLineSeriesComponent;
    IgxLineSeriesComponent.prototype.createImplementation = function () {
        return new LineSeries();
    };
    Object.defineProperty(IgxLineSeriesComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxLineSeriesComponent.prototype, "isAreaOrLine", {
        /**
         * Gets whether the current series shows an area or line shape.
        *
        * Gets whether the `IgxSeriesComponent` shows an area or a line shape.
        *
        * ```ts
        * let isAreaOrLine: boolean = series.isAreaOrLine;
        * ```
        */
        get: function () {
            return this.i.ck;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxLineSeriesComponent.prototype, "unknownValuePlotting", {
        /**
         * Determines how unknown values will be plotted on the chart.
         * Null and Double.NaN are two examples of unknown values.
        *
        * Determines how unknown values will be plotted for the `IgxSeriesComponent`. This can be set to `LinearInterpolate` to create a continous shape or `DontPlot` to create breaks in the series.
        *
        * Null and Double.NaN are two examples of unknown values.
        *
        * ```html
        * <igx-data-chart #chart
        *     [dataSource]="data">
        *     <igx-category-x-axis
        *         label="label"
        *         #xAxis>
        *     </igx-category-x-axis>
        *     <igx-numeric-y-axis
        *         #yAxis>
        *     </igx-numeric-y-axis>
        *     <igx-line-series
        *         [xAxis]="xAxis"
        *         [yAxis]="yAxis"
        *         valueMemberPath="value"
        *         unknownValuePlotting="LinearInterpolate">
        *     </igx-line-series>
        * </igx-data-chart>
        * ```
        *
        * ```ts
        * series.unknownValuePlotting = UnknownValuePlotting.LinearInterpolate;
        * ```
        */
        get: function () {
            return this.i.unknownValuePlotting;
        },
        set: function (v) {
            this.i.unknownValuePlotting = ensureEnum(UnknownValuePlotting_$type, v);
        },
        enumerable: true,
        configurable: true
    });
    var IgxLineSeriesComponent_1;
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxLineSeriesComponent.prototype, "unknownValuePlotting", null);
    IgxLineSeriesComponent = IgxLineSeriesComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-line-series',
            template: "",
            providers: [{ provide: IgxHorizontalAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxLineSeriesComponent_1; }) }, { provide: IgxAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxLineSeriesComponent_1; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxLineSeriesComponent_1; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxLineSeriesComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxLineSeriesComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxLineSeriesComponent);
    return IgxLineSeriesComponent;
}(IgxHorizontalAnchoredCategorySeriesComponent));
export { IgxLineSeriesComponent };
