// igniteui-angular-charts/esm5/lib/igx-spline-fragment-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxSplineFragmentBaseComponent } from "./igx-spline-fragment-base-component";
import { IgxSplineSeriesBaseComponent } from "./igx-spline-series-base-component";
import { IgxHorizontalAnchoredCategorySeriesComponent } from "./igx-horizontal-anchored-category-series-component";
import { IgxAnchoredCategorySeriesComponent } from "./igx-anchored-category-series-component";
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { SplineFragment } from "./SplineFragment";
/**
 * Represents one part of a StackedSplineSeries.
*/
export var IgxSplineFragmentComponent_PROVIDERS = [{ provide: IgxSplineFragmentBaseComponent, useExisting: forwardRef(function () { return IgxSplineFragmentComponent; }) }, { provide: IgxSplineSeriesBaseComponent, useExisting: forwardRef(function () { return IgxSplineFragmentComponent; }) }, { provide: IgxHorizontalAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxSplineFragmentComponent; }) }, { provide: IgxAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxSplineFragmentComponent; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxSplineFragmentComponent; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxSplineFragmentComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxSplineFragmentComponent; }) }];
var IgxSplineFragmentComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxSplineFragmentComponent, _super);
    function IgxSplineFragmentComponent() {
        return _super.call(this) || this;
    }
    IgxSplineFragmentComponent_1 = IgxSplineFragmentComponent;
    IgxSplineFragmentComponent.prototype.createImplementation = function () {
        return new SplineFragment();
    };
    Object.defineProperty(IgxSplineFragmentComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxSplineFragmentComponent.prototype, "isAreaOrLine", {
        /**
         * Gets whether the current series shows an area or line shape.
        */
        get: function () {
            return this.i.ck;
        },
        enumerable: true,
        configurable: true
    });
    var IgxSplineFragmentComponent_1;
    IgxSplineFragmentComponent = IgxSplineFragmentComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-spline-fragment',
            template: "",
            providers: [{ provide: IgxSplineFragmentBaseComponent, useExisting: forwardRef(function () { return IgxSplineFragmentComponent_1; }) }, { provide: IgxSplineSeriesBaseComponent, useExisting: forwardRef(function () { return IgxSplineFragmentComponent_1; }) }, { provide: IgxHorizontalAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxSplineFragmentComponent_1; }) }, { provide: IgxAnchoredCategorySeriesComponent, useExisting: forwardRef(function () { return IgxSplineFragmentComponent_1; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxSplineFragmentComponent_1; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxSplineFragmentComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxSplineFragmentComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxSplineFragmentComponent);
    return IgxSplineFragmentComponent;
}(IgxSplineFragmentBaseComponent));
export { IgxSplineFragmentComponent };
