// igniteui-angular-charts/esm5/lib/igx-percentage-volume-oscillator-indicator-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxStrategyBasedIndicatorComponent } from "./igx-strategy-based-indicator-component";
import { IgxFinancialIndicatorComponent } from "./igx-financial-indicator-component";
import { IgxFinancialSeriesComponent } from "./igx-financial-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { PercentageVolumeOscillatorIndicator } from "./PercentageVolumeOscillatorIndicator";
/**
 * Represents a IgxDataChartComponent Percentage Volume Oscillator (PVO) series.
 * Default required members: Volume
*
* You can use the `IgxPercentageVolumeOscillatorIndicatorComponent` to show the percentage difference volume.
*
* ```html
* <igx-data-chart
*     [dataSource]="data" >
*     <igx-category-x-axis
* 		label="label"
* 		#xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
* 		#yAxis>
*     </igx-numeric-y-axis>
*     <igx-percentage-volume-oscillator-indicator
* 		[xAxis]="xAxis"
* 		[yAxis]="yAxis"
* 		openMemberPath="open"
* 		highMemberPath="high"
* 		lowMemberPath="low"
* 		closeMemberPath="close">
*     </igx-percentage-volume-oscillator-indicator>
* </igx-data-chart>
* ```
*
* ```ts
* let series = new IgxPercentageVolumeOscillatorIndicatorComponent();
* series.xAxis = this.xAxis;
* series.yAxis = this.yAxis;
* series.openMemberPath = "open";
* series.highMemberPath = "high";
* series.lowMemberPath = "low";
* series.closeMemberPath = "close";
* this.chart.series.add(series);
* ```
*/
export var IgxPercentageVolumeOscillatorIndicatorComponent_PROVIDERS = [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxPercentageVolumeOscillatorIndicatorComponent; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxPercentageVolumeOscillatorIndicatorComponent; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxPercentageVolumeOscillatorIndicatorComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxPercentageVolumeOscillatorIndicatorComponent; }) }];
var IgxPercentageVolumeOscillatorIndicatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxPercentageVolumeOscillatorIndicatorComponent, _super);
    function IgxPercentageVolumeOscillatorIndicatorComponent() {
        return _super.call(this) || this;
    }
    IgxPercentageVolumeOscillatorIndicatorComponent_1 = IgxPercentageVolumeOscillatorIndicatorComponent;
    IgxPercentageVolumeOscillatorIndicatorComponent.prototype.createImplementation = function () {
        return new PercentageVolumeOscillatorIndicator();
    };
    Object.defineProperty(IgxPercentageVolumeOscillatorIndicatorComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPercentageVolumeOscillatorIndicatorComponent.prototype, "shortPeriod", {
        /**
         * Gets or sets the short moving average period for the current PercentageVolumeOscillatorIndicator object.
         * The typical, and initial, value for short PVO periods is 10.
        *
        * You can use the `shortPeriod` to set the short moving average period.
        *
        * ```html
        * <igx-data-chart
        *     [dataSource]="data" >
        *     <igx-category-x-axis
        * 		label="label"
        * 		#xAxis>
        *     </igx-category-x-axis>
        *     <igx-numeric-y-axis
        * 		#yAxis>
        *     </igx-numeric-y-axis>
        *     <igx-percentage-volume-oscillator-indicator
        * 		#series
        * 		[xAxis]="xAxis"
        * 		[yAxis]="yAxis"
        * 		openMemberPath="open"
        * 		highMemberPath="high"
        * 		lowMemberPath="low"
        * 		closeMemberPath="close"
        * 		shortPeriod="10">
        *     </igx-percentage-volume-oscillator-indicator>
        * </igx-data-chart>
        * ```
        *
        * ```ts
        * this.series.shortPeriod = 30;
        * ```
        */
        get: function () {
            return this.i.shortPeriod;
        },
        set: function (v) {
            this.i.shortPeriod = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPercentageVolumeOscillatorIndicatorComponent.prototype, "longPeriod", {
        /**
         * Gets or sets the short moving average period for the current PercentageVolumeOscillatorIndicator object.
         * The typical, and initial, value for long PVO periods is 30.
        *
        * You can use the `longPeriod` to set the long moving average period.
        *
        * ```html
        * <igx-data-chart
        *     [dataSource]="data" >
        *     <igx-category-x-axis
        * 		label="label"
        * 		#xAxis>
        *     </igx-category-x-axis>
        *     <igx-numeric-y-axis
        * 		#yAxis>
        *     </igx-numeric-y-axis>
        *     <igx-percentage-volume-oscillator-indicator
        * 		#series
        * 		[xAxis]="xAxis"
        * 		[yAxis]="yAxis"
        * 		openMemberPath="open"
        * 		highMemberPath="high"
        * 		lowMemberPath="low"
        * 		closeMemberPath="close"
        * 		longPeriod="30">
        *     </igx-percentage-volume-oscillator-indicator>
        * </igx-data-chart>
        * ```
        *
        * ```ts
        * this.series.longPeriod = 30;
        * ```
        */
        get: function () {
            return this.i.longPeriod;
        },
        set: function (v) {
            this.i.longPeriod = +v;
        },
        enumerable: true,
        configurable: true
    });
    var IgxPercentageVolumeOscillatorIndicatorComponent_1;
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPercentageVolumeOscillatorIndicatorComponent.prototype, "shortPeriod", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPercentageVolumeOscillatorIndicatorComponent.prototype, "longPeriod", null);
    IgxPercentageVolumeOscillatorIndicatorComponent = IgxPercentageVolumeOscillatorIndicatorComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-percentage-volume-oscillator-indicator',
            template: "",
            providers: [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxPercentageVolumeOscillatorIndicatorComponent_1; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxPercentageVolumeOscillatorIndicatorComponent_1; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxPercentageVolumeOscillatorIndicatorComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxPercentageVolumeOscillatorIndicatorComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxPercentageVolumeOscillatorIndicatorComponent);
    return IgxPercentageVolumeOscillatorIndicatorComponent;
}(IgxStrategyBasedIndicatorComponent));
export { IgxPercentageVolumeOscillatorIndicatorComponent };
