// igniteui-angular-charts/esm5/lib/igx-polar-scatter-series-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxPolarBaseComponent } from "./igx-polar-base-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { PolarScatterSeries } from "./PolarScatterSeries";
/**
 * Represents a IgxDataChartComponent polar scatter series.
*/
export var IgxPolarScatterSeriesComponent_PROVIDERS = [{ provide: IgxPolarBaseComponent, useExisting: forwardRef(function () { return IgxPolarScatterSeriesComponent; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxPolarScatterSeriesComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxPolarScatterSeriesComponent; }) }];
var IgxPolarScatterSeriesComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxPolarScatterSeriesComponent, _super);
    function IgxPolarScatterSeriesComponent() {
        return _super.call(this) || this;
    }
    IgxPolarScatterSeriesComponent_1 = IgxPolarScatterSeriesComponent;
    IgxPolarScatterSeriesComponent.prototype.createImplementation = function () {
        return new PolarScatterSeries();
    };
    Object.defineProperty(IgxPolarScatterSeriesComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    var IgxPolarScatterSeriesComponent_1;
    IgxPolarScatterSeriesComponent = IgxPolarScatterSeriesComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-polar-scatter-series',
            template: "",
            providers: [{ provide: IgxPolarBaseComponent, useExisting: forwardRef(function () { return IgxPolarScatterSeriesComponent_1; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxPolarScatterSeriesComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxPolarScatterSeriesComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxPolarScatterSeriesComponent);
    return IgxPolarScatterSeriesComponent;
}(IgxPolarBaseComponent));
export { IgxPolarScatterSeriesComponent };
