// igniteui-angular-charts/esm5/lib/igx-range-area-series-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxHorizontalRangeCategorySeriesComponent } from "./igx-horizontal-range-category-series-component";
import { IgxRangeCategorySeriesComponent } from "./igx-range-category-series-component";
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { RangeAreaSeries } from "./RangeAreaSeries";
/**
 * Represents a IgxDataChartComponent range area series.
*
* The `IgxRangeAreaSeriesComponent` class represents the range area series of the IgxDataChartComponent.
*
* ```html
* <igx-data-chart
* [dataSource]="data">
* <igx-category-x-axis
* 	label="label"
* 	#xAxis>
* </igx-category-x-axis>
* <igx-numeric-y-axis
* 	#yAxis>
* </igx-numeric-y-axis>
*     <igx-range-area-series
*       [xAxis]="xAxis"
*       [yAxis]="yAxis"
*       highMemberPath="high"
*       lowMemberPath="low">
*   </igx-range-area-series>
*     </igx-data-chart>
* </igx-data-chart>
* ```
*
* ```ts
* let series: IgxRangeAreaSeriesComponent = new IgxRangeAreaSeriesComponent();
* series.xAxis = this.xAxis;
* series.yAxis = this.yAxis;
* series.highMemberPath = "High";
* series.lowMemberPath = "Low";
* this.chart.series.add(series);
* ```
*/
export var IgxRangeAreaSeriesComponent_PROVIDERS = [{ provide: IgxHorizontalRangeCategorySeriesComponent, useExisting: forwardRef(function () { return IgxRangeAreaSeriesComponent; }) }, { provide: IgxRangeCategorySeriesComponent, useExisting: forwardRef(function () { return IgxRangeAreaSeriesComponent; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxRangeAreaSeriesComponent; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxRangeAreaSeriesComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxRangeAreaSeriesComponent; }) }];
var IgxRangeAreaSeriesComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxRangeAreaSeriesComponent, _super);
    function IgxRangeAreaSeriesComponent() {
        return _super.call(this) || this;
    }
    IgxRangeAreaSeriesComponent_1 = IgxRangeAreaSeriesComponent;
    IgxRangeAreaSeriesComponent.prototype.createImplementation = function () {
        return new RangeAreaSeries();
    };
    Object.defineProperty(IgxRangeAreaSeriesComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxRangeAreaSeriesComponent.prototype, "isAreaOrLine", {
        /**
         * Gets whether the current series shows an area or line shape.
        *
        * Use the `isAreaOrLine` property to check if the current series shows an area or line shape.
        *
        * ```ts
        * var x= series.isAreaOrLine;
        * ```
        */
        get: function () {
            return this.i.ck;
        },
        enumerable: true,
        configurable: true
    });
    var IgxRangeAreaSeriesComponent_1;
    IgxRangeAreaSeriesComponent = IgxRangeAreaSeriesComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-range-area-series',
            template: "",
            providers: [{ provide: IgxHorizontalRangeCategorySeriesComponent, useExisting: forwardRef(function () { return IgxRangeAreaSeriesComponent_1; }) }, { provide: IgxRangeCategorySeriesComponent, useExisting: forwardRef(function () { return IgxRangeAreaSeriesComponent_1; }) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(function () { return IgxRangeAreaSeriesComponent_1; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxRangeAreaSeriesComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxRangeAreaSeriesComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxRangeAreaSeriesComponent);
    return IgxRangeAreaSeriesComponent;
}(IgxHorizontalRangeCategorySeriesComponent));
export { IgxRangeAreaSeriesComponent };
