// igniteui-angular-charts/esm5/lib/igx-stoch-rsi-indicator-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxStrategyBasedIndicatorComponent } from "./igx-strategy-based-indicator-component";
import { IgxFinancialIndicatorComponent } from "./igx-financial-indicator-component";
import { IgxFinancialSeriesComponent } from "./igx-financial-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { StochRSIIndicator } from "./StochRSIIndicator";
/**
 * Represents a IgxDataChartComponent StochRSI indicator series.
 * Default required members: Close
*
* `IgxStochRSIIndicatorComponent` or Stochastic Relative Strength IndexI Indicator (SRSI) measures when a security is overbought or oversold within a specified period of time. The values range from 0 to 1. The StochRSI Indicator is calculated by applying the Stochastic Oscillator formula to RelativeStrengthIndexIndicator (RSI) data.
*
* Using this indicator requires setting the `CloseMemberPath`.
*
* ```html
* <igx-data-chart
*     width=500px
*     height=500px
*     [dataSource]="data" >
*     <igx-ordinal-time-x-axis
*         dateTimeMemberPath="TimeStamp"
*         #xAxis>
*       </igx-ordinal-time-x-axis>
*     <igx-numeric-y-axis
*        #yAxis></igx-numeric-y-axis>
*     <igx-stoch-rsi-indicator
*        [xAxis]="xAxis"
*        [yAxis]="yAxis"
*        closeMemberPath="Close">
*      </igx-stoch-rsi-indicator>
* </igx-data-chart>
* ```
*
* ```ts
* let series = new IgxStochRSIIndicatorComponent();
* series.xAxis = this.xAxis;
* series.yAxis = this.yAxis;
* series.closeMemberPath = "close";
* this.chart.series.add(series);
* ```
*/
export var IgxStochRSIIndicatorComponent_PROVIDERS = [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxStochRSIIndicatorComponent; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxStochRSIIndicatorComponent; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxStochRSIIndicatorComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxStochRSIIndicatorComponent; }) }];
var IgxStochRSIIndicatorComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxStochRSIIndicatorComponent, _super);
    function IgxStochRSIIndicatorComponent() {
        return _super.call(this) || this;
    }
    IgxStochRSIIndicatorComponent_1 = IgxStochRSIIndicatorComponent;
    IgxStochRSIIndicatorComponent.prototype.createImplementation = function () {
        return new StochRSIIndicator();
    };
    Object.defineProperty(IgxStochRSIIndicatorComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxStochRSIIndicatorComponent.prototype, "period", {
        /**
         * Gets or sets the moving average period for the current StochRSIIndicator object.
         * The typical, and initial, value for StochRSI periods is 14.
        *
        * ```html
        * <igx-data-chart
        *     width=500px
        *     height=500px
        *     [dataSource]="data" >
        *     <igx-ordinal-time-x-axis
        *         dateTimeMemberPath="TimeStamp"
        *         #xAxis>
        *       </igx-ordinal-time-x-axis>
        *     <igx-numeric-y-axis
        *        #yAxis></igx-numeric-y-axis>
        *     <igx-stoch-rsi-indicator
        *        [xAxis]="xAxis"
        *        [yAxis]="yAxis"
        *        closeMemberPath="Close"
        *        period=10>
        *      </igx-stoch-rsi-indicator>
        * </igx-data-chart>
        * ```
        *
        * ```ts
        * this.series.period = 10;
        * ```
        */
        get: function () {
            return this.i.qy;
        },
        set: function (v) {
            this.i.qy = +v;
        },
        enumerable: true,
        configurable: true
    });
    var IgxStochRSIIndicatorComponent_1;
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxStochRSIIndicatorComponent.prototype, "period", null);
    IgxStochRSIIndicatorComponent = IgxStochRSIIndicatorComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-stoch-rsi-indicator',
            template: "",
            providers: [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(function () { return IgxStochRSIIndicatorComponent_1; }) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(function () { return IgxStochRSIIndicatorComponent_1; }) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(function () { return IgxStochRSIIndicatorComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxStochRSIIndicatorComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxStochRSIIndicatorComponent);
    return IgxStochRSIIndicatorComponent;
}(IgxStrategyBasedIndicatorComponent));
export { IgxStochRSIIndicatorComponent };
