// igniteui-angular-charts/esm5/lib/igx-polar-spline-series-component.js
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxPolarLineSeriesBaseComponent } from "./igx-polar-line-series-base-component";
import { IgxPolarBaseComponent } from "./igx-polar-base-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { PolarSplineSeries } from "./PolarSplineSeries";
/**
 * Represents a IgxDataChartComponent polar spline series.
*/
export var IgxPolarSplineSeriesComponent_PROVIDERS = [{ provide: IgxPolarLineSeriesBaseComponent, useExisting: forwardRef(function () { return IgxPolarSplineSeriesComponent; }) }, { provide: IgxPolarBaseComponent, useExisting: forwardRef(function () { return IgxPolarSplineSeriesComponent; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxPolarSplineSeriesComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxPolarSplineSeriesComponent; }) }];
var IgxPolarSplineSeriesComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxPolarSplineSeriesComponent, _super);
    function IgxPolarSplineSeriesComponent() {
        return _super.call(this) || this;
    }
    IgxPolarSplineSeriesComponent_1 = IgxPolarSplineSeriesComponent;
    IgxPolarSplineSeriesComponent.prototype.createImplementation = function () {
        return new PolarSplineSeries();
    };
    Object.defineProperty(IgxPolarSplineSeriesComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxPolarSplineSeriesComponent.prototype, "stiffness", {
        /**
         * Gets or sets the Stiffness property.
        */
        get: function () {
            return this.i.pi;
        },
        set: function (v) {
            this.i.pi = +v;
        },
        enumerable: true,
        configurable: true
    });
    var IgxPolarSplineSeriesComponent_1;
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxPolarSplineSeriesComponent.prototype, "stiffness", null);
    IgxPolarSplineSeriesComponent = IgxPolarSplineSeriesComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-polar-spline-series',
            template: "",
            providers: [{ provide: IgxPolarLineSeriesBaseComponent, useExisting: forwardRef(function () { return IgxPolarSplineSeriesComponent_1; }) }, { provide: IgxPolarBaseComponent, useExisting: forwardRef(function () { return IgxPolarSplineSeriesComponent_1; }) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(function () { return IgxPolarSplineSeriesComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxPolarSplineSeriesComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxPolarSplineSeriesComponent);
    return IgxPolarSplineSeriesComponent;
}(IgxPolarLineSeriesBaseComponent));
export { IgxPolarSplineSeriesComponent };
