// igniteui-angular-charts/esm5/lib/igx-category-chart-component.js
import * as tslib_1 from "tslib";
import { Component, ViewContainerRef, Injector, ComponentFactoryResolver, Renderer2, ViewChild, Input, NgZone, ChangeDetectionStrategy } from '@angular/core';
import { AngularRenderer, AngularWrapper } from "igniteui-angular-core";
import { TypeRegistrar } from "igniteui-angular-core";
import { CategoryChart } from './CategoryChart';
import { CategoryChartType, CategoryChartType_$type } from './CategoryChartType';
import { CategorySeriesMarkerCollisionAvoidance, CategorySeriesMarkerCollisionAvoidance_$type } from './CategorySeriesMarkerCollisionAvoidance';
import { CategoryTransitionInMode, CategoryTransitionInMode_$type } from './CategoryTransitionInMode';
import { TransitionInSpeedType, TransitionInSpeedType_$type } from './TransitionInSpeedType';
import { fromBrushCollection, toBrushCollection, ensureEnum, ensureBool, NamePatcher } from "igniteui-angular-core";
import { IgxXYChartComponent } from './igx-xy-chart-component';
import { DataChartStylingDefaults } from './DataChartStylingDefaults';
import { DataSeriesType } from "igniteui-angular-core";
import { IgxDataContext } from "igniteui-angular-core";
import { IgxCategoryXAxisComponent } from './igx-category-x-axis-component';
import { IgxNumericYAxisComponent } from './igx-numeric-y-axis-component';
export var IgxCategoryChartComponent_PROVIDERS = [];
/**
 * Represents a chart with an ordinal X-axis and a numeric Y-axis.
*
* **Ignite UI for Angular Category Chart** - [Documentation](https://www.infragistics.com/products/ignite-ui-angular/angular/components/CategoryChart.html)
*
* The Ignite UI Category Chart is a lightweight, highly performant chart. It can be easily configured to display category data using an extremely simple and intuitive API. All you need to do is provide your data to the chart and it will take care of everything else.
*
* Example:
*
* ```html
* <igx-category-chart
*     [dataSource]="data"    >
* </igx-category-chart>
* ```
*/
var IgxCategoryChartComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxCategoryChartComponent, _super);
    function IgxCategoryChartComponent(renderer, _elRef, ngZone, injector, componentFactoryResolver) {
        var _this = _super.call(this) || this;
        _this.renderer = renderer;
        _this._elRef = _elRef;
        _this.ngZone = ngZone;
        _this.injector = injector;
        _this.componentFactoryResolver = componentFactoryResolver;
        _this._dataSource = null;
        _this._defaultTooltips = null;
        if (_this._styling) {
            NamePatcher.ensureStylablePatched(Object.getPrototypeOf(_this));
        }
        _this._zoneRunner = function (act) { return ngZone.run(act); };
        _this.container = renderer.createElement("div");
        renderer.appendChild(_elRef.element.nativeElement, _this.container);
        renderer.setStyle(_this.container, "display", "block");
        renderer.setStyle(_this.container, "width", "100%");
        renderer.setStyle(_this.container, "height", "100%");
        var root;
        root = _this.container;
        if (_this.container != null) {
            root = _this.container;
        }
        var ren = new AngularRenderer(root, _this.renderer, window.document, _this.ngZone, true, DataChartStylingDefaults);
        _this._wrapper = ren;
        var chart = _this.i;
        _this._chart = chart;
        if (chart.xAxis) {
            var x = new IgxCategoryXAxisComponent();
            x._provideRenderer(ren);
            x._implementation = chart.xAxis;
            chart.xAxis.externalObject = x;
        }
        if (chart.yAxis) {
            var y = new IgxNumericYAxisComponent();
            y._provideRenderer(ren);
            y._implementation = chart.yAxis;
            chart.yAxis.externalObject = y;
        }
        chart.provideContainer(ren);
        _this.bindData();
        chart.notifyResized();
        ren.addSizeWatcher(function () {
            _this._chart.notifyResized();
        });
        return _this;
    }
    Object.defineProperty(IgxCategoryChartComponent.prototype, "height", {
        get: function () {
            return this._height;
        },
        set: function (value) {
            this._height = value;
            this.renderer.setStyle(this._elRef.element.nativeElement, "height", value);
            this._chart.notifyResized();
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "width", {
        get: function () {
            return this._width;
        },
        set: function (value) {
            this._width = value;
            this.renderer.setStyle(this._elRef.element.nativeElement, "width", value);
            this._chart.notifyResized();
        },
        enumerable: true,
        configurable: true
    });
    IgxCategoryChartComponent.prototype.ngOnDestroy = function () {
        this._chart.destroy();
        this._wrapper.destroy();
    };
    IgxCategoryChartComponent.prototype.createImplementation = function () {
        return new CategoryChart();
    };
    Object.defineProperty(IgxCategoryChartComponent.prototype, "i", {
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    IgxCategoryChartComponent.prototype.createSeriesComponent = function (type) {
        if (TypeRegistrar.isRegistered(type)) {
            var s = TypeRegistrar.create(type);
            s.owner = this;
            s._provideRenderer(this.renderer);
            return s;
        }
        else {
            //we shouldn't get here, hopefully.
            throw Error("series type not loaded: " + type);
        }
    };
    IgxCategoryChartComponent.prototype.createSeries = function (type) {
        switch (type) {
            case DataSeriesType.Area:
                return this.createSeriesComponent('IgxAreaSeriesComponent');
            case DataSeriesType.Column:
                return this.createSeriesComponent('IgxColumnSeriesComponent');
            case DataSeriesType.Line:
                return this.createSeriesComponent('IgxLineSeriesComponent');
            case DataSeriesType.Point:
                return this.createSeriesComponent('IgxPointSeriesComponent');
            case DataSeriesType.Spline:
                return this.createSeriesComponent('IgxSplineSeriesComponent');
            case DataSeriesType.SplineArea:
                return this.createSeriesComponent('IgxSplineAreaSeriesComponent');
            case DataSeriesType.StepArea:
                return this.createSeriesComponent('IgxStepAreaSeriesComponent');
            case DataSeriesType.StepLine:
                return this.createSeriesComponent('IgxStepLineSeriesComponent');
            case DataSeriesType.Waterfall:
                return this.createSeriesComponent('IgxWaterfallSeriesComponent');
            case DataSeriesType.ItemToolTipLayer:
                return this.createSeriesComponent('IgxItemToolTipLayerComponent');
            case DataSeriesType.CategoryToolTipLayer:
                return this.createSeriesComponent('IgxCategoryToolTipLayerComponent');
            case DataSeriesType.CrosshairLayer:
                return this.createSeriesComponent('IgxCrosshairLayerComponent');
            case DataSeriesType.FinalValueLayer:
                return this.createSeriesComponent('IgxFinalValueLayerComponent');
            case DataSeriesType.CalloutLayer:
                return this.createSeriesComponent('IgxCalloutLayerComponent');
            case DataSeriesType.CategoryHighlightLayer:
                return this.createSeriesComponent('IgxCategoryHighlightLayerComponent');
            case DataSeriesType.CategoryItemHighlightLayer:
                return this.createSeriesComponent('IgxCategoryItemHighlightLayerComponent');
            default:
                return this.createSeriesComponent('IgxColumnSeriesComponent');
        }
    };
    Object.defineProperty(IgxCategoryChartComponent.prototype, "dataSource", {
        get: function () {
            return this._dataSource;
        },
        set: function (value) {
            this._dataSource = value;
            this.bindData();
        },
        enumerable: true,
        configurable: true
    });
    IgxCategoryChartComponent.prototype.bindData = function () {
        if (this._chart != null && this._chart !== undefined) {
            this._chart.itemsSource = this._dataSource;
        }
    };
    IgxCategoryChartComponent.prototype.ngAfterContentInit = function () {
        var _this = this;
        if (TypeRegistrar.isRegistered("IgxDataChartDefaultTooltipsComponent")) {
            var c = TypeRegistrar.get("IgxDataChartDefaultTooltipsComponent");
            var cf = this.componentFactoryResolver.resolveComponentFactory(c);
            //let cr = cf.create(this.injector);
            var cr_1 = this._dynamicContent.createComponent(cf);
            this._defaultTooltips = cr_1;
            cr_1.instance.onContentReady.subscribe(function () {
                _this._onDefaultTooltipsReady(cr_1);
            });
            this.container.appendChild(cr_1.location.nativeElement);
        }
        this._styling(this.container, this);
    };
    IgxCategoryChartComponent.prototype.createTooltip = function () {
        if (!TypeRegistrar.isRegistered("IgxTooltipContainerComponent")) {
            return null;
        }
        var t = TypeRegistrar.get("IgxTooltipContainerComponent");
        var cf = this.componentFactoryResolver.resolveComponentFactory(t);
        //let cr = cf.create(this.injector);
        var cr = this._dynamicContent.createComponent(cf);
        var ele = cr.location.nativeElement;
        var self = this;
        ele.updateToolTip = function (c, isSubContent) {
            if (c.externalObject) {
                c = c.externalObject;
            }
            else {
                var ext = new IgxDataContext();
                ext._implementation = c;
                c = ext;
            }
            if (!isSubContent) {
                if (ele.parentElement != self.container) {
                    if (ele.parentElement != null) {
                        ele.parentElement.removeChild(ele);
                    }
                    self.container.appendChild(ele);
                }
            }
            else {
                c.isSubContent = true;
            }
            cr.instance.context = c;
            ele.style.display = "block";
            return true;
        };
        ele.hideToolTip = function () {
            ele.style.display = "none";
        };
        ele.style.display = "none";
        return cr;
    };
    IgxCategoryChartComponent.prototype._ensureTooltipCreated = function (series) {
        var _this = this;
        series._ensureTooltipCreated(function () { return _this.createTooltip(); }, function (ele) {
            var wrapper = new AngularWrapper(ele, _this.renderer, _this.ngZone);
            wrapper.updateToolTip = ele.updateToolTip;
            wrapper.hideToolTip = ele.hideToolTip;
            return wrapper;
        });
    };
    IgxCategoryChartComponent.prototype._createXAxis = function () {
        var x = new IgxCategoryXAxisComponent();
        x._provideRenderer(this.renderer);
        return x.i;
    };
    IgxCategoryChartComponent.prototype._createYAxis = function () {
        var y = new IgxNumericYAxisComponent();
        y.provideRenderer(this.renderer);
        return y.i;
    };
    IgxCategoryChartComponent.prototype._ensureDefaultTooltip = function (series) {
        if (this._defaultTooltips == null) {
            return;
        }
        this._defaultTooltips.instance["ensureDefaultTooltip"](series);
    };
    IgxCategoryChartComponent.prototype._onDefaultTooltipsReady = function (cr) {
        if (this.i.dataChart) {
            var currSeries = this.i.dataChart.series;
            for (var i = 0; i < currSeries.count; i++) {
                if (currSeries.item(i).externalObject.showDefaultTooltip) {
                    this._ensureDefaultTooltip(currSeries.item(i).externalObject);
                }
            }
        }
    };
    Object.defineProperty(IgxCategoryChartComponent.prototype, "transitionInDuration", {
        /**
     * Gets or sets the duration used for animating series plots when the chart is loading into view
    *
    * `transitionInDuration` controls the length of time taken by the transition-in animation.  Try setting it to 2 seconds:
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     isTransitionInEnabled="true"
    *     transitionInDuration="2000">
    * </igx-category-chart>
    * ```
    */
        get: function () {
            return this.i.ql;
        },
        set: function (v) {
            this.i.ql = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "transitionInEasingFunction", {
        /**
         * Gets or sets the easing function used for animating series plots when the chart is loading into view
         * This can be set to one of the known values "linear" or "cubic," or it can be set to an easing function which takes a single numeric parameter and returns a number.
        */
        get: function () {
            return this.i.pk;
        },
        set: function (v) {
            this.i.pk = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "chartType", {
        /**
         * Gets or sets the type of chart series to generate from the data.
        *
        * The `chartType` property determines the type of data series to display.
        *
        * ```html
        * <igx-category-chart
        *     [dataSource]="data"
        *     chartType="stepArea">
        * </igx-category-chart>
        * ```
        */
        get: function () {
            return this.i.o4;
        },
        set: function (v) {
            this.i.o4 = ensureEnum(CategoryChartType_$type, v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "markerCollisionAvoidance", {
        /**
         * Gets or sets the behavior for markers in each series which are placed too close together for the current view, resulting in a collision.
        *
        * `markerCollisionAvoidance` controls the technique the chart uses to avoid overlapping markers.
        *
        * ```html
        * <igx-category-chart
        *     [dataSource]="data"
        *     markerCollisionAvoidance="none">
        * </igx-category-chart>
        * ```
        */
        get: function () {
            return this.i.o6;
        },
        set: function (v) {
            this.i.o6 = ensureEnum(CategorySeriesMarkerCollisionAvoidance_$type, v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "isTransitionInEnabled", {
        /**
         * Gets or sets whether animation of series plots is enabled when the chart is loading into view
        *
        * Set `isTransitionInEnabled` to true if you want your chart series to animate into position when the chart is loading into view.
        *
        * ```html
        * <igx-category-chart
        *     [dataSource]="data"
        *     isTransitionInEnabled="true">
        * </igx-category-chart>
        * ```
        */
        get: function () {
            return this.i.ps;
        },
        set: function (v) {
            this.i.ps = ensureBool(v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "transitionInMode", {
        /**
         * Gets or sets the method that determines how to animate series plots when the chart is loading into view
        *
        * `transitionInMode` controls the direction of the transition-in animation.
        *
        * ```html
        * <igx-category-chart
        *     [dataSource]="data"
        *     isTransitionInEnabled="true"
        *     transitionInMode="accordionfromRight">
        * </igx-category-chart>
        * ```
        */
        get: function () {
            return this.i.o8;
        },
        set: function (v) {
            this.i.o8 = ensureEnum(CategoryTransitionInMode_$type, v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "transitionInSpeedType", {
        /**
         * Gets or sets the arrival speed used for animating series plots when the chart is loading into view
        *
        * `transitionInSpeedType` controls the speed of the transition-in animation.
        *
        * ```html
        * <igx-category-chart
        *     [dataSource]="data"
        *     isTransitionInEnabled="true"
        *     transitionInSpeedType="indexScaled">
        * </igx-category-chart>
        * ```
        */
        get: function () {
            return this.i.pi;
        },
        set: function (v) {
            this.i.pi = ensureEnum(TransitionInSpeedType_$type, v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "xAxisInterval", {
        /**
         * Gets or sets the frequency of displayed labels along the X-axis.
         * Gets or sets the set value is a factor that determines which labels will be hidden. For example, an interval of 2 will display every other label.
        *
        * `xAxisInterval` determines how often to show a label, tickmark, and/or gridline along the x-axis.  Set this property to _n_ to display a label every _nth_ item.
        *
        * ```html
        * <igx-category-chart
        *     [dataSource]="data"
        *     xAxisInterval="3">
        * </igx-category-chart>
        * ```
        */
        get: function () {
            return this.i.qa;
        },
        set: function (v) {
            this.i.qa = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "xAxisMinorInterval", {
        /**
         * Gets or sets the frequency of displayed minor lines along the X-axis.
         * Gets or sets the set value is a factor that determines how the minor lines will be displayed.
        *
        * `xAxisMinorInterval` determines how often to show a minor gridline along the x-axis.  This property is relevant only when the displayed series is a type with grouping, like column series.
        *
        * `xAxisMinorInterval` is expressed as a number between 0 and 1, representing the frequency of the interval.  To display minor gridlines representing 10ths of an item width, set `xAxisMinorInterval` to 0.1.
        *
        * ```html
        * <igx-category-chart
        *     [dataSource]="data"
        *     xAxisMinorInterval="0.1"
        *     xAxisMinorStroke="green"
        *     xAxisMinorStrokeThickness="1">
        * </igx-category-chart>
        * ```
        */
        get: function () {
            return this.i.qb;
        },
        set: function (v) {
            this.i.qb = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "xAxisGap", {
        /**
         * Gets or sets the amount of space between adjacent categories for the X-axis.
         * The gap is silently clamped to the range [0, 1] when used.
        *
        * Use the `xAxisGap` property to configure the spacing between items on the x-axis.  This property is relevant only when the displayed series is a type with item spacing, like column series.
        *
        * An `xAxisGap` of 0 allocates no space between items.  An `xAxisGap` of 1 allocates a space between items equal to the width of one item.
        *
        * To set the item spacing to 75% the width of one item, set the `xAxisGap` to 0.75, as in this code:
        *
        * ```html
        * <igx-category-chart
        *     [dataSource]="data"
        *     xAxisGap="0.75">
        * </igx-category-chart>
        * ```
        */
        get: function () {
            return this.i.p9;
        },
        set: function (v) {
            this.i.p9 = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "xAxisOverlap", {
        /**
         * Gets or sets the amount of overlap between adjacent categories for the X-axis.
         * Gets or sets the overlap is silently clamped to the range [-1, 1] when used.
        *
        * Use the `xAxisOverlap` property to configure the spacing between items on the x-axis.  This property is relevant only when the displayed series is a type with item spacing, like column series.
        *
        * An `xAxisOverlap` of 0 places grouped items adjacent to each other.  An `xAxisOverlap` of 1 places grouped items in the same axis space, completely overlapping.  An `xAxisOverlap` of -1 places a space between grouped items equal to the width of one item.
        *
        * To place grouped items with 75% overlap, set the `xAxisOverlap` to 0.75, as in this code:
        *
        * ```html
        * <igx-category-chart
        *     [dataSource]="data"
        *     xAxisOverlap="0.75">
        * </igx-category-chart>
        * ```
        */
        get: function () {
            return this.i.qc;
        },
        set: function (v) {
            this.i.qc = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "yAxisInterval", {
        /**
         * Gets or sets the distance between each label and grid line along the Y-axis.
        *
        * `yAxisInterval` determines how often to show a label, tickmark, and/or gridline along the y-axis.  Set this property to a number less than the numeric range of the y-axis.
        *
        * ```html
        * <igx-category-chart
        *     [dataSource]="data"
        *     yAxisInterval="10">
        * </igx-category-chart>
        * ```
        */
        get: function () {
            return this.i.qf;
        },
        set: function (v) {
            this.i.qf = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "yAxisIsLogarithmic", {
        /**
         * Gets or sets whether the Y-axis should use a logarithmic scale instead of a linear one.
         * Since log(-1) is imaginary and log(0) is undefined, it is recommended to enable this property only when the Y-axis minimum is greater than zero.
        *
        * Set `yAxisIsLogarithmic` to true if you want the y-axis to be a logarithmic scale.
        *
        * ```html
        * <igx-category-chart
        *     [dataSource]="data"
        *     yAxisIsLogarithmic="true">
        * </igx-category-chart>
        * ```
        */
        get: function () {
            return this.i.pu;
        },
        set: function (v) {
            this.i.pu = ensureBool(v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "yAxisLogarithmBase", {
        /**
         * Gets or sets the base value to use in the log function when mapping the position of data items along the Y-axis.
         * This property is effective only when YAxisIsLogarithmic is true.
        *
        * When `yAxisIsLogarithmic` is true, `yAxisLogarithmBase` sets the logarithm base of the y-axis scale.
        *
        * ```html
        * <igx-category-chart
        *     [dataSource]="data"
        *     yAxisIsLogarithmic="true"
        *     yAxisLogarithmBase="2">
        * </igx-category-chart>
        * ```
        */
        get: function () {
            return this.i.qm;
        },
        set: function (v) {
            this.i.qm = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "yAxisMinimumValue", {
        /**
         * Gets or sets the data value corresponding to the minimum value of the Y-axis.
        *
        * `yAxisMaximumValue` determines the minimum of the y-axis.
        *
        * ```html
        * <igx-category-chart
        *     [dataSource]="data"
        *     yAxisMinimumValue="42">
        * </igx-category-chart>
        * ```
        */
        get: function () {
            return this.i.qh;
        },
        set: function (v) {
            this.i.qh = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "yAxisMaximumValue", {
        /**
         * Gets or sets the data value corresponding to the maximum value of the Y-axis.
        *
        * `yAxisMaximumValue` determines the maximum of the y-axis.
        *
        * ```html
        * <igx-category-chart
        *     [dataSource]="data"
        *     yAxisMaximumValue="42">
        * </igx-category-chart>
        * ```
        */
        get: function () {
            return this.i.qg;
        },
        set: function (v) {
            this.i.qg = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "yAxisMinorInterval", {
        /**
         * Gets or sets the frequency of displayed minor lines along the Y-axis.
        *
        * `yAxisMinorInterval` determines how often to show a minor gridline along the y-axis.  Set this property to a number less than the y-axis interval.
        *
        * ```html
        * <igx-category-chart
        *     [dataSource]="data"
        *     yAxisMinorInterval="2"
        *     yAxisMinorStroke="green"
        *     yAxisMinorStrokeThickness="1">
        * </igx-category-chart>
        * ```
        */
        get: function () {
            return this.i.qi;
        },
        set: function (v) {
            this.i.qi = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "xAxisActualMinimum", {
        /**
         * Gets the current minimum value for the X-axis.
        */
        get: function () {
            return this.i.p8;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "xAxisActualMaximum", {
        /**
         * Gets the current maximum value for the X-axis.
        */
        get: function () {
            return this.i.p7;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "yAxisActualMinimum", {
        /**
         * Gets the current minimum value for the Y-axis.
        */
        get: function () {
            return this.i.qe;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "yAxisActualMaximum", {
        /**
         * Gets the current maximum value for the Y-axis.
        */
        get: function () {
            return this.i.qd;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "negativeBrushes", {
        /**
         * Gets or sets the palette used for coloring negative items of Waterfall chart type.
        *
        * `negativeBrushes` controls the brushes used by the chart for negative data, when the `chartType` is one that supports negative brushes.
        *
        * ```html
        * <igx-category-chart
        *     [dataSource]="data"
        *     chartType="waterfall"
        *     negativeBrushes="#ff0000, #cc0000, #aa0000">
        * </igx-category-chart>
        * ```
        */
        get: function () {
            return fromBrushCollection(this.i.o1);
        },
        set: function (v) {
            this.i.o1 = toBrushCollection(v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "negativeOutlines", {
        /**
         * Brushes to use for drawing negative elements, when using a chart type with contextual coloring, such as Waterfall.
        *
        * `negativeOutlines` controls the outlines used by the chart for negative data, when the `chartType` is one that supports negative brushes.
        *
        * ```html
        * <igx-category-chart
        *     [dataSource]="data"
        *     chartType="waterfall"
        *     negativeOutlines="#0000ff, #0000cc, #0000aa">
        * </igx-category-chart>
        * ```
        */
        get: function () {
            return fromBrushCollection(this.i.o2);
        },
        set: function (v) {
            this.i.o2 = toBrushCollection(v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "yAxisAbbreviateLargeNumbers", {
        /**
         * Gets or sets whether the large numbers on the Y-axis labels are abbreviated.
        *
        * To abbreviate on the y-axis using formats like "1K" or "1M," set `yAxisAbbreviateLargeNumbers` to true.
        *
        * ```html
        * <igx-category-chart
        *     [dataSource]="data"
        *     yAxisAbbreviateLargeNumbers="true">
        * </igx-category-chart>
        * ```
        */
        get: function () {
            return this.i.pt;
        },
        set: function (v) {
            this.i.pt = ensureBool(v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "isCategoryHighlightingEnabled", {
        /**
         * Gets or sets whether the category should be highlighted when hovered
        */
        get: function () {
            return this.i.pq;
        },
        set: function (v) {
            this.i.pq = ensureBool(v);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxCategoryChartComponent.prototype, "isItemHighlightingEnabled", {
        /**
         * Gets or sets whether the item should be highlighted when hovered
        */
        get: function () {
            return this.i.pr;
        },
        set: function (v) {
            this.i.pr = ensureBool(v);
        },
        enumerable: true,
        configurable: true
    });
    /**
 * Export serialized visual data.

*/
    IgxCategoryChartComponent.prototype.exportSerializedVisualData = function () {
        var iv = this.i.qo();
        return (iv);
    };
    IgxCategoryChartComponent.prototype.getCurrentXAxisActualMinimum = function () {
        var iv = this.i.p4();
        return (iv);
    };
    IgxCategoryChartComponent.prototype.getCurrentYAxisActualMinimum = function () {
        var iv = this.i.p6();
        return (iv);
    };
    IgxCategoryChartComponent.prototype.getCurrentXAxisActualMaximum = function () {
        var iv = this.i.p3();
        return (iv);
    };
    IgxCategoryChartComponent.prototype.getCurrentYAxisActualMaximum = function () {
        var iv = this.i.p5();
        return (iv);
    };
    tslib_1.__decorate([
        ViewChild("dynamicContent", { read: ViewContainerRef, static: true }),
        tslib_1.__metadata("design:type", ViewContainerRef)
    ], IgxCategoryChartComponent.prototype, "_dynamicContent", void 0);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxCategoryChartComponent.prototype, "height", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxCategoryChartComponent.prototype, "width", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Array),
        tslib_1.__metadata("design:paramtypes", [Array])
    ], IgxCategoryChartComponent.prototype, "dataSource", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxCategoryChartComponent.prototype, "transitionInDuration", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Function),
        tslib_1.__metadata("design:paramtypes", [Function])
    ], IgxCategoryChartComponent.prototype, "transitionInEasingFunction", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxCategoryChartComponent.prototype, "chartType", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxCategoryChartComponent.prototype, "markerCollisionAvoidance", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [Boolean])
    ], IgxCategoryChartComponent.prototype, "isTransitionInEnabled", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxCategoryChartComponent.prototype, "transitionInMode", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxCategoryChartComponent.prototype, "transitionInSpeedType", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxCategoryChartComponent.prototype, "xAxisInterval", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxCategoryChartComponent.prototype, "xAxisMinorInterval", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxCategoryChartComponent.prototype, "xAxisGap", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxCategoryChartComponent.prototype, "xAxisOverlap", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxCategoryChartComponent.prototype, "yAxisInterval", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [Boolean])
    ], IgxCategoryChartComponent.prototype, "yAxisIsLogarithmic", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxCategoryChartComponent.prototype, "yAxisLogarithmBase", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxCategoryChartComponent.prototype, "yAxisMinimumValue", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxCategoryChartComponent.prototype, "yAxisMaximumValue", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxCategoryChartComponent.prototype, "yAxisMinorInterval", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Array),
        tslib_1.__metadata("design:paramtypes", [Array])
    ], IgxCategoryChartComponent.prototype, "negativeBrushes", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Array),
        tslib_1.__metadata("design:paramtypes", [Array])
    ], IgxCategoryChartComponent.prototype, "negativeOutlines", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [Boolean])
    ], IgxCategoryChartComponent.prototype, "yAxisAbbreviateLargeNumbers", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [Boolean])
    ], IgxCategoryChartComponent.prototype, "isCategoryHighlightingEnabled", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [Boolean])
    ], IgxCategoryChartComponent.prototype, "isItemHighlightingEnabled", null);
    IgxCategoryChartComponent = tslib_1.__decorate([
        Component({
            selector: 'igx-category-chart',
            template: "<ng-container #dynamicContent></ng-container>",
            changeDetection: ChangeDetectionStrategy.OnPush,
            providers: IgxCategoryChartComponent_PROVIDERS,
            host: { 'class': 'ig-category-chart igx-category-chart' },
            styles: ["\n\t\t:host {\n\t\t\tdisplay: block;\n\t\t}\n\t"]
        }),
        tslib_1.__metadata("design:paramtypes", [Renderer2,
            ViewContainerRef,
            NgZone,
            Injector,
            ComponentFactoryResolver])
    ], IgxCategoryChartComponent);
    return IgxCategoryChartComponent;
}(IgxXYChartComponent));
export { IgxCategoryChartComponent };
