// igniteui-angular-charts/esm2015/lib/AssigningCategoryStyleEventArgsBase.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "igniteui-angular-core";
/**
 * @hidden
 */
export class AssigningCategoryStyleEventArgsBase extends Base {
    constructor() {
        super(...arguments);
        this.m = 0;
        this.l = 0;
        this.h = new Date();
        this.g = new Date();
        this.a = null;
        this.n = null;
        this.o = null;
        this.j = 0;
        this.b = null;
        this.i = 0;
        this.k = 0;
        this.d = false;
        this.c = false;
        this.e = false;
        this.f = false;
    }
}
AssigningCategoryStyleEventArgsBase.$t = markType(AssigningCategoryStyleEventArgsBase, 'AssigningCategoryStyleEventArgsBase');
