// igniteui-angular-charts/esm2015/lib/FastStochasticOscillatorIndicator.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { StrategyBasedIndicator } from "./StrategyBasedIndicator";
import { FastStochasticOscillatorIndicatorStrategy } from "./FastStochasticOscillatorIndicatorStrategy";
import { typeGetValue, markType } from "igniteui-angular-core";
import { Brush } from "igniteui-angular-core";
/**
 * @hidden
 */
export class FastStochasticOscillatorIndicator extends StrategyBasedIndicator {
    constructor() {
        super();
        this.trendLineBrush = ((() => {
            let $ret = new Brush();
            $ret.fill = "#565656";
            return $ret;
        })());
        this.trendLineType = 10;
        this.trendLinePeriod = 3;
    }
    get_qd() {
        return new FastStochasticOscillatorIndicatorStrategy();
    }
    get qd() {
        return this.get_qd();
    }
    get_qq() {
        return FastStochasticOscillatorIndicator.$;
    }
    get qq() {
        return this.get_qq();
    }
    get qy() {
        return typeGetValue(this.c(FastStochasticOscillatorIndicator.qz));
    }
    set qy(a) {
        this.h(FastStochasticOscillatorIndicator.qz, a);
    }
    qj() {
        return this.qy;
    }
    qs(a) {
        this.qy = a;
    }
}
FastStochasticOscillatorIndicator.$t = markType(FastStochasticOscillatorIndicator, 'FastStochasticOscillatorIndicator', StrategyBasedIndicator.$);
FastStochasticOscillatorIndicator.qz = StrategyBasedIndicator.qv(14, FastStochasticOscillatorIndicator.$);
