// igniteui-angular-charts/esm2015/lib/igx-relative-strength-index-indicator-component.js
var IgxRelativeStrengthIndexIndicatorComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxStrategyBasedIndicatorComponent } from "./igx-strategy-based-indicator-component";
import { IgxFinancialIndicatorComponent } from "./igx-financial-indicator-component";
import { IgxFinancialSeriesComponent } from "./igx-financial-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { RelativeStrengthIndexIndicator } from "./RelativeStrengthIndexIndicator";
/**
 * Represents a IgxDataChartComponent Relative Strength Index indicator series.
 * Default required members: Close
*
* The `IgxRelativeStrengthIndexIndicatorComponent` class represents Relative Strength Index indicator series for the IgxDataChartComponent .
*
* ```html
*  <igx-data-chart
*     [dataSource]="data">
* 	<igx-category-x-axis
* 		label="label"
* 		#xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
* 		#yAxis>
*     </igx-numeric-y-axis>
* 	<igx-relative-strength-index-indicator
*         #series
*         [xAxis]="xAxis"
*         [yAxis]="yAxis"
*         highMemberPath="high"
* 		lowMemberPath="low"
* 		volumeMemberPath="volume">
*     </igx-relative-strength-index-indicator>
* </igx-data-chart>
* ```
*
* ```ts
* let series = new IgxRelativeStrengthIndexIndicatorComponent();
* series.xAxis = this.xAxis;
* series.yAxis = this.yAxis;
* series.openMemberPath = "open";
* series.highMemberPath = "high";
* series.lowMemberPath = "low";
* series.closeMemberPath = "close";
* this.chart.series.add(series);
* ```
*/
export const IgxRelativeStrengthIndexIndicatorComponent_PROVIDERS = [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(() => IgxRelativeStrengthIndexIndicatorComponent) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(() => IgxRelativeStrengthIndexIndicatorComponent) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(() => IgxRelativeStrengthIndexIndicatorComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxRelativeStrengthIndexIndicatorComponent) }];
let IgxRelativeStrengthIndexIndicatorComponent = IgxRelativeStrengthIndexIndicatorComponent_1 = class IgxRelativeStrengthIndexIndicatorComponent extends IgxStrategyBasedIndicatorComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new RelativeStrengthIndexIndicator();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Gets or sets the moving average period for the current RelativeStrengthIndexIndicator object.
     * The typical, and initial, value for RSI periods is 14.
    *
    * You can use the `period` property for the current RelativeStrengthIndexIndicator object.
    *
    * ```html
    *  <igx-data-chart
    *     [dataSource]="data">
    * 	<igx-category-x-axis
    * 		label="label"
    * 		#xAxis>
    *     </igx-category-x-axis>
    *     <igx-numeric-y-axis
    * 		#yAxis>
    *     </igx-numeric-y-axis>
    * 	<igx-relative-strength-index-indicator
    *         #series
    *         [xAxis]="xAxis"
    *         [yAxis]="yAxis"
    *         highMemberPath="high"
    * 		lowMemberPath="low"
    * 		volumeMemberPath="volume"
    * 		period=30>
    *     </igx-relative-strength-index-indicator>
    * </igx-data-chart>
    * ```
    *
    * ```ts
    * this.series.period = 30;
    * ```
    */
    get period() {
        return this.i.qy;
    }
    set period(v) {
        this.i.qy = +v;
    }
};
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxRelativeStrengthIndexIndicatorComponent.prototype, "period", null);
IgxRelativeStrengthIndexIndicatorComponent = IgxRelativeStrengthIndexIndicatorComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-relative-strength-index-indicator',
        template: ``,
        providers: [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(() => IgxRelativeStrengthIndexIndicatorComponent_1) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(() => IgxRelativeStrengthIndexIndicatorComponent_1) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(() => IgxRelativeStrengthIndexIndicatorComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxRelativeStrengthIndexIndicatorComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxRelativeStrengthIndexIndicatorComponent);
export { IgxRelativeStrengthIndexIndicatorComponent };
