// igniteui-angular-charts/esm2015/lib/igx-spline-series-component.js
var IgxSplineSeriesComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxSplineSeriesBaseComponent } from "./igx-spline-series-base-component";
import { IgxHorizontalAnchoredCategorySeriesComponent } from "./igx-horizontal-anchored-category-series-component";
import { IgxAnchoredCategorySeriesComponent } from "./igx-anchored-category-series-component";
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { SplineSeries } from "./SplineSeries";
/**
 * Represents a IgxDataChartComponent spline series.
*
* ```html
* <igx-data-chart
*     [dataSource]="data">
*     <igx-category-x-axis
*         label="label"
*         #xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
*         #yAxis>
*     </igx-numeric-y-axis>
*     <igx-spline-series
*         [xAxis]="xAxis"
*         [yAxis]="yAxis"
*         valueMemberPath="value">
*     </igx-spline-series>
* </igx-data-chart>
* ```
*/
export const IgxSplineSeriesComponent_PROVIDERS = [{ provide: IgxSplineSeriesBaseComponent, useExisting: forwardRef(() => IgxSplineSeriesComponent) }, { provide: IgxHorizontalAnchoredCategorySeriesComponent, useExisting: forwardRef(() => IgxSplineSeriesComponent) }, { provide: IgxAnchoredCategorySeriesComponent, useExisting: forwardRef(() => IgxSplineSeriesComponent) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(() => IgxSplineSeriesComponent) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxSplineSeriesComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxSplineSeriesComponent) }];
let IgxSplineSeriesComponent = IgxSplineSeriesComponent_1 = class IgxSplineSeriesComponent extends IgxSplineSeriesBaseComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new SplineSeries();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
};
IgxSplineSeriesComponent = IgxSplineSeriesComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-spline-series',
        template: ``,
        providers: [{ provide: IgxSplineSeriesBaseComponent, useExisting: forwardRef(() => IgxSplineSeriesComponent_1) }, { provide: IgxHorizontalAnchoredCategorySeriesComponent, useExisting: forwardRef(() => IgxSplineSeriesComponent_1) }, { provide: IgxAnchoredCategorySeriesComponent, useExisting: forwardRef(() => IgxSplineSeriesComponent_1) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(() => IgxSplineSeriesComponent_1) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxSplineSeriesComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxSplineSeriesComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxSplineSeriesComponent);
export { IgxSplineSeriesComponent };
