// igniteui-angular-charts/esm2015/lib/igx-point-series-component.js
var IgxPointSeriesComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxHorizontalAnchoredCategorySeriesComponent } from "./igx-horizontal-anchored-category-series-component";
import { IgxAnchoredCategorySeriesComponent } from "./igx-anchored-category-series-component";
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { PointSeries } from "./PointSeries";
/**
 * Represents a IgxDataChartComponent point series.
*
* You can use the `IgxPointSeriesComponent` plot a value along a category or date.
*
* ```html
* <igx-data-chart
*     [dataSource]="data" >
*     <igx-category-x-axis
* 		label="label"
* 		#xAxis>
*     </igx-category-x-axis
* 		#yAxis>
*     <igx-numeric-y-axis>
*     </igx-numeric-y-axis>
*     <igx-point-series
* 		[xAxis]="xAxis"
* 		[yAxis]="yAxis"
* 		valueMemberPath="coal">
*     </igx-point-series>
* </igx-data-chart>
* ```
*
* ```ts
* let series: IgxPointSeriesComponent = new IgxPointSeriesComponent();
* series.xAxis = this.xAxis;
* series.yAxis = this.yAxis;
* series.valueMemberPath = "coal";
* this.chart.series.add(series);
* ```
*/
export const IgxPointSeriesComponent_PROVIDERS = [{ provide: IgxHorizontalAnchoredCategorySeriesComponent, useExisting: forwardRef(() => IgxPointSeriesComponent) }, { provide: IgxAnchoredCategorySeriesComponent, useExisting: forwardRef(() => IgxPointSeriesComponent) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(() => IgxPointSeriesComponent) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxPointSeriesComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxPointSeriesComponent) }];
let IgxPointSeriesComponent = IgxPointSeriesComponent_1 = class IgxPointSeriesComponent extends IgxHorizontalAnchoredCategorySeriesComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new PointSeries();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Overridden by derived series classes to indicate when markerless display is preferred or not.
    */
    get isMarkerlessDisplayPreferred() {
        return this.i.cy;
    }
};
IgxPointSeriesComponent = IgxPointSeriesComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-point-series',
        template: ``,
        providers: [{ provide: IgxHorizontalAnchoredCategorySeriesComponent, useExisting: forwardRef(() => IgxPointSeriesComponent_1) }, { provide: IgxAnchoredCategorySeriesComponent, useExisting: forwardRef(() => IgxPointSeriesComponent_1) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(() => IgxPointSeriesComponent_1) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxPointSeriesComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxPointSeriesComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxPointSeriesComponent);
export { IgxPointSeriesComponent };
