// igniteui-angular-charts/esm2015/lib/igx-range-category-series-component.js
import * as tslib_1 from "tslib";
import { forwardRef, Input } from '@angular/core';
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
/**
 * Represents the base class for all IgxDataChartComponent ranged category/value series.
*
* The `IgxRangeCategorySeriesComponent` represents the base class for all IgxDataChartComponent ranged category/value series.
*
* ```html
* <igx-data-chart  #chart
*     [dataSource]="data">
*     <igx-category-x-axis
*     label="label"
*     #xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
*     minimumValue="0"
*     #yAxis>
*     </igx-numeric-y-axis>
*     <igx-range-column-series
*     [xAxis]="xAxis"
*     [yAxis]="yAxis"
*     highMemberPath="high"
*     lowMemberPath="low">
* </igx-range-column-series>
* </igx-data-chart>
* ```
*
* ```ts
* let series = new IgxRangeColumnSeriesComponent();
* series.xAxis = this.xAxis;
* series.yAxis = this.yAxis;
* series.highMemberPath = "high";
* series.lowMemberPath = "low";
* this.chart.series.add(series);
* ```
*/
export const IgxRangeCategorySeriesComponent_PROVIDERS = [{ provide: IgxCategorySeriesComponent, useExisting: forwardRef(() => IgxRangeCategorySeriesComponent) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxRangeCategorySeriesComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxRangeCategorySeriesComponent) }];
export class IgxRangeCategorySeriesComponent extends IgxCategorySeriesComponent {
    constructor() {
        super();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Gets or sets the value mapping property for the current series object.
    *
    * The `lowMemberPath` is used for low mapping property of the current series object.
    *
    * ```html
    * <igx-data-chart  #chart
    *     [dataSource]="data">
    *     <igx-category-x-axis
    *     label="label"
    *     #xAxis>
    *     </igx-category-x-axis>
    *     <igx-numeric-y-axis
    *     minimumValue="0"
    *     #yAxis>
    *     </igx-numeric-y-axis>
    *     <igx-range-column-series
    *     [xAxis]="xAxis"
    *     [yAxis]="yAxis"
    *     highMemberPath="high"
    *     lowMemberPath="low">
    * </igx-range-column-series>
    * </igx-data-chart>
    * ```
    *
    * ```ts
    * let series = new IgxRangeColumnSeriesComponent();
    * series.xAxis = this.xAxis;
    * series.yAxis = this.yAxis;
    * series.highMemberPath = "high";
    * series.lowMemberPath = "low";
    * this.chart.series.add(series);
    * ```
    */
    get lowMemberPath() {
        return this.i.pa;
    }
    set lowMemberPath(v) {
        this.i.pa = v;
    }
    /**
     * Gets or sets the value mapping property for the current series object.
    *
    * Use `highMemberPath` propert for high mapping of the current series object.
    *
    * ```html
    * <igx-data-chart  #chart
    *     [dataSource]="data">
    *     <igx-category-x-axis
    *     label="label"
    *     #xAxis>
    *     </igx-category-x-axis>
    *     <igx-numeric-y-axis
    *     minimumValue="0"
    *     #yAxis>
    *     </igx-numeric-y-axis>
    *     <igx-range-column-series
    *     [xAxis]="xAxis"
    *     [yAxis]="yAxis"
    *     highMemberPath="high"
    *     lowMemberPath="low">
    * </igx-range-column-series>
    * </igx-data-chart>
    * ```
    *
    * ```ts
    * series.highMemberPath = "high";
    * ```
    */
    get highMemberPath() {
        return this.i.o7;
    }
    set highMemberPath(v) {
        this.i.o7 = v;
    }
    getItemValue(item, memberPathName) {
        let iv = this.i.fs(item, memberPathName);
        return (iv);
    }
    /**
     * Scrolls the series to display the item for the specified data item.
     * The series is scrolled by the minimum amount required to place the specified data item within
     * the central 80% of the visible axis.
    
    * @param item  * The data item (item) to scroll to.
    *
    * `scrollIntoView` method Scrolls the series to display the item for the specified data item.
    */
    scrollIntoView(item) {
        let iv = this.i.c9(item);
        return (iv);
    }
}
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxRangeCategorySeriesComponent.prototype, "lowMemberPath", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxRangeCategorySeriesComponent.prototype, "highMemberPath", null);
