// igniteui-angular-charts/esm2015/lib/igx-radial-area-series-component.js
var IgxRadialAreaSeriesComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { UnknownValuePlotting, UnknownValuePlotting_$type } from "igniteui-angular-core";
import { IgxAnchoredRadialSeriesComponent } from "./igx-anchored-radial-series-component";
import { IgxRadialBaseComponent } from "./igx-radial-base-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { RadialAreaSeries } from "./RadialAreaSeries";
import { ensureEnum } from "igniteui-angular-core";
/**
 * Represents a IgxDataChartComponent radial area series.
*/
export const IgxRadialAreaSeriesComponent_PROVIDERS = [{ provide: IgxAnchoredRadialSeriesComponent, useExisting: forwardRef(() => IgxRadialAreaSeriesComponent) }, { provide: IgxRadialBaseComponent, useExisting: forwardRef(() => IgxRadialAreaSeriesComponent) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxRadialAreaSeriesComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxRadialAreaSeriesComponent) }];
let IgxRadialAreaSeriesComponent = IgxRadialAreaSeriesComponent_1 = class IgxRadialAreaSeriesComponent extends IgxAnchoredRadialSeriesComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new RadialAreaSeries();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Determines how unknown values will be plotted on the chart.
     * Null and Double.NaN are two examples of unknown values.
    */
    get unknownValuePlotting() {
        return this.i.ov;
    }
    set unknownValuePlotting(v) {
        this.i.ov = ensureEnum(UnknownValuePlotting_$type, v);
    }
};
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxRadialAreaSeriesComponent.prototype, "unknownValuePlotting", null);
IgxRadialAreaSeriesComponent = IgxRadialAreaSeriesComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-radial-area-series',
        template: ``,
        providers: [{ provide: IgxAnchoredRadialSeriesComponent, useExisting: forwardRef(() => IgxRadialAreaSeriesComponent_1) }, { provide: IgxRadialBaseComponent, useExisting: forwardRef(() => IgxRadialAreaSeriesComponent_1) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxRadialAreaSeriesComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxRadialAreaSeriesComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxRadialAreaSeriesComponent);
export { IgxRadialAreaSeriesComponent };
