// igniteui-angular-charts/esm2015/lib/igx-financial-legend-component.js
var IgxFinancialLegendComponent_1;
import * as tslib_1 from "tslib";
import { Component, Renderer2, ElementRef, forwardRef, NgZone, ChangeDetectionStrategy } from '@angular/core';
import { IgxLegendBaseComponent } from './igx-legend-base-component';
import { Legend } from "./Legend";
import { AngularRenderer } from "igniteui-angular-core";
import { DataChartStylingDefaults } from './DataChartStylingDefaults';
export const IgxFinancialLegendComponent_PROVIDERS = [{ provide: IgxLegendBaseComponent, useExisting: forwardRef(() => IgxFinancialLegendComponent) }];
/**
 * Represents a legend that displays information about each series, most often used in financial charting.
*/
let IgxFinancialLegendComponent = IgxFinancialLegendComponent_1 = class IgxFinancialLegendComponent extends IgxLegendBaseComponent {
    constructor(renderer, elRef, ngZone) {
        super();
        this.renderer = renderer;
        this.elRef = elRef;
        this.ngZone = ngZone;
        this._zoneRunner = (act) => this.ngZone.run(act);
        this.container = renderer.createElement("div");
        renderer.appendChild(elRef.nativeElement, this.container);
        var root;
        root = this.container;
        if (this.container != null) {
            root = this.container;
        }
        var ren = new AngularRenderer(root, this.renderer, window.document, this.ngZone, true, DataChartStylingDefaults);
        this._wrapper = ren;
        this.i.provideContainer(ren);
    }
    ngOnDestroy() {
        this._wrapper.destroy();
        this.i.provideContainer(null);
    }
    createImplementation() {
        return new Legend();
    }
    get i() {
        return this._implementation;
    }
    /**
 * Gets if the legend is a financial legend.
*/
    get isFinancial() {
        return this.i.isFinancial;
    }
};
IgxFinancialLegendComponent = IgxFinancialLegendComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-financial-legend',
        template: ``,
        providers: [{ provide: IgxLegendBaseComponent, useExisting: forwardRef(() => IgxFinancialLegendComponent_1) }],
        host: { 'class': 'ig-financial-legend igx-financial-legend' },
        changeDetection: ChangeDetectionStrategy.OnPush,
        styles: [`:host {
    display: block;
}
.ig-legend {
    border: none;
    border: var(--legend-border, none);
    overflow: auto;
    overflow: var(--legend-item-overflow, auto);
}
.ig-chart-legend-item-text {
    font: 12px Verdana, Arial, sans-serif;
    font: var(--legend-item-font, 12px Verdana, Arial, sans-serif);
    vertical-align: middle;
    vertical-align: var(--legend-item-vertical-align, middle);
}
.ig-chart-legend-items-list
{
    padding: 5px;
    padding: var(--legend-item-padding, 5px);
    margin: 0px;
    margin: var(--legend-item-margin, 0px);
    background-color: transparent;
    background-color: var(--legend-list-background-color, transparent);
}
.ig-chart-legend-item-badge, .ui-chart-legend-item-badge canvas
{
    vertical-align: middle;
    vertical-align: var(--legend-item-badge-vertical-align, middle);
}
.ig-chart-legend-item
{
    background-color: transparent;
    background-color: var(--legend-item-background-color, transparent);
}`]
    }),
    tslib_1.__metadata("design:paramtypes", [Renderer2,
        ElementRef,
        NgZone])
], IgxFinancialLegendComponent);
export { IgxFinancialLegendComponent };
