// igniteui-angular-charts/esm2015/lib/igx-category-tool-tip-layer-component.js
var IgxCategoryToolTipLayerComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxAxisComponent } from "./igx-axis-component";
import { IgxAnnotationLayerComponent } from "./igx-annotation-layer-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { CategoryToolTipLayer } from "./CategoryToolTipLayer";
import { ensureBool } from "igniteui-angular-core";
/**
 * Represents an annotation layer that displays grouped tooltips for series that use a category axis.
*/
export const IgxCategoryToolTipLayerComponent_PROVIDERS = [{ provide: IgxAnnotationLayerComponent, useExisting: forwardRef(() => IgxCategoryToolTipLayerComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxCategoryToolTipLayerComponent) }];
let IgxCategoryToolTipLayerComponent = IgxCategoryToolTipLayerComponent_1 = class IgxCategoryToolTipLayerComponent extends IgxAnnotationLayerComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new CategoryToolTipLayer();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Gets whether the default tooltip behaviors for the chart are disabled if this layer is present.
    */
    get isDefaultTooltipBehaviorDisabled() {
        return this.i.cn;
    }
    /**
     * Gets or sets the Axis to target this annotation to. If null, this annotation layer will not render content.
    */
    get targetAxis() {
        if (this.i.m5 == null) {
            return null;
        }
        if (!this.i.m5.externalObject) {
            let e = IgxAxisComponent._createFromInternal(this.i.m5);
            if (e) {
                e._implementation = this.i.m5;
            }
            this.i.m5.externalObject = e;
        }
        return this.i.m5.externalObject;
    }
    set targetAxis(v) {
        if (v != null && this._stylingContainer && v._styling)
            v._styling(this._stylingContainer, this, this);
        v == null ? this.i.m5 = null : this.i.m5 = v.i;
    }
    /**
     * Gets or sets whether to use value interpolation when drawing the tooltips.
    */
    get useInterpolation() {
        return this.i.m9;
    }
    set useInterpolation(v) {
        this.i.m9 = ensureBool(v);
    }
    findByName(name) {
        var baseResult = super.findByName(name);
        if (baseResult) {
            return baseResult;
        }
        if (this.targetAxis && this.targetAxis.name && this.targetAxis.name == name) {
            return this.targetAxis;
        }
        return null;
    }
    _styling(container, component, parent) {
        super._styling(container, component, parent);
        this._inStyling = true;
        if (this.targetAxis && this.targetAxis._styling) {
            this.targetAxis._styling(container, component, this);
        }
        this._inStyling = false;
    }
    /**
     * Hides any tooltips presented by the layer, if any.
    
    */
    hideToolTips() {
        this.i.ib();
    }
};
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", IgxAxisComponent),
    tslib_1.__metadata("design:paramtypes", [IgxAxisComponent])
], IgxCategoryToolTipLayerComponent.prototype, "targetAxis", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Boolean),
    tslib_1.__metadata("design:paramtypes", [Boolean])
], IgxCategoryToolTipLayerComponent.prototype, "useInterpolation", null);
IgxCategoryToolTipLayerComponent = IgxCategoryToolTipLayerComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-category-tool-tip-layer',
        template: ``,
        providers: [{ provide: IgxAnnotationLayerComponent, useExisting: forwardRef(() => IgxCategoryToolTipLayerComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxCategoryToolTipLayerComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxCategoryToolTipLayerComponent);
export { IgxCategoryToolTipLayerComponent };
