// igniteui-angular-charts/esm2015/lib/igx-scatter-series-component.js
var IgxScatterSeriesComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxScatterBaseComponent } from "./igx-scatter-base-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { ScatterSeries } from "./ScatterSeries";
/**
 * Represents a IgxDataChartComponent scatter series
*/
export const IgxScatterSeriesComponent_PROVIDERS = [{ provide: IgxScatterBaseComponent, useExisting: forwardRef(() => IgxScatterSeriesComponent) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxScatterSeriesComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxScatterSeriesComponent) }];
let IgxScatterSeriesComponent = IgxScatterSeriesComponent_1 = class IgxScatterSeriesComponent extends IgxScatterBaseComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new ScatterSeries();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Overridden by derived series classes to indicate when markerless display is preferred or not.
    */
    get isMarkerlessDisplayPreferred() {
        return this.i.cy;
    }
};
IgxScatterSeriesComponent = IgxScatterSeriesComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-scatter-series',
        template: ``,
        providers: [{ provide: IgxScatterBaseComponent, useExisting: forwardRef(() => IgxScatterSeriesComponent_1) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxScatterSeriesComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxScatterSeriesComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxScatterSeriesComponent);
export { IgxScatterSeriesComponent };
