// igniteui-angular-charts/esm2015/lib/igx-custom-palette-brush-scale-component.js
var IgxCustomPaletteBrushScaleComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { BrushSelectionMode, BrushSelectionMode_$type } from "./BrushSelectionMode";
import { IgxBrushScaleComponent } from "./igx-brush-scale-component";
import { CustomPaletteBrushScale } from "./CustomPaletteBrushScale";
import { ensureEnum, brushToString } from "igniteui-angular-core";
/**
 * Represents a brush scale that uses index-based brush selection mode.
*/
export const IgxCustomPaletteBrushScaleComponent_PROVIDERS = [{ provide: IgxBrushScaleComponent, useExisting: forwardRef(() => IgxCustomPaletteBrushScaleComponent) }];
let IgxCustomPaletteBrushScaleComponent = IgxCustomPaletteBrushScaleComponent_1 = class IgxCustomPaletteBrushScaleComponent extends IgxBrushScaleComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new CustomPaletteBrushScale();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Gets or sets the brush selection mode.
    */
    get brushSelectionMode() {
        return this.i.brushSelectionMode;
    }
    set brushSelectionMode(v) {
        this.i.brushSelectionMode = ensureEnum(BrushSelectionMode_$type, v);
    }
    /**
     * Checks if this scale is ready for useage with a series
    */
    get isReady() {
        return this.i.isReady;
    }
    getBrush1(index, total) {
        let iv = this.i.getBrush1(index, total);
        return brushToString(iv);
    }
};
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxCustomPaletteBrushScaleComponent.prototype, "brushSelectionMode", null);
IgxCustomPaletteBrushScaleComponent = IgxCustomPaletteBrushScaleComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-custom-palette-brush-scale',
        template: ``,
        providers: [{ provide: IgxBrushScaleComponent, useExisting: forwardRef(() => IgxCustomPaletteBrushScaleComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxCustomPaletteBrushScaleComponent);
export { IgxCustomPaletteBrushScaleComponent };
