// igniteui-angular-charts/esm2015/lib/igx-polar-line-series-component.js
var IgxPolarLineSeriesComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { UnknownValuePlotting, UnknownValuePlotting_$type } from "igniteui-angular-core";
import { IgxPolarLineSeriesBaseComponent } from "./igx-polar-line-series-base-component";
import { IgxPolarBaseComponent } from "./igx-polar-base-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { PolarLineSeries } from "./PolarLineSeries";
import { ensureEnum } from "igniteui-angular-core";
/**
 * Represents a IgxDataChartComponent polar line series.
*/
export const IgxPolarLineSeriesComponent_PROVIDERS = [{ provide: IgxPolarLineSeriesBaseComponent, useExisting: forwardRef(() => IgxPolarLineSeriesComponent) }, { provide: IgxPolarBaseComponent, useExisting: forwardRef(() => IgxPolarLineSeriesComponent) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxPolarLineSeriesComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxPolarLineSeriesComponent) }];
let IgxPolarLineSeriesComponent = IgxPolarLineSeriesComponent_1 = class IgxPolarLineSeriesComponent extends IgxPolarLineSeriesBaseComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new PolarLineSeries();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Determines how unknown values will be plotted on the chart.
     * Null and Double.NaN are two examples of unknown values.
    */
    get unknownValuePlotting() {
        return this.i.pi;
    }
    set unknownValuePlotting(v) {
        this.i.pi = ensureEnum(UnknownValuePlotting_$type, v);
    }
};
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxPolarLineSeriesComponent.prototype, "unknownValuePlotting", null);
IgxPolarLineSeriesComponent = IgxPolarLineSeriesComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-polar-line-series',
        template: ``,
        providers: [{ provide: IgxPolarLineSeriesBaseComponent, useExisting: forwardRef(() => IgxPolarLineSeriesComponent_1) }, { provide: IgxPolarBaseComponent, useExisting: forwardRef(() => IgxPolarLineSeriesComponent_1) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxPolarLineSeriesComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxPolarLineSeriesComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxPolarLineSeriesComponent);
export { IgxPolarLineSeriesComponent };
