// igniteui-angular-charts/esm2015/lib/igx-linear-contour-value-resolver-component.js
var IgxLinearContourValueResolverComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxContourValueResolverComponent } from "./igx-contour-value-resolver-component";
import { LinearContourValueResolver } from "./LinearContourValueResolver";
/**
 * ContourValueResolver class which resolves contour lines to be distributed at linear intervals from the data minimum to the data maximum.
*/
export const IgxLinearContourValueResolverComponent_PROVIDERS = [{ provide: IgxContourValueResolverComponent, useExisting: forwardRef(() => IgxLinearContourValueResolverComponent) }];
let IgxLinearContourValueResolverComponent = IgxLinearContourValueResolverComponent_1 = class IgxLinearContourValueResolverComponent extends IgxContourValueResolverComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new LinearContourValueResolver();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Gets or sets how many contour values to return.
    */
    get valueCount() {
        return this.i.valueCount;
    }
    set valueCount(v) {
        this.i.valueCount = +v;
    }
};
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxLinearContourValueResolverComponent.prototype, "valueCount", null);
IgxLinearContourValueResolverComponent = IgxLinearContourValueResolverComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-linear-contour-value-resolver',
        template: ``,
        providers: [{ provide: IgxContourValueResolverComponent, useExisting: forwardRef(() => IgxLinearContourValueResolverComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxLinearContourValueResolverComponent);
export { IgxLinearContourValueResolverComponent };
