// igniteui-angular-charts/esm2015/lib/igx-radial-line-series-component.js
var IgxRadialLineSeriesComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { UnknownValuePlotting, UnknownValuePlotting_$type } from "igniteui-angular-core";
import { IgxAnchoredRadialSeriesComponent } from "./igx-anchored-radial-series-component";
import { IgxRadialBaseComponent } from "./igx-radial-base-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { RadialLineSeries } from "./RadialLineSeries";
import { ensureEnum } from "igniteui-angular-core";
/**
 * Represents a IgxDataChartComponent radial line series.
*/
export const IgxRadialLineSeriesComponent_PROVIDERS = [{ provide: IgxAnchoredRadialSeriesComponent, useExisting: forwardRef(() => IgxRadialLineSeriesComponent) }, { provide: IgxRadialBaseComponent, useExisting: forwardRef(() => IgxRadialLineSeriesComponent) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxRadialLineSeriesComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxRadialLineSeriesComponent) }];
let IgxRadialLineSeriesComponent = IgxRadialLineSeriesComponent_1 = class IgxRadialLineSeriesComponent extends IgxAnchoredRadialSeriesComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new RadialLineSeries();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Determines how unknown values will be plotted on the chart.
     * Null and Double.NaN are two examples of unknown values.
    */
    get unknownValuePlotting() {
        return this.i.ov;
    }
    set unknownValuePlotting(v) {
        this.i.ov = ensureEnum(UnknownValuePlotting_$type, v);
    }
    /**
     * When overridden in a derived class, is invoked whenever application code or internal processes
     * call ApplyTemplate.
    
    */
    onApplyTemplate() {
        this.i.ac();
    }
};
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxRadialLineSeriesComponent.prototype, "unknownValuePlotting", null);
IgxRadialLineSeriesComponent = IgxRadialLineSeriesComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-radial-line-series',
        template: ``,
        providers: [{ provide: IgxAnchoredRadialSeriesComponent, useExisting: forwardRef(() => IgxRadialLineSeriesComponent_1) }, { provide: IgxRadialBaseComponent, useExisting: forwardRef(() => IgxRadialLineSeriesComponent_1) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxRadialLineSeriesComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxRadialLineSeriesComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxRadialLineSeriesComponent);
export { IgxRadialLineSeriesComponent };
