// igniteui-angular-charts/esm2015/lib/igx-step-line-series-component.js
var IgxStepLineSeriesComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxHorizontalAnchoredCategorySeriesComponent } from "./igx-horizontal-anchored-category-series-component";
import { IgxAnchoredCategorySeriesComponent } from "./igx-anchored-category-series-component";
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { StepLineSeries } from "./StepLineSeries";
/**
 * Represents a IgxDataChartComponent step line series.
*
* ```html
* <igx-data-chart
*     [dataSource]="data">
*     <igx-category-x-axis
*         label="label"
*         #xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
*         #yAxis>
*     </igx-numeric-y-axis>
*     <igx-step-line-series
*         [xAxis]="xAxis"
*         [yAxis]="yAxis"
*         valueMemberPath="value">
*     </igx-step-line-series>
* </igx-data-chart>
* ```
*
* ```ts
*    let series = new IgxStepLineSeriesComponent();
*    series.xAxis = this.xAxis;
*    series.yAxis = this.yAxis;
*    series.valueMemberPath="value";
*    this.chart.series.add(series);
* ```
*/
export const IgxStepLineSeriesComponent_PROVIDERS = [{ provide: IgxHorizontalAnchoredCategorySeriesComponent, useExisting: forwardRef(() => IgxStepLineSeriesComponent) }, { provide: IgxAnchoredCategorySeriesComponent, useExisting: forwardRef(() => IgxStepLineSeriesComponent) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(() => IgxStepLineSeriesComponent) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxStepLineSeriesComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxStepLineSeriesComponent) }];
let IgxStepLineSeriesComponent = IgxStepLineSeriesComponent_1 = class IgxStepLineSeriesComponent extends IgxHorizontalAnchoredCategorySeriesComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new StepLineSeries();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Gets whether the current series shows an area or line shape.
    */
    get isAreaOrLine() {
        return this.i.ck;
    }
};
IgxStepLineSeriesComponent = IgxStepLineSeriesComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-step-line-series',
        template: ``,
        providers: [{ provide: IgxHorizontalAnchoredCategorySeriesComponent, useExisting: forwardRef(() => IgxStepLineSeriesComponent_1) }, { provide: IgxAnchoredCategorySeriesComponent, useExisting: forwardRef(() => IgxStepLineSeriesComponent_1) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(() => IgxStepLineSeriesComponent_1) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxStepLineSeriesComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxStepLineSeriesComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxStepLineSeriesComponent);
export { IgxStepLineSeriesComponent };
