// igniteui-angular-charts/esm2015/lib/igx-xy-chart-component.js
import * as tslib_1 from "tslib";
import { forwardRef, Input } from '@angular/core';
import { HorizontalAlignment, HorizontalAlignment_$type } from "igniteui-angular-core";
import { VerticalAlignment, VerticalAlignment_$type } from "igniteui-angular-core";
import { Visibility, Visibility_$type } from "igniteui-angular-core";
import { AxisLabelsLocation, AxisLabelsLocation_$type } from "./AxisLabelsLocation";
import { IgxDomainChartComponent } from "./igx-domain-chart-component";
import { brushToString, stringToBrush, ensureBool, ensureEnum, arrayFindByName } from "igniteui-angular-core";
/**
 * Represents a base class for domain charts with X/Y axes
*/
export const IgxXYChartComponent_PROVIDERS = [{ provide: IgxDomainChartComponent, useExisting: forwardRef(() => IgxXYChartComponent) }];
export class IgxXYChartComponent extends IgxDomainChartComponent {
    constructor() {
        super();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Gets or sets function which takes an context object and returns a formatted label for the X-axis.
    *
    * To customize the text of the x-axis label, set `xAxisFormatLabel` to a single-parameter function.  The argument passed to that function will be your data item.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     [xAxisFormatLabel]="formatX"
    *     xAxisExtent="250">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * formatX(item: any): string {
    *     return "[ " + item.label + "! ]";
    * }
    * ```
    */
    get xAxisFormatLabel() {
        return this.i.ii;
    }
    set xAxisFormatLabel(v) {
        this.i.ii = v;
    }
    /**
     * Gets or sets function which takes a context object and returns a formatted label for the Y-axis.
    *
    * To customize the text of the y-axis label, set `yAxisFormatLabel` to a single-parameter function.  The argument passed to that function will be the numeric value corresponding to a position on the y-axis.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     [yAxisFormatLabel]="formatY">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * formatY(value: any): string {
    *     return "[ " + value+ "! ]";
    * }
    * ```
    */
    get yAxisFormatLabel() {
        return this.i.ij;
    }
    set yAxisFormatLabel(v) {
        this.i.ij = v;
    }
    /**
     * Gets or sets the left margin of labels on the X-axis
    *
    * Use the `xAxisLabelLeftMargin` property to apply a margin left of the x-axis labels.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisLabelLeftMargin="100">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisLabelLeftMargin ="100";
    * ```
    */
    get xAxisLabelLeftMargin() {
        return this.i.jw;
    }
    set xAxisLabelLeftMargin(v) {
        this.i.jw = +v;
    }
    /**
     * Gets or sets the top margin of labels on the X-axis
    *
    * Use the `xAxisLabelTopMargin` property to apply a margin above the x-axis labels.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisTitle="x axis"
    *     xAxisTitleTopMargin="100">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisTitleTopMargin="250";
    * ```
    */
    get xAxisLabelTopMargin() {
        return this.i.jy;
    }
    set xAxisLabelTopMargin(v) {
        this.i.jy = +v;
    }
    /**
     * Gets or sets the right margin of labels on the X-axis
    *
    * Use the `xAxisLabelRightMargin` property to apply a margin right of the x-axis labels.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisLabelRightMargin="100">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisLabelRightMargin="100";
    * ```
    */
    get xAxisLabelRightMargin() {
        return this.i.jx;
    }
    set xAxisLabelRightMargin(v) {
        this.i.jx = +v;
    }
    /**
     * Gets or sets the bottom margin of labels on the X-axis
    *
    * Use the `xAxisLabelBottomMargin` property to apply a margin below the x-axis labels.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisLabelBottomMargin="100">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisLabelBottomMargin ="100";
    * ```
    */
    get xAxisLabelBottomMargin() {
        return this.i.jv;
    }
    set xAxisLabelBottomMargin(v) {
        this.i.jv = +v;
    }
    /**
     * Gets or sets the left margin of labels on the Y-axis
    *
    * Use the `yAxisLabelLeftMargin` property to apply a margin left of the y-axis labels.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisLabelLeftMargin="100">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart. yAxisLabelLeftMargin="250";
    * ```
    */
    get yAxisLabelLeftMargin() {
        return this.i.kd;
    }
    set yAxisLabelLeftMargin(v) {
        this.i.kd = +v;
    }
    /**
     * Gets or sets the top margin of labels on the Y-axis
    *
    * Use the `yAxisLabelTopMargin` property to apply a margin above the y-axis labels.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisLabelTopMargin="100">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisLabelTopMargin="250";
    * ```
    */
    get yAxisLabelTopMargin() {
        return this.i.kf;
    }
    set yAxisLabelTopMargin(v) {
        this.i.kf = +v;
    }
    /**
     * Gets or sets the right margin of labels on the Y-axis
    *
    * Use the `yAxisLabelRightMargin` property to apply a margin right of the y-axis labels.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisLabelRightMargin="100">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisLabelRightMargin="250";
    * ```
    */
    get yAxisLabelRightMargin() {
        return this.i.ke;
    }
    set yAxisLabelRightMargin(v) {
        this.i.ke = +v;
    }
    /**
     * Gets or sets the bottom margin of labels on the Y-axis
    *
    * Use the `yAxisLabelBottomMargin` property to apply a margin below the y-axis labels.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisLabelBottomMargin="100">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisLabelBottomMargin="250";
    * ```
    */
    get yAxisLabelBottomMargin() {
        return this.i.kc;
    }
    set yAxisLabelBottomMargin(v) {
        this.i.kc = +v;
    }
    /**
     * Gets or sets color of labels on the X-axis
    *
    * To change the color of x-axis labels, set the `xAxisLabelTextColor` property to a color string.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisLabelTextColor="green">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisLabelTextColor="green";
    * ```
    */
    get xAxisLabelTextColor() {
        return brushToString(this.i.ob);
    }
    set xAxisLabelTextColor(v) {
        this.i.ob = stringToBrush(v);
    }
    /**
     * Gets or sets color of labels on the Y-axis
    *
    * To change the color of y-axis labels, set the `yAxisLabelTextColor` property to a color string.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisLabelTextColor="green">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisLabelTextColor="green";
    * ```
    */
    get yAxisLabelTextColor() {
        return brushToString(this.i.oi);
    }
    set yAxisLabelTextColor(v) {
        this.i.oi = stringToBrush(v);
    }
    /**
     * Gets or sets the margin around a title on the X-axis
    *
    * Use the `xAxisTitleMargin` property to set a margin around the y-axis title.
    */
    get xAxisTitleMargin() {
        return this.i.j7;
    }
    set xAxisTitleMargin(v) {
        this.i.j7 = +v;
    }
    /**
     * Gets or sets the margin around a title on the Y-axis
    *
    * Use the `yAxisTitleMargin` property to set a margin around the y-axis title.
    */
    get yAxisTitleMargin() {
        return this.i.ko;
    }
    set yAxisTitleMargin(v) {
        this.i.ko = +v;
    }
    /**
     * Gets or sets the left margin of a title on the X-axis
    *
    * Use the `xAxisTitleLeftMargin` property to apply a margin left of the x-axis title.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisTitle="x axis"
    *     xAxisTitleAlignment="left"
    *     xAxisTitleLeftMargin="100">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisTitleLeftMargin="150";
    * ```
    */
    get xAxisTitleLeftMargin() {
        return this.i.j6;
    }
    set xAxisTitleLeftMargin(v) {
        this.i.j6 = +v;
    }
    /**
     * Gets or sets the left margin of a title on the Y-axis
    *
    * Use the `yAxisTitleLeftMargin` property to apply a margin left of the y-axis title.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisTitle="y axis"
    *     yAxisTitleLeftMargin="100">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisTitleLeftMargin="150";
    * ```
    */
    get yAxisTitleLeftMargin() {
        return this.i.kn;
    }
    set yAxisTitleLeftMargin(v) {
        this.i.kn = +v;
    }
    /**
     * Gets or sets the top margin of a title on the X-axis
    *
    * Use the `xAxisTitleTopMargin` property to apply a margin above the x-axis title.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisTitle="x axis"
    *     xAxisTitleTopMargin="100">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisTitleTopMargin="150";
    * ```
    */
    get xAxisTitleTopMargin() {
        return this.i.j9;
    }
    set xAxisTitleTopMargin(v) {
        this.i.j9 = +v;
    }
    /**
     * Gets or sets the top margin of a title on the Y-axis
    *
    * Use the `yAxisTitleTopMargin` property to apply a margin above the y-axis title.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisTitle="y axis"
    *     yAxisTitleAlignment="top"
    *     yAxisTitleTopMargin="100">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisTitleTopMargin="150";
    * ```
    */
    get yAxisTitleTopMargin() {
        return this.i.kq;
    }
    set yAxisTitleTopMargin(v) {
        this.i.kq = +v;
    }
    /**
     * Gets or sets the right margin of a title on the X-axis
    *
    * Use the `xAxisTitleRightMargin` property to apply a margin right of the x-axis title.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisTitle="x axis"
    *     xAxisTitleAlignment="right"
    *     xAxisTitleRightMargin="100">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisTitleRightMargin="100";
    * ```
    */
    get xAxisTitleRightMargin() {
        return this.i.j8;
    }
    set xAxisTitleRightMargin(v) {
        this.i.j8 = +v;
    }
    /**
     * Gets or sets the right margin of a title on the Y-axis
    *
    * Use the `yAxisTitleRightMargin` property to apply a margin right of the y-axis title.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisTitle="y axis"
    *     yAxisTitleRightMargin="100">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisTitleRightMargin="150";
    * ```
    */
    get yAxisTitleRightMargin() {
        return this.i.kp;
    }
    set yAxisTitleRightMargin(v) {
        this.i.kp = +v;
    }
    /**
     * Gets or sets the bottom margin of a title on the X-axis
    *
    * Use the `xAxisTitleBottomMargin` property to apply a margin below the x-axis title.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisTitle="x axis"
    *     xAxisTitleBottomMargin="100">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisTitleBottomMargin="150";
    * ```
    */
    get xAxisTitleBottomMargin() {
        return this.i.j5;
    }
    set xAxisTitleBottomMargin(v) {
        this.i.j5 = +v;
    }
    /**
     * Gets or sets the bottom margin of a title on the Y-axis
    *
    * Use the `yAxisTitleBottomMargin` property to apply a margin below the y-axis title.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisTitle="y axis"
    *     yAxisTitleAlignment="bottom"
    *     yAxisTitleBottomMargin="100">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisTitleBottomMargin="150";
    * ```
    */
    get yAxisTitleBottomMargin() {
        return this.i.km;
    }
    set yAxisTitleBottomMargin(v) {
        this.i.km = +v;
    }
    /**
     * Gets or sets color of title on the X-axis
    *
    * To change the color of the x-axis title, set the `xAxisTitleTextColor` property to a color string.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisTitle="x axis"
    *     xAxisTitleTextColor="green">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisTitleTextColor="green";
    * ```
    */
    get xAxisTitleTextColor() {
        return brushToString(this.i.oh);
    }
    set xAxisTitleTextColor(v) {
        this.i.oh = stringToBrush(v);
    }
    /**
     * Gets or sets color of title on the Y-axis
    *
    * To change the color of the y-axis title, set the `yAxisTitleTextColor` property to a color string.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisTitle="y axis"
    *     yAxisTitleTextColor="green">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisTitleTextColor="green";
    * ```
    */
    get yAxisTitleTextColor() {
        return brushToString(this.i.oo);
    }
    set yAxisTitleTextColor(v) {
        this.i.oo = stringToBrush(v);
    }
    /**
     * Gets or sets CSS font property for labels on X-axis
    *
    * To change the font of x-axis labels, set the `xAxisLabelTextStyle` property.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisLabelTextStyle="italic 15px arial, sans-serif">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisLabelTextStyle="italic 15px arial, sans-serif";
    * ```
    */
    get xAxisLabelTextStyle() {
        return this.i.lh;
    }
    set xAxisLabelTextStyle(v) {
        this.i.lh = v;
    }
    /**
     * Gets or sets CSS font property for labels on Y-axis
    *
    * To change the font of x-axis labels, set the `yAxisLabelTextStyle` property.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisLabelTextStyle="italic 15px arial, sans-serif">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart. yAxisLabelTextStyle="italic 15px arial, sans-serif";
    * ```
    */
    get yAxisLabelTextStyle() {
        return this.i.mm;
    }
    set yAxisLabelTextStyle(v) {
        this.i.mm = v;
    }
    /**
     * Gets or sets CSS font property for title on X-axis
    *
    * To change the font of the x-axis title, set the `xAxisTitleTextStyle` property.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisTitle="x axis"
    *     xAxisTitleTextStyle="italic 15px arial, sans-serif">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisTitleTextStyle="italic 15px arial, sans-serif";
    * ```
    */
    get xAxisTitleTextStyle() {
        return this.i.l5;
    }
    set xAxisTitleTextStyle(v) {
        this.i.l5 = v;
    }
    /**
     * Gets or sets CSS font property for title on Y-axis
    *
    * To change the font of the x-axis title, set the `yAxisTitleTextStyle` property.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisTitle="y axis"
    *     yAxisTitleTextStyle="italic 15px arial, sans-serif">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisTitleTextStyle="italic 15px arial, sans-serif";
    * ```
    */
    get yAxisTitleTextStyle() {
        return this.i.na;
    }
    set yAxisTitleTextStyle(v) {
        this.i.na = v;
    }
    /**
     * Gets or sets the format for labels along the X-axis.
    */
    get xAxisLabel() {
        return this.i.kt;
    }
    set xAxisLabel(v) {
        this.i.kt = v;
    }
    /**
     * Gets or sets the property or string from which the labels are derived.
    */
    get yAxisLabel() {
        return this.i.ku;
    }
    set yAxisLabel(v) {
        this.i.ku = v;
    }
    /**
     * Gets or sets the color to apply to major gridlines along the X-axis.
    *
    * Set `xAxisMajorStroke` to control the color of major gridlines extending vertically from the x-axis.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisMajorStroke="green">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisMajorStroke="green";
    * ```
    */
    get xAxisMajorStroke() {
        return brushToString(this.i.oc);
    }
    set xAxisMajorStroke(v) {
        this.i.oc = stringToBrush(v);
    }
    /**
     * Gets or sets the color to apply to major gridlines along the Y-axis.
    *
    * Set `yAxisMajorStroke` to control the color of major gridlines extending horizontally from the y-axis.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisMajorStroke="green">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisMajorStroke="green";
    * ```
    */
    get yAxisMajorStroke() {
        return brushToString(this.i.oj);
    }
    set yAxisMajorStroke(v) {
        this.i.oj = stringToBrush(v);
    }
    /**
     * Gets or sets the thickness to apply to major gridlines along the X-axis.
    *
    * Set `xAxisMajorStrokeThickness` to control the thickness of major gridlines extending vertically from the x-axis.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisMajorStroke="green"
    *     xAxisMajorStrokeThickness="3">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisMajorStrokeThickness="3" ;
    * ```
    */
    get xAxisMajorStrokeThickness() {
        return this.i.jz;
    }
    set xAxisMajorStrokeThickness(v) {
        this.i.jz = +v;
    }
    /**
     * Gets or sets the thickness to apply to major gridlines along the Y-axis.
    *
    * Set `yAxisMajorStrokeThickness` to control the thickness of major gridlines extending horizontally from the y-axis.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisMajorStrokeThickness="3">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisMajorStrokeThickness="2";
    * ```
    */
    get yAxisMajorStrokeThickness() {
        return this.i.kg;
    }
    set yAxisMajorStrokeThickness(v) {
        this.i.kg = +v;
    }
    /**
     * Gets or sets the thickness to apply to minor gridlines along the X-axis.
    *
    * Set `xAxisMinorStrokeThickness` to control the thickness of minor gridlines extending vertically from the x-axis.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisMinorStroke="green"
    *     xAxisMinorStrokeThickness="3">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisMinorStrokeThickness="3";
    * ```
    */
    get xAxisMinorStrokeThickness() {
        return this.i.j0;
    }
    set xAxisMinorStrokeThickness(v) {
        this.i.j0 = +v;
    }
    /**
     * Gets or sets the thickness to apply to minor gridlines along the Y-axis.
    *
    * Set `yAxisMinorStrokeThickness` to control the thickness of minor gridlines extending horizontally from the y-axis.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisMinorStroke="green"
    *     yAxisMinorStrokeThickness="3">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisMinorStrokeThickness="3";
    * ```
    */
    get yAxisMinorStrokeThickness() {
        return this.i.kh;
    }
    set yAxisMinorStrokeThickness(v) {
        this.i.kh = +v;
    }
    /**
     * Gets or sets the color to apply to stripes along the X-axis.
    *
    * Set `xAxisStrip` to control the alternating color of stripes extending vertically from the x-axis.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisStrip="green">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisStrip="green";
    * ```
    */
    get xAxisStrip() {
        return brushToString(this.i.oe);
    }
    set xAxisStrip(v) {
        this.i.oe = stringToBrush(v);
    }
    /**
     * Gets or sets the color to apply to stripes along the Y-axis.
    *
    * Set `yAxisStrip` to control the alternating color of stripes extending horizontally from the y-axis.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisStrip="green">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisStrip="green";
    * ```
    */
    get yAxisStrip() {
        return brushToString(this.i.ol);
    }
    set yAxisStrip(v) {
        this.i.ol = stringToBrush(v);
    }
    /**
     * Gets or sets the color to apply to the X-axis line.
    *
    * Set `xAxisStroke` to control the color of the x-axis line.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisStroke="green">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisStroke="green"";
    * ```
    */
    get xAxisStroke() {
        return brushToString(this.i.of);
    }
    set xAxisStroke(v) {
        this.i.of = stringToBrush(v);
    }
    /**
     * Gets or sets the color to apply to the Y-axis line.
    *
    * Set `yAxisStroke` to control the color of the y-axis line.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisStroke="green">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisStroke="green";
    * ```
    */
    get yAxisStroke() {
        return brushToString(this.i.om);
    }
    set yAxisStroke(v) {
        this.i.om = stringToBrush(v);
    }
    /**
     * Gets or sets the thickness to apply to the X-axis line.
    *
    * Set `xAxisStrokeThickness` to control the thickness of the x-axis line.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisStrokeThickness="3">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisStrokeThickness="2";
    * ```
    */
    get xAxisStrokeThickness() {
        return this.i.j1;
    }
    set xAxisStrokeThickness(v) {
        this.i.j1 = +v;
    }
    /**
     * Gets or sets the thickness to apply to the Y-axis line.
    *
    * Set `yAxisStrokeThickness` to control the thickness of the y-axis line.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisStroke="green"
    *     yAxisStrokeThickness="3">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart. yAxisStrokeThickness="2";
    * ```
    */
    get yAxisStrokeThickness() {
        return this.i.ki;
    }
    set yAxisStrokeThickness(v) {
        this.i.ki = +v;
    }
    /**
     * Gets or sets the length of tickmarks along the X-axis.
    *
    * Set `xAxisTickLength` to control the length of tickmarks extending vertically from the x-axis line.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisTickLength="23">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisTickLength="25";
    * ```
    */
    get xAxisTickLength() {
        return this.i.j2;
    }
    set xAxisTickLength(v) {
        this.i.j2 = +v;
    }
    /**
     * Gets or sets the length of tickmarks along the Y-axis.
    *
    * Set `yAxisTickLength` to control the length of tickmarks extending horizontally from the y-axis line.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisTickLength="23">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisTickLength="25";
    * ```
    */
    get yAxisTickLength() {
        return this.i.kj;
    }
    set yAxisTickLength(v) {
        this.i.kj = +v;
    }
    /**
     * Gets or sets the color to apply to tickmarks along the X-axis.
    *
    * Set `xAxisTickStroke` to control the color of tickmarks extending vertically from the x-axis line.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisTickStroke="green">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart. xAxisTickStroke="green";
    * ```
    */
    get xAxisTickStroke() {
        return brushToString(this.i.og);
    }
    set xAxisTickStroke(v) {
        this.i.og = stringToBrush(v);
    }
    /**
     * Gets or sets the color to apply to tickmarks along the Y-axis.
    *
    * Set `yAxisTickStroke` to control the color of tickmarks extending horizontally from the y-axis line.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisTickLength="5"
    *     yAxisTickStroke="green">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisTickStroke="green";
    * ```
    */
    get yAxisTickStroke() {
        return brushToString(this.i.on);
    }
    set yAxisTickStroke(v) {
        this.i.on = stringToBrush(v);
    }
    /**
     * Gets or sets the thickness to apply to tickmarks along the X-axis.
    *
    * Set `xAxisTickStrokeThickness` to control the thickness of tickmarks extending vertically from the x-axis line.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisTickStrokeThickness="10">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisTickStrokeThickness="20";
    * ```
    */
    get xAxisTickStrokeThickness() {
        return this.i.j3;
    }
    set xAxisTickStrokeThickness(v) {
        this.i.j3 = +v;
    }
    /**
     * Gets or sets the thickness to apply to tickmarks along the Y-axis.
    *
    * Set `yAxisTickStrokeThickness` to control the thickness of tickmarks extending vertically from the y-axis line.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisTickLength="5"
    *     yAxisTickStrokeThickness="10">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisTickStrokeThickness="25";
    * ```
    */
    get yAxisTickStrokeThickness() {
        return this.i.kk;
    }
    set yAxisTickStrokeThickness(v) {
        this.i.kk = +v;
    }
    /**
     * Gets or sets the Text to display below the X-axis.
    *
    * Set `xAxisTitle` to display a label adjacent to the x-axis.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisTitle="x axis">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart. xAxisTitle="x axis";
    * ```
    */
    get xAxisTitle() {
        return this.i.lw;
    }
    set xAxisTitle(v) {
        this.i.lw = v;
    }
    /**
     * Gets or sets the Text to display to the left of the Y-axis.
    *
    * Set `yAxisTitle` to display a label adjacent to the y-axis.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisTitle="y axis">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisTitle="y axis";
    * ```
    */
    get yAxisTitle() {
        return this.i.m1;
    }
    set yAxisTitle(v) {
        this.i.m1 = v;
    }
    /**
     * Gets or sets the color to apply to minor gridlines along the X-axis.
    *
    * Set `xAxisMinorStroke` to control the color of minor gridlines extending vertically from the x-axis.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisMinorStroke="green">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisExtent="250";
    * ```
    */
    get xAxisMinorStroke() {
        return brushToString(this.i.od);
    }
    set xAxisMinorStroke(v) {
        this.i.od = stringToBrush(v);
    }
    /**
     * Gets or sets the color to apply to minor gridlines along the Y-axis.
    *
    * Set `yAxisMinorStroke` to control the color of minor gridlines extending horizontally from the y-axis.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisMinorStroke="green"
    *     yAxisMinorStrokeThickness="1">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisMinorStroke="green";
    * ```
    */
    get yAxisMinorStroke() {
        return brushToString(this.i.ok);
    }
    set yAxisMinorStroke(v) {
        this.i.ok = stringToBrush(v);
    }
    /**
     * Gets or sets the angle of rotation for labels along the X-axis.
    *
    * To control the rotation of labels on the x-axis, set the `xAxisLabelAngle` property to a number of rotational degrees.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisLabelAngle="45">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisLabelAngle="45";
    * ```
    */
    get xAxisLabelAngle() {
        return this.i.ju;
    }
    set xAxisLabelAngle(v) {
        this.i.ju = +v;
    }
    /**
     * Gets or sets the angle of rotation for labels along the Y-axis.
    *
    * To control the rotation of labels on the y-axis, set the `yAxisLabelAngle` property to a number of rotational degrees.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisLabelAngle="45">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisLabelAngle="25";
    * ```
    */
    get yAxisLabelAngle() {
        return this.i.kb;
    }
    set yAxisLabelAngle(v) {
        this.i.kb = +v;
    }
    /**
     * Gets or sets the distance between the X-axis and the bottom of the chart.
    *
    * To allocate space between the x-axis and the edge of the chart, set the `xAxisExtent` property.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisExtent="250">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisExtent="250";
    * ```
    */
    get xAxisExtent() {
        return this.i.jt;
    }
    set xAxisExtent(v) {
        this.i.jt = +v;
    }
    /**
     * Gets or sets the distance between the Y-axis and the left edge of the chart.
    *
    * To allocate space between the x-axis and the edge of the chart, set the `xAxisExtent` property.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisExtent="250">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisExtent="250";
    * ```
    */
    get yAxisExtent() {
        return this.i.ka;
    }
    set yAxisExtent(v) {
        this.i.ka = +v;
    }
    /**
     * Gets or sets the angle of rotation for the X-axis title.
    *
    * Set `xAxisTitleAngle` to control the rotation of the x-axis title.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisTitle="x axis"
    *     xAxisTitleAngle="45">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisTitleAngle="50";
    * ```
    */
    get xAxisTitleAngle() {
        return this.i.j4;
    }
    set xAxisTitleAngle(v) {
        this.i.j4 = +v;
    }
    /**
     * Gets or sets the angle of rotation for the Y-axis title.
    *
    * Set `yAxisTitleAngle` to control the rotation of the y-axis title.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisTitle="y axis"
    *     yAxisTitleAngle="45">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisTitleAngle="25";
    * ```
    */
    get yAxisTitleAngle() {
        return this.i.kl;
    }
    set yAxisTitleAngle(v) {
        this.i.kl = +v;
    }
    /**
     * Gets or sets whether to invert the direction of the X-axis by placing the first data items on the right side of the chart.
    *
    * To reverse the direction of items along the x-axis, set the `xAxisInverted` property to true.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisInverted="true">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.XAxisInverted="True";
    * ```
    */
    get xAxisInverted() {
        return this.i.ip;
    }
    set xAxisInverted(v) {
        this.i.ip = ensureBool(v);
    }
    /**
     * Gets or sets whether to invert the direction of the Y-axis by placing the minimum numeric value at the top of the chart.
    *
    * You can use `yAxisInverted` to invert the scale of the y axis like this:
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisInverted="true"
    * >
    *
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisInverted="true";
    * ```
    *
    * The default value of `yAxisInverted` is false.
    */
    get yAxisInverted() {
        return this.i.iq;
    }
    set yAxisInverted(v) {
        this.i.iq = ensureBool(v);
    }
    /**
     * Gets or sets Horizontal alignment of the X-axis title.
    *
    * Set `xAxisTitleAlignment` to control the horizontal alignment of the x-axis title.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisTitle="x axis"
    *     xAxisTitleAlignment="right">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisTitleAlignment="right";
    * ```
    */
    get xAxisTitleAlignment() {
        return this.i.nv;
    }
    set xAxisTitleAlignment(v) {
        this.i.nv = ensureEnum(HorizontalAlignment_$type, v);
    }
    /**
     * Gets or sets Vertical alignment of the Y-axis title.
    *
    * Set `yAxisTitleAlignment` to control the vertical alignment of the x-axis title.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisTitle="y axis"
    *     yAxisTitleAlignment="bottom">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisTitleAlignment="bottom";
    * ```
    */
    get yAxisTitleAlignment() {
        return this.i.ou;
    }
    set yAxisTitleAlignment(v) {
        this.i.ou = ensureEnum(VerticalAlignment_$type, v);
    }
    /**
     * Gets or sets Horizontal alignment of X-axis labels.
    */
    get xAxisLabelHorizontalAlignment() {
        return this.i.nu;
    }
    set xAxisLabelHorizontalAlignment(v) {
        this.i.nu = ensureEnum(HorizontalAlignment_$type, v);
    }
    /**
     * Gets or sets Horizontal alignment of Y-axis labels.
    *
    * Set the `yAxisLabelHorizontalAlignment` property to change the horizontal alignment of y-axis labels.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisExtent="250"
    *     yAxisLabelHorizontalAlignment="left">
    * </igx-category-chart>
    * ```
    */
    get yAxisLabelHorizontalAlignment() {
        return this.i.nw;
    }
    set yAxisLabelHorizontalAlignment(v) {
        this.i.nw = ensureEnum(HorizontalAlignment_$type, v);
    }
    /**
     * Gets or sets Vertical alignment of X-axis labels.
    *
    * Use the `xAxisLabelVerticalAlignment` property to change the vertical position of x-axis labels.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisExtent="250"
    *     xAxisLabelVerticalAlignment="bottom">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisLabelVerticalAlignment="bottom";
    * ```
    */
    get xAxisLabelVerticalAlignment() {
        return this.i.os;
    }
    set xAxisLabelVerticalAlignment(v) {
        this.i.os = ensureEnum(VerticalAlignment_$type, v);
    }
    /**
     * Gets or sets Vertical alignment of Y-axis labels.
    */
    get yAxisLabelVerticalAlignment() {
        return this.i.ot;
    }
    set yAxisLabelVerticalAlignment(v) {
        this.i.ot = ensureEnum(VerticalAlignment_$type, v);
    }
    /**
     * Gets or sets Visibility of X-axis labels.
    *
    * To hide the x-axis labels, set `xAxisLabelVisibility` to collapsed.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     xAxisLabelVisibility="collapsed">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.xAxisLabelVisibility="collapsed";
    * ```
    */
    get xAxisLabelVisibility() {
        return this.i.ox;
    }
    set xAxisLabelVisibility(v) {
        this.i.ox = ensureEnum(Visibility_$type, v);
    }
    /**
     * Gets or sets Visibility of Y-axis labels.
    *
    * To hide the y-axis labels, set `yAxisLabelVisibility` to collapsed.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisLabelVisibility="collapsed">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisLabelVisibility="collapsed"";
    * ```
    */
    get yAxisLabelVisibility() {
        return this.i.oy;
    }
    set yAxisLabelVisibility(v) {
        this.i.oy = ensureEnum(Visibility_$type, v);
    }
    /**
     * The location of Y-axis labels, relative to the plot area.
    *
    * Set the `yAxisLabelLocation` property to change the location of the y-axis and its labels.
    *
    * ```html
    * <igx-category-chart
    *     [dataSource]="data"
    *     yAxisLabelLocation="outsideRight">
    * </igx-category-chart>
    * ```
    *
    * ```ts
    * this.chart.yAxisLabelLocation="outsideRight";
    * ```
    */
    get yAxisLabelLocation() {
        return this.i.il;
    }
    set yAxisLabelLocation(v) {
        this.i.il = ensureEnum(AxisLabelsLocation_$type, v);
    }
    /**
     * Gets or sets the format string for the X axis label.
    */
    get xAxisLabelFormat() {
        return this.i.k9;
    }
    set xAxisLabelFormat(v) {
        this.i.k9 = v;
    }
    /**
     * Gets or sets the format specifiers to use with the XAxisLabelFormat string.
    */
    get xAxisLabelFormatSpecifiers() {
        return this.i.ib;
    }
    set xAxisLabelFormatSpecifiers(v) {
        this.i.ib = v;
    }
    /**
     * Gets or sets the format string for the Y axis label.
    */
    get yAxisLabelFormat() {
        return this.i.md;
    }
    set yAxisLabelFormat(v) {
        this.i.md = v;
    }
    /**
     * Gets or sets the format specifiers to use with the YAxisLabelFormat string.
    */
    get yAxisLabelFormatSpecifiers() {
        return this.i.ic;
    }
    set yAxisLabelFormatSpecifiers(v) {
        this.i.ic = v;
    }
    findByName(name) {
        var baseResult = super.findByName(name);
        if (baseResult) {
            return baseResult;
        }
        if (this.xAxisLabelFormatSpecifiers != null && arrayFindByName(this.xAxisLabelFormatSpecifiers, name)) {
            return arrayFindByName(this.xAxisLabelFormatSpecifiers, name);
        }
        if (this.yAxisLabelFormatSpecifiers != null && arrayFindByName(this.yAxisLabelFormatSpecifiers, name)) {
            return arrayFindByName(this.yAxisLabelFormatSpecifiers, name);
        }
        return null;
    }
    /**
     * Called by the UI framework to provide a UI container for rendering this control.
    
    * @param container  * The UI container element.
    */
    provideContainer(container) {
        this.i.provideContainer(container);
    }
    /**
     * Converts the given visual location to a data value.
    
    * @param unscaledValue  * The x-coordinate of the location to scale.
    */
    getScaledValueX(unscaledValue) {
        let iv = this.i.jp(unscaledValue);
        return (iv);
    }
    /**
     * Converts the given data value to a visual location.
    
    * @param scaledValue  * The data value to un-scale.
    */
    getUnscaledValueX(scaledValue) {
        let iv = this.i.jr(scaledValue);
        return (iv);
    }
    /**
     * Converts the given visual location to a data value.
    
    * @param unscaledValue  * The y-coordinate of the location to scale.
    */
    getScaledValueY(unscaledValue) {
        let iv = this.i.jq(unscaledValue);
        return (iv);
    }
    /**
     * Converts the given data value to a visual location.
    
    * @param scaledValue  * The data value to un-scale.
    */
    getUnscaledValueY(scaledValue) {
        let iv = this.i.js(scaledValue);
        return (iv);
    }
}
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Function])
], IgxXYChartComponent.prototype, "xAxisFormatLabel", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Function])
], IgxXYChartComponent.prototype, "yAxisFormatLabel", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "xAxisLabelLeftMargin", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "xAxisLabelTopMargin", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "xAxisLabelRightMargin", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "xAxisLabelBottomMargin", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisLabelLeftMargin", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisLabelTopMargin", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisLabelRightMargin", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisLabelBottomMargin", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "xAxisLabelTextColor", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "yAxisLabelTextColor", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "xAxisTitleMargin", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisTitleMargin", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "xAxisTitleLeftMargin", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisTitleLeftMargin", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "xAxisTitleTopMargin", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisTitleTopMargin", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "xAxisTitleRightMargin", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisTitleRightMargin", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "xAxisTitleBottomMargin", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisTitleBottomMargin", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "xAxisTitleTextColor", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "yAxisTitleTextColor", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "xAxisLabelTextStyle", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "yAxisLabelTextStyle", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "xAxisTitleTextStyle", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "yAxisTitleTextStyle", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Object),
    tslib_1.__metadata("design:paramtypes", [Object])
], IgxXYChartComponent.prototype, "xAxisLabel", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Object),
    tslib_1.__metadata("design:paramtypes", [Object])
], IgxXYChartComponent.prototype, "yAxisLabel", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "xAxisMajorStroke", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "yAxisMajorStroke", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "xAxisMajorStrokeThickness", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisMajorStrokeThickness", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "xAxisMinorStrokeThickness", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisMinorStrokeThickness", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "xAxisStrip", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "yAxisStrip", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "xAxisStroke", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "yAxisStroke", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "xAxisStrokeThickness", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisStrokeThickness", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "xAxisTickLength", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisTickLength", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "xAxisTickStroke", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "yAxisTickStroke", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "xAxisTickStrokeThickness", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisTickStrokeThickness", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "xAxisTitle", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "yAxisTitle", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "xAxisMinorStroke", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "yAxisMinorStroke", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "xAxisLabelAngle", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisLabelAngle", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "xAxisExtent", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisExtent", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "xAxisTitleAngle", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisTitleAngle", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Boolean),
    tslib_1.__metadata("design:paramtypes", [Boolean])
], IgxXYChartComponent.prototype, "xAxisInverted", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Boolean),
    tslib_1.__metadata("design:paramtypes", [Boolean])
], IgxXYChartComponent.prototype, "yAxisInverted", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "xAxisTitleAlignment", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisTitleAlignment", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "xAxisLabelHorizontalAlignment", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisLabelHorizontalAlignment", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "xAxisLabelVerticalAlignment", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisLabelVerticalAlignment", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "xAxisLabelVisibility", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisLabelVisibility", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxXYChartComponent.prototype, "yAxisLabelLocation", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "xAxisLabelFormat", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Array),
    tslib_1.__metadata("design:paramtypes", [Array])
], IgxXYChartComponent.prototype, "xAxisLabelFormatSpecifiers", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxXYChartComponent.prototype, "yAxisLabelFormat", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Array),
    tslib_1.__metadata("design:paramtypes", [Array])
], IgxXYChartComponent.prototype, "yAxisLabelFormatSpecifiers", null);
