// igniteui-angular-charts/esm2015/lib/igx-scatter-contour-series-component.js
var IgxScatterContourSeriesComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxValueBrushScaleComponent } from "./igx-value-brush-scale-component";
import { IgxContourValueResolverComponent } from "./igx-contour-value-resolver-component";
import { IgxScatterTriangulationSeriesComponent } from "./igx-scatter-triangulation-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { ScatterContourSeries } from "./ScatterContourSeries";
/**
 * Series class for rendering isarithmic contours based on a triangulation of X+Y+Value points in the ItemsSource.
*/
export const IgxScatterContourSeriesComponent_PROVIDERS = [{ provide: IgxScatterTriangulationSeriesComponent, useExisting: forwardRef(() => IgxScatterContourSeriesComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxScatterContourSeriesComponent) }];
let IgxScatterContourSeriesComponent = IgxScatterContourSeriesComponent_1 = class IgxScatterContourSeriesComponent extends IgxScatterTriangulationSeriesComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new ScatterContourSeries();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Gets or set the name of the property from which to retrieve the numeric values from the ItemsSource items.
    */
    get valueMemberPath() {
        return this.i.nq;
    }
    set valueMemberPath(v) {
        this.i.nq = v;
    }
    /**
     * Gets or sets the ValueBrushScale to use when determining Brushes for each contour line, based on the values found in ValueMemberPath.
    */
    get fillScale() {
        if (this.i.nl == null) {
            return null;
        }
        if (!this.i.nl.externalObject) {
            let e = IgxValueBrushScaleComponent._createFromInternal(this.i.nl);
            if (e) {
                e._implementation = this.i.nl;
            }
            this.i.nl.externalObject = e;
        }
        return this.i.nl.externalObject;
    }
    set fillScale(v) {
        if (v != null && this._stylingContainer && v._styling)
            v._styling(this._stylingContainer, this, this);
        v == null ? this.i.nl = null : this.i.nl = v.i;
    }
    /**
     * Gets or set the ContourValueResolver used to determine the numeric values of contours.
    */
    get valueResolver() {
        if (this.i.nj == null) {
            return null;
        }
        if (!this.i.nj.externalObject) {
            let e = IgxContourValueResolverComponent._createFromInternal(this.i.nj);
            if (e) {
                e._implementation = this.i.nj;
            }
            this.i.nj.externalObject = e;
        }
        return this.i.nj.externalObject;
    }
    set valueResolver(v) {
        if (v != null && this._stylingContainer && v._styling)
            v._styling(this._stylingContainer, this, this);
        v == null ? this.i.nj = null : this.i.nj = v.i;
    }
    findByName(name) {
        var baseResult = super.findByName(name);
        if (baseResult) {
            return baseResult;
        }
        if (this.fillScale && this.fillScale.name && this.fillScale.name == name) {
            return this.fillScale;
        }
        if (this.valueResolver && this.valueResolver.name && this.valueResolver.name == name) {
            return this.valueResolver;
        }
        return null;
    }
    _styling(container, component, parent) {
        super._styling(container, component, parent);
        this._inStyling = true;
        if (this.fillScale && this.fillScale._styling) {
            this.fillScale._styling(container, component, this);
        }
        if (this.valueResolver && this.valueResolver._styling) {
            this.valueResolver._styling(container, component, this);
        }
        this._inStyling = false;
    }
    getItemValue(item, memberPathName) {
        let iv = this.i.fs(item, memberPathName);
        return (iv);
    }
};
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", String),
    tslib_1.__metadata("design:paramtypes", [String])
], IgxScatterContourSeriesComponent.prototype, "valueMemberPath", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", IgxValueBrushScaleComponent),
    tslib_1.__metadata("design:paramtypes", [IgxValueBrushScaleComponent])
], IgxScatterContourSeriesComponent.prototype, "fillScale", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", IgxContourValueResolverComponent),
    tslib_1.__metadata("design:paramtypes", [IgxContourValueResolverComponent])
], IgxScatterContourSeriesComponent.prototype, "valueResolver", null);
IgxScatterContourSeriesComponent = IgxScatterContourSeriesComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-scatter-contour-series',
        template: ``,
        providers: [{ provide: IgxScatterTriangulationSeriesComponent, useExisting: forwardRef(() => IgxScatterContourSeriesComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxScatterContourSeriesComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxScatterContourSeriesComponent);
export { IgxScatterContourSeriesComponent };
