// igniteui-angular-charts/esm2015/lib/igx-average-true-range-indicator-component.js
var IgxAverageTrueRangeIndicatorComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxStrategyBasedIndicatorComponent } from "./igx-strategy-based-indicator-component";
import { IgxFinancialIndicatorComponent } from "./igx-financial-indicator-component";
import { IgxFinancialSeriesComponent } from "./igx-financial-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { AverageTrueRangeIndicator } from "./AverageTrueRangeIndicator";
/**
 * Represents a IgxDataChartComponent Average True Range indicator series.
 * Default required members: High, Low, Close
*
* You can use the `IgxAverageTrueRangeIndicatorComponent` to measures a security's degree of price movement or volatility within a given period of time.
*
* ```html
* <igx-data-chart
* 	#chart
*     [dataSource]="data" >
*     <igx-category-x-axis
* 		label="label"
* 		#xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
* 		#yAxis>
*     </igx-numeric-y-axis>
*     <igx-average-true-range-indicator
* 		[xAxis]="xAxis"
* 		[yAxis]="yAxis"
* 		openMemberPath="open"
* 		highMemberPath="high"
* 		lowMemberPath="low"
* 		closeMemberPath="close">
*     </igx-average-true-range-indicator>
* </igx-data-chart>
* ```
*
* ```ts
* let series = new IgxAverageTrueRangeIndicatorComponent();
* series.xAxis = this.xAxis;
* series.yAxis = this.yAxis;
* series.openMemberPath = "open";
* series.highMemberPath = "high";
* series.lowMemberPath = "low";
* series.closeMemberPath = "close";
* this.chart.series.add(series);
* ```
*/
export const IgxAverageTrueRangeIndicatorComponent_PROVIDERS = [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(() => IgxAverageTrueRangeIndicatorComponent) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(() => IgxAverageTrueRangeIndicatorComponent) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(() => IgxAverageTrueRangeIndicatorComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxAverageTrueRangeIndicatorComponent) }];
let IgxAverageTrueRangeIndicatorComponent = IgxAverageTrueRangeIndicatorComponent_1 = class IgxAverageTrueRangeIndicatorComponent extends IgxStrategyBasedIndicatorComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new AverageTrueRangeIndicator();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Gets or sets the moving average period for the current AverageTrueRangeSeries object.
     * The typical, and initial, value for AverageTrueRange periods is 14.
    *
    * You can use the `period` to set the moving average.
    *
    * ```html
    * <igx-data-chart
    *     [dataSource]="data" >
    *     <igx-category-x-axis
    * 		label="label"
    * 		#xAxis>
    *     </igx-category-x-axis>
    *     <igx-numeric-y-axis
    * 		#yAxis>
    *     </igx-numeric-y-axis>
    *     <igx-average-true-range-indicator
    * 		#series
    * 		[xAxis]="xAxis"
    * 		[yAxis]="yAxis"
    * 		openMemberPath="open"
    * 		highMemberPath="high"
    * 		lowMemberPath="low"
    * 		closeMemberPath="close"
    * 		period="14">
    *     </igx-average-true-range-indicator>
    * </igx-data-chart>
    * ```
    *
    * ```ts
    * this.series.period = 14;
    * ```
    */
    get period() {
        return this.i.qy;
    }
    set period(v) {
        this.i.qy = +v;
    }
};
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxAverageTrueRangeIndicatorComponent.prototype, "period", null);
IgxAverageTrueRangeIndicatorComponent = IgxAverageTrueRangeIndicatorComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-average-true-range-indicator',
        template: ``,
        providers: [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(() => IgxAverageTrueRangeIndicatorComponent_1) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(() => IgxAverageTrueRangeIndicatorComponent_1) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(() => IgxAverageTrueRangeIndicatorComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxAverageTrueRangeIndicatorComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxAverageTrueRangeIndicatorComponent);
export { IgxAverageTrueRangeIndicatorComponent };
