// igniteui-angular-charts/esm2015/lib/igx-percent-change-y-axis-component.js
var IgxPercentChangeYAxisComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxNumericYAxisComponent } from "./igx-numeric-y-axis-component";
import { IgxStraightNumericAxisBaseComponent } from "./igx-straight-numeric-axis-base-component";
import { IgxNumericAxisBaseComponent } from "./igx-numeric-axis-base-component";
import { IgxAxisComponent } from "./igx-axis-component";
import { PercentChangeYAxis } from "./PercentChangeYAxis";
/**
 * A horizontal axis that uses a DateTime scale.
*
* You can use the `IgxPercentChangeYAxisComponent` show percentage values.
*
* ```html
* <igx-data-chart
*     [dataSource]="data">
*     <igx-percent-change-y-axis
* 		#yAxis>
*     </igx-percent-change-y-axis>
* </igx-data-chart>
* ```
*
* ```ts
* let yAxis = new IgxPercentChangeYAxisComponent();
* this.chart.axes.add(yAxis);
* ```
*/
export const IgxPercentChangeYAxisComponent_PROVIDERS = [{ provide: IgxNumericYAxisComponent, useExisting: forwardRef(() => IgxPercentChangeYAxisComponent) }, { provide: IgxStraightNumericAxisBaseComponent, useExisting: forwardRef(() => IgxPercentChangeYAxisComponent) }, { provide: IgxNumericAxisBaseComponent, useExisting: forwardRef(() => IgxPercentChangeYAxisComponent) }, { provide: IgxAxisComponent, useExisting: forwardRef(() => IgxPercentChangeYAxisComponent) }];
let IgxPercentChangeYAxisComponent = IgxPercentChangeYAxisComponent_1 = class IgxPercentChangeYAxisComponent extends IgxNumericYAxisComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new PercentChangeYAxis();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
};
IgxPercentChangeYAxisComponent = IgxPercentChangeYAxisComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-percent-change-y-axis',
        template: ``,
        providers: [{ provide: IgxNumericYAxisComponent, useExisting: forwardRef(() => IgxPercentChangeYAxisComponent_1) }, { provide: IgxStraightNumericAxisBaseComponent, useExisting: forwardRef(() => IgxPercentChangeYAxisComponent_1) }, { provide: IgxNumericAxisBaseComponent, useExisting: forwardRef(() => IgxPercentChangeYAxisComponent_1) }, { provide: IgxAxisComponent, useExisting: forwardRef(() => IgxPercentChangeYAxisComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxPercentChangeYAxisComponent);
export { IgxPercentChangeYAxisComponent };
