// igniteui-angular-charts/esm2015/lib/igx-market-facilitation-index-indicator-component.js
var IgxMarketFacilitationIndexIndicatorComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxItemwiseStrategyBasedIndicatorComponent } from "./igx-itemwise-strategy-based-indicator-component";
import { IgxStrategyBasedIndicatorComponent } from "./igx-strategy-based-indicator-component";
import { IgxFinancialIndicatorComponent } from "./igx-financial-indicator-component";
import { IgxFinancialSeriesComponent } from "./igx-financial-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { MarketFacilitationIndexIndicator } from "./MarketFacilitationIndexIndicator";
/**
 * Represents a IgxDataChartComponent Market Facilitation Index indicator series.
 * The Market Facilitation Index (MFI) indicator is represented by
 * the difference of the high and the low for a bar divided by the volume.
 * Default required members: High, Low, Volume
*
* The Market Facilitation Index (MFI) indicator is represented by the difference of the high and the low for a bar divided by the volume.
*
* Example:
*
* ```html
* <igx-data-chart
*     [dataSource]="data">
* 	<igx-category-x-axis
* 		label="label"
* 		#xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
* 		#yAxis>
*     </igx-numeric-y-axis>
* 	<igx-market-facilitation-index-indicator
*         #series
*         [xAxis]="xAxis"
*         [yAxis]="yAxis"
*         highMemberPath="high"
* 		lowMemberPath="low"
* 		volumeMemberPath="volume">
*     </igx-market-facilitation-index-indicator>
* </igx-data-chart>
* ```
*/
export const IgxMarketFacilitationIndexIndicatorComponent_PROVIDERS = [{ provide: IgxItemwiseStrategyBasedIndicatorComponent, useExisting: forwardRef(() => IgxMarketFacilitationIndexIndicatorComponent) }, { provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(() => IgxMarketFacilitationIndexIndicatorComponent) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(() => IgxMarketFacilitationIndexIndicatorComponent) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(() => IgxMarketFacilitationIndexIndicatorComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxMarketFacilitationIndexIndicatorComponent) }];
let IgxMarketFacilitationIndexIndicatorComponent = IgxMarketFacilitationIndexIndicatorComponent_1 = class IgxMarketFacilitationIndexIndicatorComponent extends IgxItemwiseStrategyBasedIndicatorComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new MarketFacilitationIndexIndicator();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
};
IgxMarketFacilitationIndexIndicatorComponent = IgxMarketFacilitationIndexIndicatorComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-market-facilitation-index-indicator',
        template: ``,
        providers: [{ provide: IgxItemwiseStrategyBasedIndicatorComponent, useExisting: forwardRef(() => IgxMarketFacilitationIndexIndicatorComponent_1) }, { provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(() => IgxMarketFacilitationIndexIndicatorComponent_1) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(() => IgxMarketFacilitationIndexIndicatorComponent_1) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(() => IgxMarketFacilitationIndexIndicatorComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxMarketFacilitationIndexIndicatorComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxMarketFacilitationIndexIndicatorComponent);
export { IgxMarketFacilitationIndexIndicatorComponent };
