// igniteui-angular-charts/esm2015/lib/igx-price-volume-trend-indicator-component.js
var IgxPriceVolumeTrendIndicatorComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxStrategyBasedIndicatorComponent } from "./igx-strategy-based-indicator-component";
import { IgxFinancialIndicatorComponent } from "./igx-financial-indicator-component";
import { IgxFinancialSeriesComponent } from "./igx-financial-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { PriceVolumeTrendIndicator } from "./PriceVolumeTrendIndicator";
/**
 * Represents a IgxDataChartComponent Price Volume Trend Indicator series.
 * Default required members: Volume, Close
*
* You can use the `IgxPriceVolumeTrendIndicatorComponent` to measure money flow by adding or subtracting a portion of the daily volume.
*
* ```html
* <igx-data-chart
*     [dataSource]="data" >
*     <igx-category-x-axis
* 		label="label"
* 		#xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
* 		#yAxis>
*     </igx-numeric-y-axis>
*     <igx-price-volume-trend-indicator
* 		[xAxis]="xAxis"
* 		[yAxis]="yAxis"
* 		openMemberPath="open"
* 		highMemberPath="high"
* 		lowMemberPath="low"
* 		closeMemberPath="close">
*     </igx-price-volume-trend-indicator>
* </igx-data-chart>
* ```
*
* ```ts
* let series = new IgxPriceVolumeTrendIndicatorComponent();
* series.xAxis = this.xAxis;
* series.yAxis = this.yAxis;
* series.openMemberPath = "open";
* series.highMemberPath = "high";
* series.lowMemberPath = "low";
* series.closeMemberPath = "close";
* this.chart.series.add(series);
* ```
*/
export const IgxPriceVolumeTrendIndicatorComponent_PROVIDERS = [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(() => IgxPriceVolumeTrendIndicatorComponent) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(() => IgxPriceVolumeTrendIndicatorComponent) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(() => IgxPriceVolumeTrendIndicatorComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxPriceVolumeTrendIndicatorComponent) }];
let IgxPriceVolumeTrendIndicatorComponent = IgxPriceVolumeTrendIndicatorComponent_1 = class IgxPriceVolumeTrendIndicatorComponent extends IgxStrategyBasedIndicatorComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new PriceVolumeTrendIndicator();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
};
IgxPriceVolumeTrendIndicatorComponent = IgxPriceVolumeTrendIndicatorComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-price-volume-trend-indicator',
        template: ``,
        providers: [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(() => IgxPriceVolumeTrendIndicatorComponent_1) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(() => IgxPriceVolumeTrendIndicatorComponent_1) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(() => IgxPriceVolumeTrendIndicatorComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxPriceVolumeTrendIndicatorComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxPriceVolumeTrendIndicatorComponent);
export { IgxPriceVolumeTrendIndicatorComponent };
