// igniteui-angular-charts/esm2015/lib/igx-scatter-line-series-component.js
var IgxScatterLineSeriesComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { UnknownValuePlotting, UnknownValuePlotting_$type } from "igniteui-angular-core";
import { IgxScatterBaseComponent } from "./igx-scatter-base-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { ScatterLineSeries } from "./ScatterLineSeries";
import { ensureEnum } from "igniteui-angular-core";
/**
 * Represents a IgxDataChartComponent scatter line series
*/
export const IgxScatterLineSeriesComponent_PROVIDERS = [{ provide: IgxScatterBaseComponent, useExisting: forwardRef(() => IgxScatterLineSeriesComponent) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxScatterLineSeriesComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxScatterLineSeriesComponent) }];
let IgxScatterLineSeriesComponent = IgxScatterLineSeriesComponent_1 = class IgxScatterLineSeriesComponent extends IgxScatterBaseComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new ScatterLineSeries();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Determines how unknown values will be plotted on the chart.
     * Null and Double.NaN are two examples of unknown values.
    */
    get unknownValuePlotting() {
        return this.i.o1;
    }
    set unknownValuePlotting(v) {
        this.i.o1 = ensureEnum(UnknownValuePlotting_$type, v);
    }
};
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxScatterLineSeriesComponent.prototype, "unknownValuePlotting", null);
IgxScatterLineSeriesComponent = IgxScatterLineSeriesComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-scatter-line-series',
        template: ``,
        providers: [{ provide: IgxScatterBaseComponent, useExisting: forwardRef(() => IgxScatterLineSeriesComponent_1) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxScatterLineSeriesComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxScatterLineSeriesComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxScatterLineSeriesComponent);
export { IgxScatterLineSeriesComponent };
