// igniteui-angular-charts/esm2015/lib/igx-annotation-layer-component.js
import * as tslib_1 from "tslib";
import { forwardRef, Input } from '@angular/core';
import { IgxSeriesComponent } from "./igx-series-component";
import { ensureBool, fromPoint, toPoint } from "igniteui-angular-core";
/**
 * Represents the base class for annotation layer types. Annotations will usually disable some of the default hover/touch behaviors of the chart when they are introduced to the series collection.
 * In return they provide some valuable information as the cursor is moved by hovering over the chart, or performing a press-hold-drag interaction in touch mode.
*/
export const IgxAnnotationLayerComponent_PROVIDERS = [{ provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxAnnotationLayerComponent) }];
export class IgxAnnotationLayerComponent extends IgxSeriesComponent {
    constructor() {
        super();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Gets or sets whether this series should take up an index for auto brush coloring.
    */
    get useIndex() {
        return this.i.mb;
    }
    set useIndex(v) {
        this.i.mb = ensureBool(v);
    }
    /**
     * Gets or sets whether this series interacts with the chart legend.
    */
    get useLegend() {
        return this.i.mc;
    }
    set useLegend(v) {
        this.i.mc = ensureBool(v);
    }
    /**
     * Gets or sets the world position to use instead of the pointer position for this annotation layer. If you set this value, rather than responding to the pointer, this layer will be fixed on the provided world position.
     * A world position has x and y values that range from 0 to 1 that represent the global position of the cursor relative to the entire ranges of the axes. The default of (NaN, NaN) means that the layer should react to pointer movement.
    */
    get cursorPosition() {
        return fromPoint(this.i.m3);
    }
    set cursorPosition(v) {
        this.i.m3 = toPoint(v);
    }
    /**
     * Gets or sets whether the series viewer's default crosshair should be disabled by the presence of this layer.
    */
    get isDefaultCrosshairDisabled() {
        return this.i.l6;
    }
    set isDefaultCrosshairDisabled(v) {
        this.i.l6 = ensureBool(v);
    }
    /**
     * Gets whether or not the default crosshair behavior is disabled.
    */
    get isDefaultCrosshairBehaviorDisabled() {
        return this.i.cm;
    }
    /**
     * Gets if the series uses an index, which can be used to auto-assign it a color.
    */
    get isIndexed() {
        return this.i.cw;
    }
    /**
     * Gets if the series should appear in any legends.
    */
    get isUsableInLegend() {
        return this.i.isUsableInLegend;
    }
    /**
     * Gets whether this series is an annotation layer.
    */
    get isAnnotationLayer() {
        return this.i.cj;
    }
    /**
     * Gets or sets whether this layer renders as an overlay or not.
    */
    get shouldRenderAsOverlay() {
        return this.i.ma;
    }
    set shouldRenderAsOverlay(v) {
        this.i.ma = ensureBool(v);
    }
}
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Boolean),
    tslib_1.__metadata("design:paramtypes", [Boolean])
], IgxAnnotationLayerComponent.prototype, "useIndex", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Boolean),
    tslib_1.__metadata("design:paramtypes", [Boolean])
], IgxAnnotationLayerComponent.prototype, "useLegend", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Object),
    tslib_1.__metadata("design:paramtypes", [Object])
], IgxAnnotationLayerComponent.prototype, "cursorPosition", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Boolean),
    tslib_1.__metadata("design:paramtypes", [Boolean])
], IgxAnnotationLayerComponent.prototype, "isDefaultCrosshairDisabled", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Boolean),
    tslib_1.__metadata("design:paramtypes", [Boolean])
], IgxAnnotationLayerComponent.prototype, "shouldRenderAsOverlay", null);
