// igniteui-angular-charts/esm2015/lib/igx-line-fragment-component.js
var IgxLineFragmentComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxFragmentBaseComponent } from "./igx-fragment-base-component";
import { IgxHorizontalAnchoredCategorySeriesComponent } from "./igx-horizontal-anchored-category-series-component";
import { IgxAnchoredCategorySeriesComponent } from "./igx-anchored-category-series-component";
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { LineFragment } from "./LineFragment";
/**
 * Represents one part of a StackedLineSeries.
*/
export const IgxLineFragmentComponent_PROVIDERS = [{ provide: IgxFragmentBaseComponent, useExisting: forwardRef(() => IgxLineFragmentComponent) }, { provide: IgxHorizontalAnchoredCategorySeriesComponent, useExisting: forwardRef(() => IgxLineFragmentComponent) }, { provide: IgxAnchoredCategorySeriesComponent, useExisting: forwardRef(() => IgxLineFragmentComponent) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(() => IgxLineFragmentComponent) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxLineFragmentComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxLineFragmentComponent) }];
let IgxLineFragmentComponent = IgxLineFragmentComponent_1 = class IgxLineFragmentComponent extends IgxFragmentBaseComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new LineFragment();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Gets whether the current series shows an area or line shape.
    */
    get isAreaOrLine() {
        return this.i.ck;
    }
};
IgxLineFragmentComponent = IgxLineFragmentComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-line-fragment',
        template: ``,
        providers: [{ provide: IgxFragmentBaseComponent, useExisting: forwardRef(() => IgxLineFragmentComponent_1) }, { provide: IgxHorizontalAnchoredCategorySeriesComponent, useExisting: forwardRef(() => IgxLineFragmentComponent_1) }, { provide: IgxAnchoredCategorySeriesComponent, useExisting: forwardRef(() => IgxLineFragmentComponent_1) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(() => IgxLineFragmentComponent_1) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxLineFragmentComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxLineFragmentComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxLineFragmentComponent);
export { IgxLineFragmentComponent };
