// igniteui-angular-charts/esm2015/lib/igx-stacked-100-spline-area-series-component.js
var IgxStacked100SplineAreaSeriesComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxStackedSplineAreaSeriesComponent } from "./igx-stacked-spline-area-series-component";
import { IgxHorizontalStackedSeriesBaseComponent } from "./igx-horizontal-stacked-series-base-component";
import { IgxStackedSeriesBaseComponent } from "./igx-stacked-series-base-component";
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { Stacked100SplineAreaSeries } from "./Stacked100SplineAreaSeries";
/**
 * Represents a IgxDataChartComponent stacked100 spline area series.
*/
export const IgxStacked100SplineAreaSeriesComponent_PROVIDERS = [{ provide: IgxStackedSplineAreaSeriesComponent, useExisting: forwardRef(() => IgxStacked100SplineAreaSeriesComponent) }, { provide: IgxHorizontalStackedSeriesBaseComponent, useExisting: forwardRef(() => IgxStacked100SplineAreaSeriesComponent) }, { provide: IgxStackedSeriesBaseComponent, useExisting: forwardRef(() => IgxStacked100SplineAreaSeriesComponent) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(() => IgxStacked100SplineAreaSeriesComponent) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxStacked100SplineAreaSeriesComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxStacked100SplineAreaSeriesComponent) }];
let IgxStacked100SplineAreaSeriesComponent = IgxStacked100SplineAreaSeriesComponent_1 = class IgxStacked100SplineAreaSeriesComponent extends IgxStackedSplineAreaSeriesComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new Stacked100SplineAreaSeries();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
};
IgxStacked100SplineAreaSeriesComponent = IgxStacked100SplineAreaSeriesComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-stacked-100-spline-area-series',
        template: ``,
        providers: [{ provide: IgxStackedSplineAreaSeriesComponent, useExisting: forwardRef(() => IgxStacked100SplineAreaSeriesComponent_1) }, { provide: IgxHorizontalStackedSeriesBaseComponent, useExisting: forwardRef(() => IgxStacked100SplineAreaSeriesComponent_1) }, { provide: IgxStackedSeriesBaseComponent, useExisting: forwardRef(() => IgxStacked100SplineAreaSeriesComponent_1) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(() => IgxStacked100SplineAreaSeriesComponent_1) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxStacked100SplineAreaSeriesComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxStacked100SplineAreaSeriesComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxStacked100SplineAreaSeriesComponent);
export { IgxStacked100SplineAreaSeriesComponent };
