// igniteui-angular-charts/esm2015/lib/igx-radial-pie-series-component.js
var IgxRadialPieSeriesComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxAnchoredRadialSeriesComponent } from "./igx-anchored-radial-series-component";
import { IgxRadialBaseComponent } from "./igx-radial-base-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { RadialPieSeries } from "./RadialPieSeries";
/**
 * Represents a IgxDataChartComponent radial pie series.
 * Compare values across categories by using radial pie slices.
*/
export const IgxRadialPieSeriesComponent_PROVIDERS = [{ provide: IgxAnchoredRadialSeriesComponent, useExisting: forwardRef(() => IgxRadialPieSeriesComponent) }, { provide: IgxRadialBaseComponent, useExisting: forwardRef(() => IgxRadialPieSeriesComponent) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxRadialPieSeriesComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxRadialPieSeriesComponent) }];
let IgxRadialPieSeriesComponent = IgxRadialPieSeriesComponent_1 = class IgxRadialPieSeriesComponent extends IgxAnchoredRadialSeriesComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new RadialPieSeries();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Gets or sets the x-radius of the ellipse that is used to round the corners of the column.
    */
    get radiusX() {
        return this.i.ov;
    }
    set radiusX(v) {
        this.i.ov = +v;
    }
    /**
     * Gets or sets the y-radius of the ellipse that is used to round the corners of the column.
    */
    get radiusY() {
        return this.i.ow;
    }
    set radiusY(v) {
        this.i.ow = +v;
    }
    /**
     * For a category plotted series, returns the current width of the items within the categories. This only returns a value if the items have some form of width (e.g. columns, bars, etc.) otherwise 0 is returned.
    
    */
    getItemSpan() {
        let iv = this.i.eg();
        return (iv);
    }
};
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxRadialPieSeriesComponent.prototype, "radiusX", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxRadialPieSeriesComponent.prototype, "radiusY", null);
IgxRadialPieSeriesComponent = IgxRadialPieSeriesComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-radial-pie-series',
        template: ``,
        providers: [{ provide: IgxAnchoredRadialSeriesComponent, useExisting: forwardRef(() => IgxRadialPieSeriesComponent_1) }, { provide: IgxRadialBaseComponent, useExisting: forwardRef(() => IgxRadialPieSeriesComponent_1) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxRadialPieSeriesComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxRadialPieSeriesComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxRadialPieSeriesComponent);
export { IgxRadialPieSeriesComponent };
