// igniteui-angular-charts/esm2015/lib/igx-stacked-100-area-series-component.js
var IgxStacked100AreaSeriesComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxStackedAreaSeriesComponent } from "./igx-stacked-area-series-component";
import { IgxHorizontalStackedSeriesBaseComponent } from "./igx-horizontal-stacked-series-base-component";
import { IgxStackedSeriesBaseComponent } from "./igx-stacked-series-base-component";
import { IgxCategorySeriesComponent } from "./igx-category-series-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { Stacked100AreaSeries } from "./Stacked100AreaSeries";
/**
 * Represents a stacked area series, where values are presented as percentages of the total.
*/
export const IgxStacked100AreaSeriesComponent_PROVIDERS = [{ provide: IgxStackedAreaSeriesComponent, useExisting: forwardRef(() => IgxStacked100AreaSeriesComponent) }, { provide: IgxHorizontalStackedSeriesBaseComponent, useExisting: forwardRef(() => IgxStacked100AreaSeriesComponent) }, { provide: IgxStackedSeriesBaseComponent, useExisting: forwardRef(() => IgxStacked100AreaSeriesComponent) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(() => IgxStacked100AreaSeriesComponent) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxStacked100AreaSeriesComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxStacked100AreaSeriesComponent) }];
let IgxStacked100AreaSeriesComponent = IgxStacked100AreaSeriesComponent_1 = class IgxStacked100AreaSeriesComponent extends IgxStackedAreaSeriesComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new Stacked100AreaSeries();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
};
IgxStacked100AreaSeriesComponent = IgxStacked100AreaSeriesComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-stacked-100-area-series',
        template: ``,
        providers: [{ provide: IgxStackedAreaSeriesComponent, useExisting: forwardRef(() => IgxStacked100AreaSeriesComponent_1) }, { provide: IgxHorizontalStackedSeriesBaseComponent, useExisting: forwardRef(() => IgxStacked100AreaSeriesComponent_1) }, { provide: IgxStackedSeriesBaseComponent, useExisting: forwardRef(() => IgxStacked100AreaSeriesComponent_1) }, { provide: IgxCategorySeriesComponent, useExisting: forwardRef(() => IgxStacked100AreaSeriesComponent_1) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxStacked100AreaSeriesComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxStacked100AreaSeriesComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxStacked100AreaSeriesComponent);
export { IgxStacked100AreaSeriesComponent };
