// igniteui-angular-charts/esm2015/lib/igx-bollinger-band-width-indicator-component.js
var IgxBollingerBandWidthIndicatorComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxStrategyBasedIndicatorComponent } from "./igx-strategy-based-indicator-component";
import { IgxFinancialIndicatorComponent } from "./igx-financial-indicator-component";
import { IgxFinancialSeriesComponent } from "./igx-financial-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { BollingerBandWidthIndicator } from "./BollingerBandWidthIndicator";
/**
 * Represents a IgxDataChartComponent Bollinger Bandwidth indicator series.
 * Represents the normalized width of the Bollinger bands for each provided value.
 * For more info see: BollingerBandsOverlay
 * Default required members: High, Low, Close
*
* You can use the `IgxBollingerBandWidthIndicatorComponent` to represents a IgxDataChartComponent Bollinger Bandwidth indicator series.
*
* ```html
* <igx-data-chart
* 	#chart
*     [dataSource]="data" >
*     <igx-category-x-axis
* 		label="label"
* 		#xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
* 		#yAxis>
*     </igx-numeric-y-axis>
*     <igx-bollinger-band-width-indicator
* 		[xAxis]="xAxis"
* 		[yAxis]="yAxis"
* 		openMemberPath="open"
* 		highMemberPath="high"
* 		lowMemberPath="low"
* 		closeMemberPath="close">
*     </igx-bollinger-band-width-indicator>
* </igx-data-chart>
* ```
*
* ```ts
* let series = new IgxBollingerBandWidthIndicatorComponent();
* series.xAxis = this.xAxis;
* series.yAxis = this.yAxis;
* series.openMemberPath = "open";
* series.highMemberPath = "high";
* series.lowMemberPath = "low";
* series.closeMemberPath = "close";
* this.chart.series.add(series);
* ```
*/
export const IgxBollingerBandWidthIndicatorComponent_PROVIDERS = [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(() => IgxBollingerBandWidthIndicatorComponent) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(() => IgxBollingerBandWidthIndicatorComponent) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(() => IgxBollingerBandWidthIndicatorComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxBollingerBandWidthIndicatorComponent) }];
let IgxBollingerBandWidthIndicatorComponent = IgxBollingerBandWidthIndicatorComponent_1 = class IgxBollingerBandWidthIndicatorComponent extends IgxStrategyBasedIndicatorComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new BollingerBandWidthIndicator();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Gets or sets the moving average period for the Bollinger Band Width Indicator.
     * The typical, and initial, value for CCI periods is 20.
    *
    * You can use the `period` to set the moving average.
    *
    * ```html
    * <igx-data-chart
    * 	#chart
    *     [dataSource]="data" >
    *     <igx-category-x-axis
    * 		label="label"
    * 		#xAxis>
    *     </igx-category-x-axis>
    *     <igx-numeric-y-axis
    * 		#yAxis>
    *     </igx-numeric-y-axis>
    *     <igx-bollinger-band-width-indicator
    * 		[xAxis]="xAxis"
    * 		[yAxis]="yAxis"
    * 		openMemberPath="open"
    * 		highMemberPath="high"
    * 		lowMemberPath="low"
    * 		closeMemberPath="close"
    * 		period="14">
    *     </igx-bollinger-band-width-indicator>
    * </igx-data-chart>
    * ```
    *
    * ```ts
    * this.series.period = 14;
    * ```
    */
    get period() {
        return this.i.qz;
    }
    set period(v) {
        this.i.qz = +v;
    }
    /**
     * Gets or sets the multiplier for the Bollinger Band width.
     * The typical, and initial, value for Bollinger Band width multiplier is 2.
    *
    * Use the `multiplier` property for the Bollinger Band width. The typical, and initial, value for Bollinger Band width multiplier is 2.
    *
    * ```ts
    * this.series.multiplier = 14;
    * ```
    */
    get multiplier() {
        return this.i.qy;
    }
    set multiplier(v) {
        this.i.qy = +v;
    }
};
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxBollingerBandWidthIndicatorComponent.prototype, "period", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxBollingerBandWidthIndicatorComponent.prototype, "multiplier", null);
IgxBollingerBandWidthIndicatorComponent = IgxBollingerBandWidthIndicatorComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-bollinger-band-width-indicator',
        template: ``,
        providers: [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(() => IgxBollingerBandWidthIndicatorComponent_1) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(() => IgxBollingerBandWidthIndicatorComponent_1) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(() => IgxBollingerBandWidthIndicatorComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxBollingerBandWidthIndicatorComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxBollingerBandWidthIndicatorComponent);
export { IgxBollingerBandWidthIndicatorComponent };
