// igniteui-angular-charts/esm2015/lib/igx-numeric-x-axis-component.js
var IgxNumericXAxisComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, ChangeDetectionStrategy } from '@angular/core';
import { IgxStraightNumericAxisBaseComponent } from "./igx-straight-numeric-axis-base-component";
import { IgxNumericAxisBaseComponent } from "./igx-numeric-axis-base-component";
import { IgxAxisComponent } from "./igx-axis-component";
import { NumericXAxis } from "./NumericXAxis";
/**
 * Represents a IgxDataChartComponent numeric X axis.
*
* The `IgxNumericXAxisComponent` treats the data as continuously varying numerical data items. Labels on this axis are placed along the X-axis. The location of labels varies according to the value in a data column that is mapped using the `XMemberPath` property of Scatter Series or `ValueMemberPath` property of Bar Series.
*
* ```html
* <igx-data-chart
*     [dataSource]="data" >
*     <igx-numeric-x-axis
* 		label="label"
* 		#xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
* 		crossingAxis= "xAxis"
* 		crossingValue= 700
* 		#yAxis>
*     </igx-numeric-y-axis>
* </igx-data-chart>
* ```
*/
export const IgxNumericXAxisComponent_PROVIDERS = [{ provide: IgxStraightNumericAxisBaseComponent, useExisting: forwardRef(() => IgxNumericXAxisComponent) }, { provide: IgxNumericAxisBaseComponent, useExisting: forwardRef(() => IgxNumericXAxisComponent) }, { provide: IgxAxisComponent, useExisting: forwardRef(() => IgxNumericXAxisComponent) }];
let IgxNumericXAxisComponent = IgxNumericXAxisComponent_1 = class IgxNumericXAxisComponent extends IgxStraightNumericAxisBaseComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new NumericXAxis();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    scrollRangeIntoView(minimum, maximum) {
        this.i.l3(minimum, maximum);
    }
};
IgxNumericXAxisComponent = IgxNumericXAxisComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-numeric-x-axis',
        template: ``,
        providers: [{ provide: IgxStraightNumericAxisBaseComponent, useExisting: forwardRef(() => IgxNumericXAxisComponent_1) }, { provide: IgxNumericAxisBaseComponent, useExisting: forwardRef(() => IgxNumericXAxisComponent_1) }, { provide: IgxAxisComponent, useExisting: forwardRef(() => IgxNumericXAxisComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxNumericXAxisComponent);
export { IgxNumericXAxisComponent };
