// igniteui-angular-charts/esm2015/lib/igx-absolute-volume-oscillator-indicator-component.js
var IgxAbsoluteVolumeOscillatorIndicatorComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxStrategyBasedIndicatorComponent } from "./igx-strategy-based-indicator-component";
import { IgxFinancialIndicatorComponent } from "./igx-financial-indicator-component";
import { IgxFinancialSeriesComponent } from "./igx-financial-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { AbsoluteVolumeOscillatorIndicator } from "./AbsoluteVolumeOscillatorIndicator";
/**
 * Represents a IgxDataChartComponent Absolute Volume Oscillator indicator series.
 * Default required members: Volume
*
* You can use the `IgxAbsoluteVolumeOscillatorIndicatorComponent` to identify whether volume trands are increasing or decreasing.
*
* ```html
* <igx-data-chart
* 	#chart
*     [dataSource]="data" >
*     <igx-category-x-axis
* 		label="label"
* 		#xAxis>
*     </igx-category-x-axis>
*     <igx-numeric-y-axis
* 		#yAxis>
*     </igx-numeric-y-axis>
*     <igx-absolute-volume-oscillator-indicator
* 		[xAxis]="xAxis"
* 		[yAxis]="yAxis"
* 		openMemberPath="open"
* 		highMemberPath="high"
* 		lowMemberPath="low"
* 		closeMemberPath="close">
*     </igx-absolute-volume-oscillator-indicator>
* </igx-data-chart>
* ```
*/
export const IgxAbsoluteVolumeOscillatorIndicatorComponent_PROVIDERS = [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(() => IgxAbsoluteVolumeOscillatorIndicatorComponent) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(() => IgxAbsoluteVolumeOscillatorIndicatorComponent) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(() => IgxAbsoluteVolumeOscillatorIndicatorComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxAbsoluteVolumeOscillatorIndicatorComponent) }];
let IgxAbsoluteVolumeOscillatorIndicatorComponent = IgxAbsoluteVolumeOscillatorIndicatorComponent_1 = class IgxAbsoluteVolumeOscillatorIndicatorComponent extends IgxStrategyBasedIndicatorComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new AbsoluteVolumeOscillatorIndicator();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Gets or sets the short moving average period for the current AbsoluteVolumeOscillatorIndicator object.
     * The typical, and initial, value for short AVO periods is 10.
    *
    * You can use the `shortPeriod` to set the short moving average period.
    *
    * ```html
    * <igx-data-chart
    *     [dataSource]="data" >
    *     <igx-category-x-axis
    * 		label="label"
    * 		#xAxis>
    *     </igx-category-x-axis>
    *     <igx-numeric-y-axis
    * 		#yAxis>
    *     </igx-numeric-y-axis>
    *     <igx-absolute-volume-oscillator-indicator
    * 		#series
    * 		[xAxis]="xAxis"
    * 		[yAxis]="yAxis"
    * 		openMemberPath="open"
    * 		highMemberPath="high"
    * 		lowMemberPath="low"
    * 		closeMemberPath="close"
    * 		shortPeriod="10">
    *     </igx-absolute-volume-oscillator-indicator>
    * </igx-data-chart>
    * ```
    *
    * ```ts
    * this.series.shortPeriod = 10;
    * ```
    */
    get shortPeriod() {
        return this.i.shortPeriod;
    }
    set shortPeriod(v) {
        this.i.shortPeriod = +v;
    }
    /**
     * Gets or sets the short moving average period for the current AbsoluteVolumeOscillatorIndicator object.
     * The typical, and initial, value for long AVO periods is 30.
    *
    * You can use the `longPeriod` to set the long moving average period.
    *
    * ```html
    * <igx-data-chart
    *     [dataSource]="data" >
    *     <igx-category-x-axis
    * 		label="label"
    * 		#xAxis>
    *     </igx-category-x-axis>
    *     <igx-numeric-y-axis
    * 		#yAxis>
    *     </igx-numeric-y-axis>
    *     <igx-absolute-volume-oscillator-indicator
    * 		#series
    * 		[xAxis]="xAxis"
    * 		[yAxis]="yAxis"
    * 		openMemberPath="open"
    * 		highMemberPath="high"
    * 		lowMemberPath="low"
    * 		closeMemberPath="close"
    * 		longPeriod="30">
    *     </igx-absolute-volume-oscillator-indicator>
    * </igx-data-chart>
    * ```
    *
    * ```ts
    * this.series.longPeriod = 30;
    * ```
    */
    get longPeriod() {
        return this.i.longPeriod;
    }
    set longPeriod(v) {
        this.i.longPeriod = +v;
    }
};
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxAbsoluteVolumeOscillatorIndicatorComponent.prototype, "shortPeriod", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxAbsoluteVolumeOscillatorIndicatorComponent.prototype, "longPeriod", null);
IgxAbsoluteVolumeOscillatorIndicatorComponent = IgxAbsoluteVolumeOscillatorIndicatorComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-absolute-volume-oscillator-indicator',
        template: ``,
        providers: [{ provide: IgxStrategyBasedIndicatorComponent, useExisting: forwardRef(() => IgxAbsoluteVolumeOscillatorIndicatorComponent_1) }, { provide: IgxFinancialIndicatorComponent, useExisting: forwardRef(() => IgxAbsoluteVolumeOscillatorIndicatorComponent_1) }, { provide: IgxFinancialSeriesComponent, useExisting: forwardRef(() => IgxAbsoluteVolumeOscillatorIndicatorComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxAbsoluteVolumeOscillatorIndicatorComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxAbsoluteVolumeOscillatorIndicatorComponent);
export { IgxAbsoluteVolumeOscillatorIndicatorComponent };
