// igniteui-angular-charts/esm2015/lib/igx-polar-area-series-component.js
var IgxPolarAreaSeriesComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { UnknownValuePlotting, UnknownValuePlotting_$type } from "igniteui-angular-core";
import { IgxPolarLineSeriesBaseComponent } from "./igx-polar-line-series-base-component";
import { IgxPolarBaseComponent } from "./igx-polar-base-component";
import { IgxMarkerSeriesComponent } from "./igx-marker-series-component";
import { IgxSeriesComponent } from "./igx-series-component";
import { PolarAreaSeries } from "./PolarAreaSeries";
import { ensureEnum } from "igniteui-angular-core";
/**
 * Represents a IgxDataChartComponent polar area series.
*/
export const IgxPolarAreaSeriesComponent_PROVIDERS = [{ provide: IgxPolarLineSeriesBaseComponent, useExisting: forwardRef(() => IgxPolarAreaSeriesComponent) }, { provide: IgxPolarBaseComponent, useExisting: forwardRef(() => IgxPolarAreaSeriesComponent) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxPolarAreaSeriesComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxPolarAreaSeriesComponent) }];
let IgxPolarAreaSeriesComponent = IgxPolarAreaSeriesComponent_1 = class IgxPolarAreaSeriesComponent extends IgxPolarLineSeriesBaseComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new PolarAreaSeries();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Determines how unknown values will be plotted on the chart.
     * Null and Double.NaN are two examples of unknown values.
    */
    get unknownValuePlotting() {
        return this.i.pi;
    }
    set unknownValuePlotting(v) {
        this.i.pi = ensureEnum(UnknownValuePlotting_$type, v);
    }
};
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxPolarAreaSeriesComponent.prototype, "unknownValuePlotting", null);
IgxPolarAreaSeriesComponent = IgxPolarAreaSeriesComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-polar-area-series',
        template: ``,
        providers: [{ provide: IgxPolarLineSeriesBaseComponent, useExisting: forwardRef(() => IgxPolarAreaSeriesComponent_1) }, { provide: IgxPolarBaseComponent, useExisting: forwardRef(() => IgxPolarAreaSeriesComponent_1) }, { provide: IgxMarkerSeriesComponent, useExisting: forwardRef(() => IgxPolarAreaSeriesComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxPolarAreaSeriesComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxPolarAreaSeriesComponent);
export { IgxPolarAreaSeriesComponent };
