// igniteui-angular-charts/esm2015/lib/MovingAverageConvergenceDivergenceIndicator.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { StrategyBasedIndicator } from "./StrategyBasedIndicator";
import { IShortPeriodLongPeriod_$type } from "./IShortPeriodLongPeriod";
import { MovingAverageConvergenceDivergenceIndicatorStrategy } from "./MovingAverageConvergenceDivergenceIndicatorStrategy";
import { typeGetValue, markType } from "igniteui-angular-core";
import { Brush } from "igniteui-angular-core";
/**
 * @hidden
 */
export class MovingAverageConvergenceDivergenceIndicator extends StrategyBasedIndicator {
    constructor() {
        super();
        this.trendLineBrush = ((() => {
            let $ret = new Brush();
            $ret.fill = "#565656";
            return $ret;
        })());
        this.trendLineType = 10;
        this.trendLinePeriod = 9;
        this.pl = 1;
    }
    get_qd() {
        return new MovingAverageConvergenceDivergenceIndicatorStrategy();
    }
    get qd() {
        return this.get_qd();
    }
    get_qq() {
        return MovingAverageConvergenceDivergenceIndicator.$;
    }
    get qq() {
        return this.get_qq();
    }
    get shortPeriod() {
        return typeGetValue(this.c(MovingAverageConvergenceDivergenceIndicator.q1));
    }
    set shortPeriod(a) {
        this.h(MovingAverageConvergenceDivergenceIndicator.q1, a);
    }
    ql() {
        return this.shortPeriod;
    }
    get longPeriod() {
        return typeGetValue(this.c(MovingAverageConvergenceDivergenceIndicator.q0));
    }
    set longPeriod(a) {
        this.h(MovingAverageConvergenceDivergenceIndicator.q0, a);
    }
    qi() {
        return this.longPeriod;
    }
    get signalPeriod() {
        return typeGetValue(this.c(MovingAverageConvergenceDivergenceIndicator.q2));
    }
    set signalPeriod(a) {
        this.h(MovingAverageConvergenceDivergenceIndicator.q2, a);
    }
    pt() {
        return this.signalPeriod;
    }
}
MovingAverageConvergenceDivergenceIndicator.$t = markType(MovingAverageConvergenceDivergenceIndicator, 'MovingAverageConvergenceDivergenceIndicator', StrategyBasedIndicator.$, [IShortPeriodLongPeriod_$type]);
MovingAverageConvergenceDivergenceIndicator.q1 = StrategyBasedIndicator.qx(10, MovingAverageConvergenceDivergenceIndicator.$);
MovingAverageConvergenceDivergenceIndicator.q0 = StrategyBasedIndicator.qt(30, MovingAverageConvergenceDivergenceIndicator.$);
MovingAverageConvergenceDivergenceIndicator.q2 = StrategyBasedIndicator.qw(9, MovingAverageConvergenceDivergenceIndicator.$, "SignalPeriod");
