// igniteui-angular-charts/esm2015/lib/PercentagePriceOscillatorIndicator.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { StrategyBasedIndicator } from "./StrategyBasedIndicator";
import { IShortPeriodLongPeriod_$type } from "./IShortPeriodLongPeriod";
import { PercentagePriceOscillatorIndicatorStrategy } from "./PercentagePriceOscillatorIndicatorStrategy";
import { typeGetValue, markType } from "igniteui-angular-core";
/**
 * @hidden
 */
export class PercentagePriceOscillatorIndicator extends StrategyBasedIndicator {
    get_qd() {
        return new PercentagePriceOscillatorIndicatorStrategy();
    }
    get qd() {
        return this.get_qd();
    }
    get_qq() {
        return PercentagePriceOscillatorIndicator.$;
    }
    get qq() {
        return this.get_qq();
    }
    get shortPeriod() {
        return typeGetValue(this.c(PercentagePriceOscillatorIndicator.qz));
    }
    set shortPeriod(a) {
        this.h(PercentagePriceOscillatorIndicator.qz, a);
    }
    ql() {
        return this.shortPeriod;
    }
    get longPeriod() {
        return typeGetValue(this.c(PercentagePriceOscillatorIndicator.qy));
    }
    set longPeriod(a) {
        this.h(PercentagePriceOscillatorIndicator.qy, a);
    }
    qi() {
        return this.longPeriod;
    }
}
PercentagePriceOscillatorIndicator.$t = markType(PercentagePriceOscillatorIndicator, 'PercentagePriceOscillatorIndicator', StrategyBasedIndicator.$, [IShortPeriodLongPeriod_$type]);
PercentagePriceOscillatorIndicator.qz = StrategyBasedIndicator.qx(10, PercentagePriceOscillatorIndicator.$);
PercentagePriceOscillatorIndicator.qy = StrategyBasedIndicator.qt(30, PercentagePriceOscillatorIndicator.$);
