// igniteui-angular-charts/esm2015/lib/ScatterSeries_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ScatterBase } from "./ScatterBase";
import { markType } from "igniteui-angular-core";
import { ScatterFrame } from "./ScatterFrame";
import { ScatterBaseView } from "./ScatterBaseView";
import { Color } from "igniteui-angular-core";
/**
 * @hidden
 */
export class ScatterSeries extends ScatterBase {
    constructor() {
        super();
        this.ab = ScatterSeries.$;
        this.nd = new ScatterFrame();
        this.nf = new ScatterFrame();
        this.nc = new ScatterFrame();
    }
    bf() {
        return new ScatterSeriesView(this);
    }
    iv(a) {
        super.iv(a);
        this.o1 = a;
    }
    get_cy() {
        return false;
    }
}
ScatterSeries.$t = markType(ScatterSeries, 'ScatterSeries', ScatterBase.$);
/**
 * @hidden
 */
export class ScatterSeriesView extends ScatterBaseView {
    constructor(a) {
        super(a);
        this.da = null;
        this.da = a;
    }
    ap() {
        super.ap();
        if (!this.n) {
            this.by.l2 = 2;
            this.e.bj = 4;
        }
    }
    ac() {
        let a = new Color();
        a.colorString = "rgba(95,95,95,0.5)";
        this.e.lb = a;
        this.e.ev = 3;
        this.e.ew = 2;
        this.e.ex = 2;
        this.e.du = false;
    }
}
ScatterSeriesView.$t = markType(ScatterSeriesView, 'ScatterSeriesView', ScatterBaseView.$);
