// igniteui-angular-charts/esm2015/lib/ScatterPolylineSeries_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ShapeSeriesBase } from "./ShapeSeriesBase";
import { StyleSelector } from "./StyleSelector";
import { DependencyProperty } from "igniteui-angular-core";
import { Style } from "igniteui-angular-core";
import { markType, markDep, typeCast } from "igniteui-angular-core";
import { ShapeSeriesViewBase } from "./ShapeSeriesViewBase";
import { PropertyMetadata } from "igniteui-angular-core";
import { Path } from "igniteui-angular-core";
import { stringIsNullOrEmpty } from "igniteui-angular-core";
/**
 * @hidden
 */
export class ScatterPolylineSeries extends ShapeSeriesBase {
    constructor() {
        super();
        this.ab = ScatterPolylineSeries.$;
    }
    bf() {
        return new ScatterPolylineSeriesView(this);
    }
    iv(a) {
        super.iv(a);
        this.nc = a;
    }
    get_mg() {
        return false;
    }
    i0(a, b, c, d) {
        super.i0(a, b, c, d);
        switch (b) {
            case ScatterPolylineSeries.$$p[0]:
                this.nc.cn(this.ni);
                this.je(false);
                this.il();
                break;
            case ScatterPolylineSeries.$$p[1]:
                this.nc.co(this.nd);
                this.je(false);
                this.il();
                break;
        }
    }
    m1(a) {
        let b = a;
        b.cn(this.ni);
        b.co(this.nd);
        super.m1(a);
    }
    ar(a, b) {
        let c = this.nc.cs(a, b, true);
        if (c != null) {
            return c.dataContext;
        }
        return super.ar(a, b);
    }
}
ScatterPolylineSeries.$t = markType(ScatterPolylineSeries, 'ScatterPolylineSeries', ShapeSeriesBase.$);
ScatterPolylineSeries.$$p = markDep(DependencyProperty, PropertyMetadata, ScatterPolylineSeries, 'raisePropertyChanged', ['ShapeStyle:ni:ng', [Style.$, null], 'ShapeStyleSelector:nd:nh', [StyleSelector.$, null]]);
/**
 * @hidden
 */
export class ScatterPolylineSeriesView extends ShapeSeriesViewBase {
    constructor(a) {
        super(a);
    }
    cu(a, b) {
        return this.cv(a, b, false);
    }
    cc(a, b) {
        super.cc(a, b);
        ShapeSeriesViewBase.cb(this, a, b);
    }
    ca(a, b) {
        let c = typeCast(Path.$, a);
        if (c == null) {
            return;
        }
        c.ak = b;
    }
    bb(a, b) {
        super.bb(a, b);
        let c = a;
        c._fill = null;
    }
    bc(a, b) {
        super.bc(a, b);
        let c = a;
        c._fill = null;
    }
    b2(a) {
        return a.width >= this.by.mq || a.height >= this.by.mq;
    }
    getDefaultTooltipTemplate() {
        let a = "<div class='ui-chart-default-tooltip-content'><span";
        let b = this.e.a0.a$q(this.e);
        if (!stringIsNullOrEmpty(b)) {
            a += " style='color:" + b + "'";
        }
        a += ">" + this.by.fx + "</span><br/>";
        a += "</div>";
        return a;
    }
}
ScatterPolylineSeriesView.$t = markType(ScatterPolylineSeriesView, 'ScatterPolylineSeriesView', ShapeSeriesViewBase.$);
