// igniteui-angular-charts/esm2015/lib/SlowStochasticOscillatorIndicator.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { StrategyBasedIndicator } from "./StrategyBasedIndicator";
import { SlowStochasticOscillatorIndicatorStrategy } from "./SlowStochasticOscillatorIndicatorStrategy";
import { typeGetValue, markType } from "igniteui-angular-core";
import { Brush } from "igniteui-angular-core";
/**
 * @hidden
 */
export class SlowStochasticOscillatorIndicator extends StrategyBasedIndicator {
    constructor() {
        super();
        this.trendLineBrush = ((() => {
            let $ret = new Brush();
            $ret.fill = "#565656";
            return $ret;
        })());
        this.trendLineType = 10;
        this.trendLinePeriod = 3;
    }
    get_qd() {
        return new SlowStochasticOscillatorIndicatorStrategy();
    }
    get qd() {
        return this.get_qd();
    }
    get_qq() {
        return SlowStochasticOscillatorIndicator.$;
    }
    get qq() {
        return this.get_qq();
    }
    get qy() {
        return typeGetValue(this.c(SlowStochasticOscillatorIndicator.qz));
    }
    set qy(a) {
        this.h(SlowStochasticOscillatorIndicator.qz, a);
    }
    qj() {
        return this.qy;
    }
    qs(a) {
        this.qy = a;
    }
}
SlowStochasticOscillatorIndicator.$t = markType(SlowStochasticOscillatorIndicator, 'SlowStochasticOscillatorIndicator', StrategyBasedIndicator.$);
SlowStochasticOscillatorIndicator.qz = StrategyBasedIndicator.qv(14, SlowStochasticOscillatorIndicator.$);
