// igniteui-angular-charts/esm2015/lib/StochRSIIndicator.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { StrategyBasedIndicator } from "./StrategyBasedIndicator";
import { StochRSIIndicatorStrategy } from "./StochRSIIndicatorStrategy";
import { typeGetValue, markType } from "igniteui-angular-core";
/**
 * @hidden
 */
export class StochRSIIndicator extends StrategyBasedIndicator {
    get_qd() {
        return new StochRSIIndicatorStrategy();
    }
    get qd() {
        return this.get_qd();
    }
    get_qq() {
        return StochRSIIndicator.$;
    }
    get qq() {
        return this.get_qq();
    }
    get qy() {
        return typeGetValue(this.c(StochRSIIndicator.qz));
    }
    set qy(a) {
        this.h(StochRSIIndicator.qz, a);
    }
    qj() {
        return this.qy;
    }
    qs(a) {
        this.qy = a;
    }
}
StochRSIIndicator.$t = markType(StochRSIIndicator, 'StochRSIIndicator', StrategyBasedIndicator.$);
StochRSIIndicator.qz = StrategyBasedIndicator.qv(14, StochRSIIndicator.$);
