// igniteui-angular-charts/esm2015/lib/XamPieChart_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { PieChartBase } from "./PieChartBase";
import { PieChartBaseView } from "./PieChartBaseView";
import { markType } from "igniteui-angular-core";
/**
 * @hidden
 */
export class XamPieChart extends PieChartBase {
    constructor() {
        super();
        this.h3 = null;
        this.ab = XamPieChart.$;
    }
    av() {
        return new XamPieChartView(this);
    }
    fq(a) {
        super.fq(a);
        this.h3 = a;
    }
}
XamPieChart.$t = markType(XamPieChart, 'XamPieChart', PieChartBase.$);
/**
 * @hidden
 */
export class XamPieChartView extends PieChartBaseView {
    constructor(a) {
        super(a);
        this.b4 = null;
        this.b4 = a;
    }
}
XamPieChartView.$t = markType(XamPieChartView, 'XamPieChartView', PieChartBaseView.$);
