// igniteui-angular-core/esm2015/lib/CellActionManagerDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export class CellActionManagerDescription extends Description {
    constructor() {
        super();
        this.g = false;
        this.f = false;
        this.h = false;
    }
    get_type() {
        return "CellActionManager";
    }
    get type() {
        return this.get_type();
    }
    get isLeaveActionDesired() {
        return this.g;
    }
    set isLeaveActionDesired(a) {
        this.g = a;
        this.e("IsLeaveActionDesired");
    }
    get isControlPressed() {
        return this.f;
    }
    set isControlPressed(a) {
        this.f = a;
        this.e("IsControlPressed");
    }
    get isShiftPressed() {
        return this.h;
    }
    set isShiftPressed(a) {
        this.h = a;
        this.e("IsShiftPressed");
    }
}
CellActionManagerDescription.$t = markType(CellActionManagerDescription, 'CellActionManagerDescription', Description.$);
