// igniteui-angular-core/esm2015/lib/EqualityComparer$1_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, IEqualityComparer_$type, IEqualityComparer$1_$type, typeCastObjTo$t, markType, getBoxIfEnum } from "./type";
/**
 * @hidden
 */
export class EqualityComparer$1 extends Base {
    constructor($t) {
        super();
        this.$t = null;
        this.$t = $t;
        this.$type = this.$type.specialize(this.$t);
    }
    static defaultEqualityComparerValue($t) {
        return new DefaultEqualityComparer$1($t);
    }
    equalsC(a, b) {
        return this.equalsC(typeCastObjTo$t(this.$t, a), typeCastObjTo$t(this.$t, b));
    }
    getHashCodeC(a) {
        return this.getHashCodeC(typeCastObjTo$t(this.$t, a));
    }
}
EqualityComparer$1.$t = markType(EqualityComparer$1, 'EqualityComparer$1', Base.$, [IEqualityComparer_$type, IEqualityComparer$1_$type.specialize(0)]);
/**
 * @hidden
 */
export class DefaultEqualityComparer$1 extends EqualityComparer$1 {
    constructor($t) {
        super($t);
        this.$t = null;
        this.$t = $t;
        this.$type = this.$type.specialize(this.$t);
    }
    equalsC(a, b) {
        return Base.equalsStatic(getBoxIfEnum(this.$t, a), getBoxIfEnum(this.$t, b));
    }
    getHashCodeC(a) {
        return Base.getHashCodeStatic(a);
    }
}
DefaultEqualityComparer$1.$t = markType(DefaultEqualityComparer$1, 'DefaultEqualityComparer$1', EqualityComparer$1.$.specialize(0));
