// igniteui-angular-core/esm2015/lib/ExpressionSyntax_combined.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Base, markType } from "./type";
/**
 * @hidden
 */
export class ExpressionSyntax extends Base {
    static c(a, b, c, ...d) {
        return null;
    }
    static a(a, ...b) {
        return null;
    }
    static b(a, b) {
        return null;
    }
    static d(a) {
        return null;
    }
    static e(a, b) {
        return null;
    }
}
ExpressionSyntax.$t = markType(ExpressionSyntax, 'ExpressionSyntax');
/**
 * @hidden
 */
export class LambdaExpression extends ExpressionSyntax {
}
LambdaExpression.$t = markType(LambdaExpression, 'LambdaExpression', ExpressionSyntax.$);
/**
 * @hidden
 */
export class MemberExpression extends ExpressionSyntax {
    constructor() {
        super(...arguments);
        this.f = null;
    }
    g(a) {
        return null;
    }
}
MemberExpression.$t = markType(MemberExpression, 'MemberExpression', ExpressionSyntax.$);
/**
 * @hidden
 */
export class MethodCallExpression extends ExpressionSyntax {
    constructor() {
        super(...arguments);
        this.f = null;
        this.g = null;
    }
    h(a, b) {
        return null;
    }
}
MethodCallExpression.$t = markType(MethodCallExpression, 'MethodCallExpression', ExpressionSyntax.$);
/**
 * @hidden
 */
export class ParameterExpression extends ExpressionSyntax {
    constructor() {
        super(...arguments);
        this.f = false;
        this.g = null;
    }
}
ParameterExpression.$t = markType(ParameterExpression, 'ParameterExpression', ExpressionSyntax.$);
