// igniteui-angular-core/esm2015/lib/ScatterAreaSeriesDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { ScatterTriangulationSeriesDescription } from "./ScatterTriangulationSeriesDescription";
import { markType } from "./type";
/**
 * @hidden
 */
export class ScatterAreaSeriesDescription extends ScatterTriangulationSeriesDescription {
    constructor() {
        super();
        this.cg = null;
        this.ci = null;
    }
    get_type() {
        return "ScatterAreaSeries";
    }
    get colorScale() {
        return this.cg;
    }
    set colorScale(a) {
        this.cg = a;
        this.e("ColorScale");
    }
    get colorMemberPath() {
        return this.ci;
    }
    set colorMemberPath(a) {
        this.ci = a;
        this.e("ColorMemberPath");
    }
}
ScatterAreaSeriesDescription.$t = markType(ScatterAreaSeriesDescription, 'ScatterAreaSeriesDescription', ScatterTriangulationSeriesDescription.$);
