// igniteui-angular-core/esm2015/lib/TimeAxisLabelFormatDescription.js
/*
THIS INFRAGISTICS ULTIMATE SOFTWARE LICENSE  AGREEMENT ("AGREEMENT") LOCATED HERE:
https://www.infragistics.com/legal/license/igultimate-la
https://www.infragistics.com/legal/license/igultimate-eula
GOVERNS THE LICENSING, INSTALLATION AND USE OF INFRAGISTICS SOFTWARE. BY DOWNLOADING AND/OR INSTALLING AND USING INFRAGISTICS SOFTWARE:  you are indicating that you have read and understand this Agreement, and agree to be legally bound by it on behalf of the yourself and your company.
*/
import { Description } from "./Description";
import { markType } from "./type";
/**
 * @hidden
 */
export class TimeAxisLabelFormatDescription extends Description {
    constructor() {
        super();
        this.f = 0;
        this.h = null;
    }
    get_type() {
        return "TimeAxisLabelFormat";
    }
    get type() {
        return this.get_type();
    }
    get range() {
        return this.f;
    }
    set range(a) {
        this.f = a;
        this.e("Range");
    }
    get format() {
        return this.h;
    }
    set format(a) {
        this.h = a;
        this.e("Format");
    }
}
TimeAxisLabelFormatDescription.$t = markType(TimeAxisLabelFormatDescription, 'TimeAxisLabelFormatDescription', Description.$);
