// igniteui-angular/esm2015/lib/grids/common/enums.js
import { mkenum } from '../../core/utils';
export const FilterMode = mkenum({
    quickFilter: 'quickFilter',
    excelStyleFilter: 'excelStyleFilter'
});
export const GridSummaryPosition = mkenum({
    top: 'top',
    bottom: 'bottom'
});
export const GridSummaryCalculationMode = mkenum({
    rootLevelOnly: 'rootLevelOnly',
    childLevelsOnly: 'childLevelsOnly',
    rootAndChildLevels: 'rootAndChildLevels'
});
export const GridSelectionMode = mkenum({
    none: 'none',
    single: 'single',
    multiple: 'multiple'
});
export const ColumnDisplayOrder = mkenum({
    Alphabetical: 'Alphabetical',
    DisplayOrder: 'DisplayOrder'
});
export var ColumnPinningPosition;
(function (ColumnPinningPosition) {
    ColumnPinningPosition[ColumnPinningPosition["Start"] = 0] = "Start";
    ColumnPinningPosition[ColumnPinningPosition["End"] = 1] = "End";
})(ColumnPinningPosition || (ColumnPinningPosition = {}));
export var RowPinningPosition;
(function (RowPinningPosition) {
    RowPinningPosition[RowPinningPosition["Top"] = 0] = "Top";
    RowPinningPosition[RowPinningPosition["Bottom"] = 1] = "Bottom";
})(RowPinningPosition || (RowPinningPosition = {}));
export var GridPagingMode;
(function (GridPagingMode) {
    GridPagingMode[GridPagingMode["local"] = 0] = "local";
    GridPagingMode[GridPagingMode["remote"] = 1] = "remote";
})(GridPagingMode || (GridPagingMode = {}));
