// igniteui-angular-maps/esm5/lib/igx-geographic-xy-triangulating-series-component.js
import * as tslib_1 from "tslib";
import { forwardRef, Input } from '@angular/core';
import { IgxGeographicMapSeriesHostComponent } from "./igx-geographic-map-series-host-component";
import { IgxSeriesComponent } from "igniteui-angular-charts";
import { IgxTriangulationDataSource } from "igniteui-angular-core";
/**
 * Base class for geographic series which triangulate XY data prior to rendering.
*/
export var IgxGeographicXYTriangulatingSeriesComponent_PROVIDERS = [{ provide: IgxGeographicMapSeriesHostComponent, useExisting: forwardRef(function () { return IgxGeographicXYTriangulatingSeriesComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxGeographicXYTriangulatingSeriesComponent; }) }];
var IgxGeographicXYTriangulatingSeriesComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxGeographicXYTriangulatingSeriesComponent, _super);
    function IgxGeographicXYTriangulatingSeriesComponent() {
        return _super.call(this) || this;
    }
    Object.defineProperty(IgxGeographicXYTriangulatingSeriesComponent.prototype, "i", {
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxGeographicXYTriangulatingSeriesComponent.prototype, "triangulationDataSource", {
        get: function () {
            return this._triangulationDataSource;
        },
        /**
         * Gets or sets the string path specifying the location of the Itf file.
        */
        set: function (value) {
            this._triangulationDataSource = value;
            this.updateTriangulationDataSource();
        },
        enumerable: true,
        configurable: true
    });
    IgxGeographicXYTriangulatingSeriesComponent.prototype.updateTriangulationDataSource = function () {
        if (this._cachedTriangulationDataSource == null) {
            this._cachedTriangulationDataSource = new IgxTriangulationDataSource();
            this.onTriangulationDataSourceImportCompleted = this.onTriangulationDataSourceImportCompleted.bind(this);
            this._cachedTriangulationDataSource.importCompleted.subscribe(this.onTriangulationDataSourceImportCompleted);
        }
        this._cachedTriangulationDataSource.source = this._triangulationDataSource;
    };
    IgxGeographicXYTriangulatingSeriesComponent.prototype.onTriangulationDataSourceImportCompleted = function (s, e) {
        this.dataSource = this._cachedTriangulationDataSource.getPointData();
        this.trianglesSource = this._cachedTriangulationDataSource.getTriangleData();
    };
    Object.defineProperty(IgxGeographicXYTriangulatingSeriesComponent.prototype, "longitudeMemberPath", {
        /**
     * The name of the property from which to extract the Longitude for each item in the ItemsSource.
    */
        get: function () {
            return this.i.mf;
        },
        set: function (v) {
            this.i.mf = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxGeographicXYTriangulatingSeriesComponent.prototype, "latitudeMemberPath", {
        /**
         * The name of the property from which to extract the Latitude for each item in the ItemsSource.
        */
        get: function () {
            return this.i.md;
        },
        set: function (v) {
            this.i.md = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxGeographicXYTriangulatingSeriesComponent.prototype, "trianglesSource", {
        /**
         * The source of triangulation data.
         * This property is optional.  If it is left as null, the triangulation will be created based on the items in the ItemsSource.  Triangulation is a demanding operation, so the runtime performance will be better when specifying a TriangulationSource, especially when a large number of data items are present.
        */
        get: function () {
            return this.i.mc;
        },
        set: function (v) {
            this.i.mc = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxGeographicXYTriangulatingSeriesComponent.prototype, "triangleVertexMemberPath1", {
        /**
         * The name of the property of the TrianglesSource items which, for each triangle, contains the index of the first vertex point in the ItemsSource.
        */
        get: function () {
            return this.i.mi;
        },
        set: function (v) {
            this.i.mi = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxGeographicXYTriangulatingSeriesComponent.prototype, "triangleVertexMemberPath2", {
        /**
         * The name of the property of the TrianglesSource items which, for each triangle, contains the index of the second vertex point in the ItemsSource.
        */
        get: function () {
            return this.i.mk;
        },
        set: function (v) {
            this.i.mk = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxGeographicXYTriangulatingSeriesComponent.prototype, "triangleVertexMemberPath3", {
        /**
         * The name of the property of the TrianglesSource items which, for each triangle, contains the index of the third vertex point in the ItemsSource.
        */
        get: function () {
            return this.i.mm;
        },
        set: function (v) {
            this.i.mm = v;
        },
        enumerable: true,
        configurable: true
    });
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxGeographicXYTriangulatingSeriesComponent.prototype, "triangulationDataSource", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxGeographicXYTriangulatingSeriesComponent.prototype, "longitudeMemberPath", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxGeographicXYTriangulatingSeriesComponent.prototype, "latitudeMemberPath", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Array),
        tslib_1.__metadata("design:paramtypes", [Array])
    ], IgxGeographicXYTriangulatingSeriesComponent.prototype, "trianglesSource", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxGeographicXYTriangulatingSeriesComponent.prototype, "triangleVertexMemberPath1", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxGeographicXYTriangulatingSeriesComponent.prototype, "triangleVertexMemberPath2", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxGeographicXYTriangulatingSeriesComponent.prototype, "triangleVertexMemberPath3", null);
    return IgxGeographicXYTriangulatingSeriesComponent;
}(IgxGeographicMapSeriesHostComponent));
export { IgxGeographicXYTriangulatingSeriesComponent };
