// igniteui-angular-maps/esm5/lib/igx-geographic-polyline-series-component.js
import * as tslib_1 from "tslib";
import { EventEmitter, Output, Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { delegateCombine } from "igniteui-angular-core";
import { IgxStyleShapeEventArgs } from "igniteui-angular-charts";
import { IgxGeographicShapeSeriesBaseComponent } from "./igx-geographic-shape-series-base-component";
import { IgxGeographicMapSeriesHostComponent } from "./igx-geographic-map-series-host-component";
import { IgxSeriesComponent } from "igniteui-angular-charts";
import { GeographicPolylineSeries } from "./GeographicPolylineSeries";
import { Style } from "igniteui-angular-core";
/**
 * Series class used for displaying multiple polylines, in a geographic context, for datasources which contain multiple lists of points.
*/
export var IgxGeographicPolylineSeriesComponent_PROVIDERS = [{ provide: IgxGeographicShapeSeriesBaseComponent, useExisting: forwardRef(function () { return IgxGeographicPolylineSeriesComponent; }) }, { provide: IgxGeographicMapSeriesHostComponent, useExisting: forwardRef(function () { return IgxGeographicPolylineSeriesComponent; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxGeographicPolylineSeriesComponent; }) }];
var IgxGeographicPolylineSeriesComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgxGeographicPolylineSeriesComponent, _super);
    function IgxGeographicPolylineSeriesComponent() {
        var _this = _super.call(this) || this;
        _this._styleShape = null;
        return _this;
    }
    IgxGeographicPolylineSeriesComponent_1 = IgxGeographicPolylineSeriesComponent;
    IgxGeographicPolylineSeriesComponent.prototype.createImplementation = function () {
        return new GeographicPolylineSeries();
    };
    Object.defineProperty(IgxGeographicPolylineSeriesComponent.prototype, "i", {
        /**
                                     * @hidden
                                     */
        get: function () {
            return this._implementation;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxGeographicPolylineSeriesComponent.prototype, "shapeFill", {
        /**
         * Gets or sets the fill brush.
        */
        get: function () {
            return this.i.mw ? this.i.mw.fill : null;
        },
        set: function (v) {
            this.ensureShapeStyle();
            this.i.mw.fill = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxGeographicPolylineSeriesComponent.prototype, "shapeStroke", {
        /**
         * Gets or sets the stroke brush.
        */
        get: function () {
            return this.i.mw ? this.i.mw.stroke : null;
        },
        set: function (v) {
            this.ensureShapeStyle();
            this.i.mw.stroke = v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxGeographicPolylineSeriesComponent.prototype, "shapeStrokeThickness", {
        /**
         * Gets or sets the stroke thickness.
        */
        get: function () {
            return this.i.mw ? this.i.mw.strokeThickness : NaN;
        },
        set: function (v) {
            this.ensureShapeStyle();
            this.i.mw.strokeThickness = +v;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(IgxGeographicPolylineSeriesComponent.prototype, "shapeOpacity", {
        /**
         * Gets or sets the opacity.
        */
        get: function () {
            return this.i.mw ? this.i.mw.opacity : NaN;
        },
        set: function (v) {
            this.ensureShapeStyle();
            this.i.mw.opacity = +v;
        },
        enumerable: true,
        configurable: true
    });
    IgxGeographicPolylineSeriesComponent.prototype.ensureShapeStyle = function () {
        if (this.i.mw) {
            return;
        }
        this.i.mw = new Style();
    };
    Object.defineProperty(IgxGeographicPolylineSeriesComponent.prototype, "styleShape", {
        get: function () {
            var _this = this;
            if (this._styleShape == null) {
                this._styleShape = new EventEmitter();
                this.i.styleShape = delegateCombine(this.i.styleShape, function (o, e) {
                    _this._runInZone(function () {
                        var outerArgs = new IgxStyleShapeEventArgs();
                        outerArgs._provideImplementation(e);
                        if (_this.beforeStyleShape) {
                            _this.beforeStyleShape(_this, outerArgs);
                        }
                        _this._styleShape.emit({
                            sender: _this,
                            args: outerArgs
                        });
                    });
                });
            }
            return this._styleShape;
        },
        enumerable: true,
        configurable: true
    });
    var IgxGeographicPolylineSeriesComponent_1;
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxGeographicPolylineSeriesComponent.prototype, "shapeFill", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], IgxGeographicPolylineSeriesComponent.prototype, "shapeStroke", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxGeographicPolylineSeriesComponent.prototype, "shapeStrokeThickness", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Number),
        tslib_1.__metadata("design:paramtypes", [Number])
    ], IgxGeographicPolylineSeriesComponent.prototype, "shapeOpacity", null);
    tslib_1.__decorate([
        Output(),
        tslib_1.__metadata("design:type", EventEmitter),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxGeographicPolylineSeriesComponent.prototype, "styleShape", null);
    IgxGeographicPolylineSeriesComponent = IgxGeographicPolylineSeriesComponent_1 = tslib_1.__decorate([
        Component({
            selector: 'igx-geographic-polyline-series',
            template: "",
            providers: [{ provide: IgxGeographicShapeSeriesBaseComponent, useExisting: forwardRef(function () { return IgxGeographicPolylineSeriesComponent_1; }) }, { provide: IgxGeographicMapSeriesHostComponent, useExisting: forwardRef(function () { return IgxGeographicPolylineSeriesComponent_1; }) }, { provide: IgxSeriesComponent, useExisting: forwardRef(function () { return IgxGeographicPolylineSeriesComponent_1; }) }],
            changeDetection: ChangeDetectionStrategy.OnPush
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], IgxGeographicPolylineSeriesComponent);
    return IgxGeographicPolylineSeriesComponent;
}(IgxGeographicShapeSeriesBaseComponent));
export { IgxGeographicPolylineSeriesComponent };
