// igniteui-angular/esm2015/lib/grids/toolbar/common.js
import { Directive, HostBinding, TemplateRef } from '@angular/core';
// tslint:disable-next-line: directive-selector
export class IgxExcelTextDirective {
}
IgxExcelTextDirective.decorators = [
    { type: Directive, args: [{ selector: '[excelText],excel-text' },] }
];
// tslint:disable-next-line: directive-selector
export class IgxCSVTextDirective {
}
IgxCSVTextDirective.decorators = [
    { type: Directive, args: [{ selector: '[csvText],csv-text' },] }
];
/**
 * Provides a way to template the title portion of the toolbar in the grid.
 *
 * @igxModule IgxGridToolbarModule
 * @igxParent IgxGridToolbarComponent
 *
 * @example
 * ```html
 * <igx-grid-toolbar-title>My custom title</igx-grid-toolbar-title>
 * ```
 */
export class IgxGridToolbarTitleDirective {
    constructor() {
        /**
         * Host `class.igx-grid-toolbar__title` binding.
         * @hidden
         * @internal
         */
        this.cssClass = 'igx-grid-toolbar__title';
    }
}
IgxGridToolbarTitleDirective.decorators = [
    { type: Directive, args: [{ selector: '[igxGridToolbarTitle],igx-grid-toolbar-title' },] }
];
IgxGridToolbarTitleDirective.propDecorators = {
    cssClass: [{ type: HostBinding, args: ['class.igx-grid-toolbar__title',] }]
};
/**
 * Provides a way to template the action portion of the toolbar in the grid.
 *
 * @igxModule IgxGridToolbarModule
 * @igxParent IgxGridToolbarComponent
 *
 * @example
 * ```html
 * <igx-grid-toolbar-actions>
 *  <some-toolbar-action-here />
 * </igx-grid-toolbar-actions>
 * ```
 */
export class IgxGridToolbarActionsDirective {
    constructor() {
        /**
         * Host `class.igx-grid-toolbar__actions` binding.
         * @hidden
         * @internal
         */
        this.cssClass = 'igx-grid-toolbar__actions';
    }
}
IgxGridToolbarActionsDirective.decorators = [
    { type: Directive, args: [{ selector: '[igxGridToolbarActions],igx-grid-toolbar-actions' },] }
];
IgxGridToolbarActionsDirective.propDecorators = {
    cssClass: [{ type: HostBinding, args: ['class.igx-grid-toolbar__actions',] }]
};
export class IgxGridToolbarDirective {
    constructor(template) {
        this.template = template;
    }
}
IgxGridToolbarDirective.decorators = [
    { type: Directive, args: [{ selector: '[igxGridToolbar]' },] }
];
IgxGridToolbarDirective.ctorParameters = () => [
    { type: TemplateRef }
];
