// igniteui-angular-maps/esm2015/lib/igx-tile-series-component.js
var IgxTileSeriesComponent_1;
import * as tslib_1 from "tslib";
import { Component, forwardRef, Input, ChangeDetectionStrategy } from '@angular/core';
import { IgxGeographicMapImagery } from "./igx-geographic-map-imagery";
import { IgxShapeSeriesBaseComponent } from "igniteui-angular-charts";
import { IgxSeriesComponent } from "igniteui-angular-charts";
import { TileSeries } from "./TileSeries";
/**
 * Represents a tile series class that can render imagery tiles
*/
export const IgxTileSeriesComponent_PROVIDERS = [{ provide: IgxShapeSeriesBaseComponent, useExisting: forwardRef(() => IgxTileSeriesComponent) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxTileSeriesComponent) }];
let IgxTileSeriesComponent = IgxTileSeriesComponent_1 = class IgxTileSeriesComponent extends IgxShapeSeriesBaseComponent {
    constructor() {
        super();
    }
    createImplementation() {
        return new TileSeries();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * The MapImagery which is used to provide the tiles for display.
    */
    get tileImagery() {
        if (this.i.ng == null) {
            return null;
        }
        return this.i.ng.externalObject;
    }
    set tileImagery(v) {
        v == null ? this.i.ng = null : this.i.ng = v.i;
    }
    findByName(name) {
        var baseResult = super.findByName(name);
        if (baseResult) {
            return baseResult;
        }
        if (this.tileImagery && this.tileImagery.name && this.tileImagery.name == name) {
            return this.tileImagery;
        }
        return null;
    }
    /**
     * Defers rendering of the series
    
    */
    deferredRefresh() {
        this.i.deferredRefresh();
    }
};
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", IgxGeographicMapImagery),
    tslib_1.__metadata("design:paramtypes", [IgxGeographicMapImagery])
], IgxTileSeriesComponent.prototype, "tileImagery", null);
IgxTileSeriesComponent = IgxTileSeriesComponent_1 = tslib_1.__decorate([
    Component({
        selector: 'igx-tile-series',
        template: ``,
        providers: [{ provide: IgxShapeSeriesBaseComponent, useExisting: forwardRef(() => IgxTileSeriesComponent_1) }, { provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxTileSeriesComponent_1) }],
        changeDetection: ChangeDetectionStrategy.OnPush
    }),
    tslib_1.__metadata("design:paramtypes", [])
], IgxTileSeriesComponent);
export { IgxTileSeriesComponent };
