// igniteui-angular-maps/esm2015/lib/igx-geographic-map-series-host-component.js
import * as tslib_1 from "tslib";
import { forwardRef, Input } from '@angular/core';
import { IgxSeriesComponent } from "igniteui-angular-charts";
import { toPoint } from "igniteui-angular-core";
/**
 * Base class for hosting chart series in a IgxGeographicMap.Component
*/
export const IgxGeographicMapSeriesHostComponent_PROVIDERS = [{ provide: IgxSeriesComponent, useExisting: forwardRef(() => IgxGeographicMapSeriesHostComponent) }];
export class IgxGeographicMapSeriesHostComponent extends IgxSeriesComponent {
    constructor() {
        super();
    }
    /**
                                 * @hidden
                                 */
    get i() {
        return this._implementation;
    }
    /**
     * Checks if this control is geographic coordinate system
    */
    get isGeographic() {
        return this.i.ct;
    }
    /**
     * The minimum scale at which this series becomes visible.
     * The default value for this property is 1.0, which means the series will always be visible.  At a VisibleFromScale setting of 0.0, the series will never be visible.  At a VisibleFromScale setting of 0.5, the series will be visible as long as the map is zoomed in to at least 200%.
    */
    get visibleFromScale() {
        return this.i.l5;
    }
    set visibleFromScale(v) {
        this.i.l5 = +v;
    }
    /**
     * Gets or sets the coercion methods to use when loading data from data sources.
     * Should be specified before setting any member paths, if being used. Setting it later
     * will not cause data to be reimported into the chart.
    */
    get coercionMethods() {
        return this.i.coercionMethods;
    }
    set coercionMethods(v) {
        this.i.coercionMethods = v;
    }
    getItemValue(item, memberPathName) {
        let iv = this.i.fs(item, memberPathName);
        return (iv);
    }
    /**
     * Gets the item that is the best match for the specified world coordinates.
    
    * @param world  * The world coordinates to use.
    */
    getItem(world) {
        let iv = this.i.fp(toPoint(world));
        return (iv);
    }
    /**
     * Renders the series.
    
    * @param animate  * True if the change should be animated.
    */
    renderSeries(animate) {
        this.i.je(animate);
    }
    /**
     * Called when this series' Style is updated.
    
    */
    styleUpdated() {
        this.i.jn();
    }
}
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Number),
    tslib_1.__metadata("design:paramtypes", [Number])
], IgxGeographicMapSeriesHostComponent.prototype, "visibleFromScale", null);
tslib_1.__decorate([
    Input(),
    tslib_1.__metadata("design:type", Object),
    tslib_1.__metadata("design:paramtypes", [Object])
], IgxGeographicMapSeriesHostComponent.prototype, "coercionMethods", null);
